{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3d9cd327",
   "metadata": {},
   "source": [
    "<p style='text-align: justify;'>\n",
    "<b>Author</b>: Abzer Kelminal (abzer.shah@uni-tuebingen.de) adapted from the scripts of Madeleine Ernst (mernst@ucsd.edu), Ricardo Silva (ridasilva@ucsd.edu) <br>\n",
    "<b>Edited by</b>: - <br>\n",
    "<b>Use case</b>: Test for association between paired samples (feature vs metadata and feature vs feature), using the Pearson's product moment correlation coefficient. The Notebook combines the asv and feature tables and creates a decoy set, calculates the correlation scores for both target and decoy tables. Finally, FDR is also calculated<br>\n",
    "<b>Input file format</b>: .csv tables of asv files and feature table. <br>\n",
    "<b>Outputs</b>: .csv table showing correlation coefficient values  <br>\n",
    "<b>Dependencies</b>: R version 3.4.2 (2017-09-28) and libraries  ggplot2, dplyr, varhandle, reshape2</p>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81b27969",
   "metadata": {},
   "source": [
    "## Load the ASVs and Feature table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c9c2d7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "install.packages(c(\"ggplot2\",\"dplyr\",\"varhandle\",\"reshape2\",\"svglite\")) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "20eb2d3d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ggplot2\n",
      "Loading required package: dplyr\n",
      "Warning message:\n",
      "\"package 'dplyr' was built under R version 3.6.3\"\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "Loading required package: varhandle\n",
      "Warning message:\n",
      "\"package 'varhandle' was built under R version 3.6.2\"Loading required package: reshape2\n",
      "Warning message:\n",
      "\"package 'reshape2' was built under R version 3.6.3\""
     ]
    }
   ],
   "source": [
    "require(ggplot2) #For Plots\n",
    "require(dplyr)  # For data manipulation\n",
    "require(varhandle) # Mainly for using unfactor function to convert factor to integer values\n",
    "require(reshape2) # Mainly for using 'melt' function\n",
    "require(svglite) #to save plots in svg format"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4b0e7a3",
   "metadata": {},
   "source": [
    "#### 'Global' settings for plot size in the output cell:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "id": "1578bda0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#options(repr.plot.width=10, repr.plot.height=8,res=600) #For google collab\n",
    "options(repr.plot.width=6, repr.plot.height=6) #For Jupyter Notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "94fd675d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enter the path of the folder with input files: D:\\Projects\\20220811_CCE_Correlation\n"
     ]
    }
   ],
   "source": [
    "# setting the current directory as the working directory\n",
    "Directory <- normalizePath(readline(\"Enter the path of the folder with input files: \"),\"/\",mustWork=FALSE)\n",
    "setwd(Directory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "01c78247",
   "metadata": {},
   "outputs": [],
   "source": [
    "asv16s <- read.csv(\"ASV_16S_With_Metadata.csv\",header=TRUE,check.names = FALSE,row.names = 1)\n",
    "asv18s <- read.csv(\"ASV_18SV9_FullDepth_With_Metadata.csv\",header=TRUE,check.names = FALSE,row.names = 1)\n",
    "FT <-  read.csv(\"CCE_Normalised_Data.csv\",header=TRUE,check.names = FALSE,row.names = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ebb3afc",
   "metadata": {},
   "source": [
    "Note that the asv files used are the ones with meatadata combined with them."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "a0787706",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>CCE_P1706_100.mzxml</th><th scope=col>CCE_P1706_102.mzxml</th><th scope=col>CCE_P1706_104.mzxml</th><th scope=col>CCE_P1706_106.mzxml</th><th scope=col>CCE_P1706_108.mzxml</th><th scope=col>CCE_P1706_110.mzxml</th><th scope=col>CCE_P1706_112.mzxml</th><th scope=col>CCE_P1706_114.mzxml</th><th scope=col>CCE_P1706_116.mzxml</th><th scope=col>CCE_P1706_118.mzxml</th><th scope=col>...</th><th scope=col>CCE_P1706_96.mzxml</th><th scope=col>CCE_P1706_98.mzxml</th><th scope=col>Class16S</th><th scope=col>Confidence</th><th scope=col>Domain16S</th><th scope=col>Family16S</th><th scope=col>Genus16S</th><th scope=col>Order16S</th><th scope=col>Phylum16S</th><th scope=col>Species16S</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>00208b62ef66aa36e069777b853c8f17</th><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>...                                              </td><td>   0                                             </td><td>   0                                             </td><td>Deltaproteobacteria                              </td><td>0.8284995                                        </td><td>Bacteria                                         </td><td>NA                                               </td><td>NA                                               </td><td>SAR324 clade(Marine group B)                     </td><td>Proteobacteria                                   </td><td>NA                                               </td></tr>\n",
       "\t<tr><th scope=row>00232f374cfc930f3fffbaef0d0c6d80</th><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>...                                              </td><td>   0                                             </td><td>   0                                             </td><td>Gammaproteobacteria                              </td><td>1.0000000                                        </td><td>Bacteria                                         </td><td>NA                                               </td><td>NA                                               </td><td>NA                                               </td><td>Proteobacteria                                   </td><td>NA                                               </td></tr>\n",
       "\t<tr><th scope=row>0023a8c071538909ed3386e4e1d52278</th><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>...                                              </td><td>   0                                             </td><td>   0                                             </td><td>Ignavibacteria                                   </td><td>0.7360358                                        </td><td>Bacteria                                         </td><td>uncultured organism                              </td><td>uncultured organism                              </td><td>OPB56                                            </td><td>Bacteroidetes                                    </td><td>uncultured organism                              </td></tr>\n",
       "\t<tr><th scope=row>00256e2eb0d211a7498c3e8b3170cedd</th><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   7                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>...                                              </td><td>   0                                             </td><td>   0                                             </td><td>NA                                               </td><td>0.9999943                                        </td><td>Bacteria                                         </td><td>NA                                               </td><td>NA                                               </td><td>NA                                               </td><td>Marinimicrobia (SAR406 clade)                    </td><td>NA                                               </td></tr>\n",
       "\t<tr><th scope=row>002ae12276a5da377dd6f7d5452a388a</th><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>  10                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>...                                              </td><td>   0                                             </td><td>   0                                             </td><td>Bacteroidia                                      </td><td>0.9964438                                        </td><td>Bacteria                                         </td><td>NS9 marine group                                 </td><td>NA                                               </td><td>Flavobacteriales                                 </td><td>Bacteroidetes                                    </td><td>NA                                               </td></tr>\n",
       "\t<tr><th scope=row>002c14075180e4484913e02e02dead50</th><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>   0                                             </td><td>...                                              </td><td>   0                                             </td><td>   0                                             </td><td>Bacteroidia                                      </td><td>0.9999990                                        </td><td>Bacteria                                         </td><td>Crocinitomicaceae                                </td><td>Crocinitomix                                     </td><td>Flavobacteriales                                 </td><td>Bacteroidetes                                    </td><td>uncultured Bacteroidetes/Chlorobi group bacterium</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & CCE\\_P1706\\_100.mzxml & CCE\\_P1706\\_102.mzxml & CCE\\_P1706\\_104.mzxml & CCE\\_P1706\\_106.mzxml & CCE\\_P1706\\_108.mzxml & CCE\\_P1706\\_110.mzxml & CCE\\_P1706\\_112.mzxml & CCE\\_P1706\\_114.mzxml & CCE\\_P1706\\_116.mzxml & CCE\\_P1706\\_118.mzxml & ... & CCE\\_P1706\\_96.mzxml & CCE\\_P1706\\_98.mzxml & Class16S & Confidence & Domain16S & Family16S & Genus16S & Order16S & Phylum16S & Species16S\\\\\n",
       "\\hline\n",
       "\t00208b62ef66aa36e069777b853c8f17 &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              & ...                                               &    0                                              &    0                                              & Deltaproteobacteria                               & 0.8284995                                         & Bacteria                                          & NA                                                & NA                                                & SAR324 clade(Marine group B)                      & Proteobacteria                                    & NA                                               \\\\\n",
       "\t00232f374cfc930f3fffbaef0d0c6d80 &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              & ...                                               &    0                                              &    0                                              & Gammaproteobacteria                               & 1.0000000                                         & Bacteria                                          & NA                                                & NA                                                & NA                                                & Proteobacteria                                    & NA                                               \\\\\n",
       "\t0023a8c071538909ed3386e4e1d52278 &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              & ...                                               &    0                                              &    0                                              & Ignavibacteria                                    & 0.7360358                                         & Bacteria                                          & uncultured organism                               & uncultured organism                               & OPB56                                             & Bacteroidetes                                     & uncultured organism                              \\\\\n",
       "\t00256e2eb0d211a7498c3e8b3170cedd &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    7                                              &    0                                              &    0                                              &    0                                              & ...                                               &    0                                              &    0                                              & NA                                                & 0.9999943                                         & Bacteria                                          & NA                                                & NA                                                & NA                                                & Marinimicrobia (SAR406 clade)                     & NA                                               \\\\\n",
       "\t002ae12276a5da377dd6f7d5452a388a &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &   10                                              &    0                                              &    0                                              &    0                                              & ...                                               &    0                                              &    0                                              & Bacteroidia                                       & 0.9964438                                         & Bacteria                                          & NS9 marine group                                  & NA                                                & Flavobacteriales                                  & Bacteroidetes                                     & NA                                               \\\\\n",
       "\t002c14075180e4484913e02e02dead50 &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              &    0                                              & ...                                               &    0                                              &    0                                              & Bacteroidia                                       & 0.9999990                                         & Bacteria                                          & Crocinitomicaceae                                 & Crocinitomix                                      & Flavobacteriales                                  & Bacteroidetes                                     & uncultured Bacteroidetes/Chlorobi group bacterium\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | CCE_P1706_100.mzxml | CCE_P1706_102.mzxml | CCE_P1706_104.mzxml | CCE_P1706_106.mzxml | CCE_P1706_108.mzxml | CCE_P1706_110.mzxml | CCE_P1706_112.mzxml | CCE_P1706_114.mzxml | CCE_P1706_116.mzxml | CCE_P1706_118.mzxml | ... | CCE_P1706_96.mzxml | CCE_P1706_98.mzxml | Class16S | Confidence | Domain16S | Family16S | Genus16S | Order16S | Phylum16S | Species16S |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 00208b62ef66aa36e069777b853c8f17 |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              | ...                                               |    0                                              |    0                                              | Deltaproteobacteria                               | 0.8284995                                         | Bacteria                                          | NA                                                | NA                                                | SAR324 clade(Marine group B)                      | Proteobacteria                                    | NA                                                |\n",
       "| 00232f374cfc930f3fffbaef0d0c6d80 |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              | ...                                               |    0                                              |    0                                              | Gammaproteobacteria                               | 1.0000000                                         | Bacteria                                          | NA                                                | NA                                                | NA                                                | Proteobacteria                                    | NA                                                |\n",
       "| 0023a8c071538909ed3386e4e1d52278 |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              | ...                                               |    0                                              |    0                                              | Ignavibacteria                                    | 0.7360358                                         | Bacteria                                          | uncultured organism                               | uncultured organism                               | OPB56                                             | Bacteroidetes                                     | uncultured organism                               |\n",
       "| 00256e2eb0d211a7498c3e8b3170cedd |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    7                                              |    0                                              |    0                                              |    0                                              | ...                                               |    0                                              |    0                                              | NA                                                | 0.9999943                                         | Bacteria                                          | NA                                                | NA                                                | NA                                                | Marinimicrobia (SAR406 clade)                     | NA                                                |\n",
       "| 002ae12276a5da377dd6f7d5452a388a |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |   10                                              |    0                                              |    0                                              |    0                                              | ...                                               |    0                                              |    0                                              | Bacteroidia                                       | 0.9964438                                         | Bacteria                                          | NS9 marine group                                  | NA                                                | Flavobacteriales                                  | Bacteroidetes                                     | NA                                                |\n",
       "| 002c14075180e4484913e02e02dead50 |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              |    0                                              | ...                                               |    0                                              |    0                                              | Bacteroidia                                       | 0.9999990                                         | Bacteria                                          | Crocinitomicaceae                                 | Crocinitomix                                      | Flavobacteriales                                  | Bacteroidetes                                     | uncultured Bacteroidetes/Chlorobi group bacterium |\n",
       "\n"
      ],
      "text/plain": [
       "                                 CCE_P1706_100.mzxml CCE_P1706_102.mzxml\n",
       "00208b62ef66aa36e069777b853c8f17    0                   0               \n",
       "00232f374cfc930f3fffbaef0d0c6d80    0                   0               \n",
       "0023a8c071538909ed3386e4e1d52278    0                   0               \n",
       "00256e2eb0d211a7498c3e8b3170cedd    0                   0               \n",
       "002ae12276a5da377dd6f7d5452a388a    0                   0               \n",
       "002c14075180e4484913e02e02dead50    0                   0               \n",
       "                                 CCE_P1706_104.mzxml CCE_P1706_106.mzxml\n",
       "00208b62ef66aa36e069777b853c8f17    0                   0               \n",
       "00232f374cfc930f3fffbaef0d0c6d80    0                   0               \n",
       "0023a8c071538909ed3386e4e1d52278    0                   0               \n",
       "00256e2eb0d211a7498c3e8b3170cedd    0                   0               \n",
       "002ae12276a5da377dd6f7d5452a388a    0                   0               \n",
       "002c14075180e4484913e02e02dead50    0                   0               \n",
       "                                 CCE_P1706_108.mzxml CCE_P1706_110.mzxml\n",
       "00208b62ef66aa36e069777b853c8f17    0                   0               \n",
       "00232f374cfc930f3fffbaef0d0c6d80    0                   0               \n",
       "0023a8c071538909ed3386e4e1d52278    0                   0               \n",
       "00256e2eb0d211a7498c3e8b3170cedd    0                   0               \n",
       "002ae12276a5da377dd6f7d5452a388a    0                   0               \n",
       "002c14075180e4484913e02e02dead50    0                   0               \n",
       "                                 CCE_P1706_112.mzxml CCE_P1706_114.mzxml\n",
       "00208b62ef66aa36e069777b853c8f17    0                   0               \n",
       "00232f374cfc930f3fffbaef0d0c6d80    0                   0               \n",
       "0023a8c071538909ed3386e4e1d52278    0                   0               \n",
       "00256e2eb0d211a7498c3e8b3170cedd    7                   0               \n",
       "002ae12276a5da377dd6f7d5452a388a   10                   0               \n",
       "002c14075180e4484913e02e02dead50    0                   0               \n",
       "                                 CCE_P1706_116.mzxml CCE_P1706_118.mzxml ...\n",
       "00208b62ef66aa36e069777b853c8f17    0                   0                ...\n",
       "00232f374cfc930f3fffbaef0d0c6d80    0                   0                ...\n",
       "0023a8c071538909ed3386e4e1d52278    0                   0                ...\n",
       "00256e2eb0d211a7498c3e8b3170cedd    0                   0                ...\n",
       "002ae12276a5da377dd6f7d5452a388a    0                   0                ...\n",
       "002c14075180e4484913e02e02dead50    0                   0                ...\n",
       "                                 CCE_P1706_96.mzxml CCE_P1706_98.mzxml\n",
       "00208b62ef66aa36e069777b853c8f17    0                  0              \n",
       "00232f374cfc930f3fffbaef0d0c6d80    0                  0              \n",
       "0023a8c071538909ed3386e4e1d52278    0                  0              \n",
       "00256e2eb0d211a7498c3e8b3170cedd    0                  0              \n",
       "002ae12276a5da377dd6f7d5452a388a    0                  0              \n",
       "002c14075180e4484913e02e02dead50    0                  0              \n",
       "                                 Class16S            Confidence Domain16S\n",
       "00208b62ef66aa36e069777b853c8f17 Deltaproteobacteria 0.8284995  Bacteria \n",
       "00232f374cfc930f3fffbaef0d0c6d80 Gammaproteobacteria 1.0000000  Bacteria \n",
       "0023a8c071538909ed3386e4e1d52278 Ignavibacteria      0.7360358  Bacteria \n",
       "00256e2eb0d211a7498c3e8b3170cedd NA                  0.9999943  Bacteria \n",
       "002ae12276a5da377dd6f7d5452a388a Bacteroidia         0.9964438  Bacteria \n",
       "002c14075180e4484913e02e02dead50 Bacteroidia         0.9999990  Bacteria \n",
       "                                 Family16S           Genus16S           \n",
       "00208b62ef66aa36e069777b853c8f17 NA                  NA                 \n",
       "00232f374cfc930f3fffbaef0d0c6d80 NA                  NA                 \n",
       "0023a8c071538909ed3386e4e1d52278 uncultured organism uncultured organism\n",
       "00256e2eb0d211a7498c3e8b3170cedd NA                  NA                 \n",
       "002ae12276a5da377dd6f7d5452a388a NS9 marine group    NA                 \n",
       "002c14075180e4484913e02e02dead50 Crocinitomicaceae   Crocinitomix       \n",
       "                                 Order16S                    \n",
       "00208b62ef66aa36e069777b853c8f17 SAR324 clade(Marine group B)\n",
       "00232f374cfc930f3fffbaef0d0c6d80 NA                          \n",
       "0023a8c071538909ed3386e4e1d52278 OPB56                       \n",
       "00256e2eb0d211a7498c3e8b3170cedd NA                          \n",
       "002ae12276a5da377dd6f7d5452a388a Flavobacteriales            \n",
       "002c14075180e4484913e02e02dead50 Flavobacteriales            \n",
       "                                 Phylum16S                    \n",
       "00208b62ef66aa36e069777b853c8f17 Proteobacteria               \n",
       "00232f374cfc930f3fffbaef0d0c6d80 Proteobacteria               \n",
       "0023a8c071538909ed3386e4e1d52278 Bacteroidetes                \n",
       "00256e2eb0d211a7498c3e8b3170cedd Marinimicrobia (SAR406 clade)\n",
       "002ae12276a5da377dd6f7d5452a388a Bacteroidetes                \n",
       "002c14075180e4484913e02e02dead50 Bacteroidetes                \n",
       "                                 Species16S                                       \n",
       "00208b62ef66aa36e069777b853c8f17 NA                                               \n",
       "00232f374cfc930f3fffbaef0d0c6d80 NA                                               \n",
       "0023a8c071538909ed3386e4e1d52278 uncultured organism                              \n",
       "00256e2eb0d211a7498c3e8b3170cedd NA                                               \n",
       "002ae12276a5da377dd6f7d5452a388a NA                                               \n",
       "002c14075180e4484913e02e02dead50 uncultured Bacteroidetes/Chlorobi group bacterium"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>7538</li>\n",
       "\t<li>79</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 7538\n",
       "\\item 79\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 7538\n",
       "2. 79\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 7538   79"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(asv16s)\n",
    "dim(asv16s)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8fdabd81",
   "metadata": {},
   "source": [
    "<p style='text-align: justify;'>The observed dimension above was 7538 rows and 79 columns. Out of this 79, <b> 8 columns were metadata info </b> (Class16S, Confidence, Domain16S, Family16S, Genus16S, Order16S, Phylum16S, Species16S). Thus, 71 files in asv16s dataframe.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "62aba349",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>CCE_P1706_100.mzxml</th><th scope=col>CCE_P1706_102.mzxml</th><th scope=col>CCE_P1706_104.mzxml</th><th scope=col>CCE_P1706_106.mzxml</th><th scope=col>CCE_P1706_108.mzxml</th><th scope=col>CCE_P1706_110.mzxml</th><th scope=col>CCE_P1706_112.mzxml</th><th scope=col>CCE_P1706_114.mzxml</th><th scope=col>CCE_P1706_116.mzxml</th><th scope=col>CCE_P1706_118.mzxml</th><th scope=col>...</th><th scope=col>Class18SV9</th><th scope=col>Confidence</th><th scope=col>Domain18SV9</th><th scope=col>Family18SV9</th><th scope=col>Genus18SV9</th><th scope=col>Kingdom18SV9</th><th scope=col>Order18SV9</th><th scope=col>Phylum18SV9</th><th scope=col>Species18SV9</th><th scope=col>X</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>000167ce642266c21c62843e4a010266</th><td>   0              </td><td>   0              </td><td>   0              </td><td>    0             </td><td>    0             </td><td>    0             </td><td>   0              </td><td>   0              </td><td>   0              </td><td>   0              </td><td>...               </td><td>Acantharea        </td><td>0.9991760         </td><td>Eukaryota         </td><td>Acantharea_XX     </td><td>Acantharea_XXX    </td><td>Rhizaria          </td><td>Acantharea_X      </td><td>Radiolaria        </td><td>Acantharea_XXX_sp.</td><td>6114              </td></tr>\n",
       "\t<tr><th scope=row>001567b7cb76eecd73f31a288fcfcb69</th><td>   0              </td><td>   0              </td><td>   0              </td><td>    0             </td><td>    0             </td><td>    0             </td><td>   0              </td><td>   0              </td><td>   0              </td><td>   0              </td><td>...               </td><td>NA                </td><td>0.7359983         </td><td>Eukaryota         </td><td>NA                </td><td>NA                </td><td>Stramenopiles     </td><td>NA                </td><td>NA                </td><td>NA                </td><td>9879              </td></tr>\n",
       "\t<tr><th scope=row>0018f741aa09de859e5af52f19a2b7c1</th><td>   0              </td><td>   0              </td><td>   0              </td><td>    0             </td><td>    0             </td><td>    0             </td><td>   0              </td><td>   0              </td><td>   0              </td><td>   0              </td><td>...               </td><td>NA                </td><td>1.0000000         </td><td>Eukaryota         </td><td>NA                </td><td>NA                </td><td>NA                </td><td>NA                </td><td>NA                </td><td>NA                </td><td>5051              </td></tr>\n",
       "\t<tr><th scope=row>00236daa14e0b5a92c512b565be61b51</th><td>   0              </td><td>   0              </td><td>   0              </td><td>    0             </td><td>    0             </td><td>    0             </td><td>   0              </td><td>   0              </td><td>   0              </td><td>   0              </td><td>...               </td><td>Chlorophyceae     </td><td>0.8520327         </td><td>Eukaryota         </td><td>NA                </td><td>NA                </td><td>Archaeplastida    </td><td>NA                </td><td>Chlorophyta       </td><td>NA                </td><td>9634              </td></tr>\n",
       "\t<tr><th scope=row>003695728276f773ed0c54ca361198e2</th><td>   0              </td><td>   0              </td><td>   0              </td><td>    0             </td><td>    0             </td><td>    0             </td><td>   0              </td><td>   0              </td><td>   0              </td><td>   0              </td><td>...               </td><td>NA                </td><td>1.0000000         </td><td>Eukaryota         </td><td>NA                </td><td>NA                </td><td>NA                </td><td>NA                </td><td>NA                </td><td>NA                </td><td>5130              </td></tr>\n",
       "\t<tr><th scope=row>0038284216ef9b1f951a5f4e5d4a3aea</th><td>   0              </td><td>   0              </td><td>   0              </td><td>    0             </td><td>    0             </td><td>    0             </td><td>   0              </td><td>   0              </td><td>   0              </td><td>   0              </td><td>...               </td><td>Pyramimonadales   </td><td>0.9981855         </td><td>Eukaryota         </td><td>Pyramimonadales_XX</td><td>Halosphaera       </td><td>Archaeplastida    </td><td>Pyramimonadales_X </td><td>Chlorophyta       </td><td>Halosphaera_sp.   </td><td>2268              </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & CCE\\_P1706\\_100.mzxml & CCE\\_P1706\\_102.mzxml & CCE\\_P1706\\_104.mzxml & CCE\\_P1706\\_106.mzxml & CCE\\_P1706\\_108.mzxml & CCE\\_P1706\\_110.mzxml & CCE\\_P1706\\_112.mzxml & CCE\\_P1706\\_114.mzxml & CCE\\_P1706\\_116.mzxml & CCE\\_P1706\\_118.mzxml & ... & Class18SV9 & Confidence & Domain18SV9 & Family18SV9 & Genus18SV9 & Kingdom18SV9 & Order18SV9 & Phylum18SV9 & Species18SV9 & X\\\\\n",
       "\\hline\n",
       "\t000167ce642266c21c62843e4a010266 &    0                   &    0                   &    0                   &     0                  &     0                  &     0                  &    0                   &    0                   &    0                   &    0                   & ...                    & Acantharea             & 0.9991760              & Eukaryota              & Acantharea\\_XX        & Acantharea\\_XXX       & Rhizaria               & Acantharea\\_X         & Radiolaria             & Acantharea\\_XXX\\_sp. & 6114                  \\\\\n",
       "\t001567b7cb76eecd73f31a288fcfcb69 &    0               &    0               &    0               &     0              &     0              &     0              &    0               &    0               &    0               &    0               & ...                & NA                 & 0.7359983          & Eukaryota          & NA                 & NA                 & Stramenopiles      & NA                 & NA                 & NA                 & 9879              \\\\\n",
       "\t0018f741aa09de859e5af52f19a2b7c1 &    0               &    0               &    0               &     0              &     0              &     0              &    0               &    0               &    0               &    0               & ...                & NA                 & 1.0000000          & Eukaryota          & NA                 & NA                 & NA                 & NA                 & NA                 & NA                 & 5051              \\\\\n",
       "\t00236daa14e0b5a92c512b565be61b51 &    0               &    0               &    0               &     0              &     0              &     0              &    0               &    0               &    0               &    0               & ...                & Chlorophyceae      & 0.8520327          & Eukaryota          & NA                 & NA                 & Archaeplastida     & NA                 & Chlorophyta        & NA                 & 9634              \\\\\n",
       "\t003695728276f773ed0c54ca361198e2 &    0               &    0               &    0               &     0              &     0              &     0              &    0               &    0               &    0               &    0               & ...                & NA                 & 1.0000000          & Eukaryota          & NA                 & NA                 & NA                 & NA                 & NA                 & NA                 & 5130              \\\\\n",
       "\t0038284216ef9b1f951a5f4e5d4a3aea &    0                 &    0                 &    0                 &     0                &     0                &     0                &    0                 &    0                 &    0                 &    0                 & ...                  & Pyramimonadales      & 0.9981855            & Eukaryota            & Pyramimonadales\\_XX & Halosphaera          & Archaeplastida       & Pyramimonadales\\_X  & Chlorophyta          & Halosphaera\\_sp.    & 2268                \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | CCE_P1706_100.mzxml | CCE_P1706_102.mzxml | CCE_P1706_104.mzxml | CCE_P1706_106.mzxml | CCE_P1706_108.mzxml | CCE_P1706_110.mzxml | CCE_P1706_112.mzxml | CCE_P1706_114.mzxml | CCE_P1706_116.mzxml | CCE_P1706_118.mzxml | ... | Class18SV9 | Confidence | Domain18SV9 | Family18SV9 | Genus18SV9 | Kingdom18SV9 | Order18SV9 | Phylum18SV9 | Species18SV9 | X |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 000167ce642266c21c62843e4a010266 |    0               |    0               |    0               |     0              |     0              |     0              |    0               |    0               |    0               |    0               | ...                | Acantharea         | 0.9991760          | Eukaryota          | Acantharea_XX      | Acantharea_XXX     | Rhizaria           | Acantharea_X       | Radiolaria         | Acantharea_XXX_sp. | 6114               |\n",
       "| 001567b7cb76eecd73f31a288fcfcb69 |    0               |    0               |    0               |     0              |     0              |     0              |    0               |    0               |    0               |    0               | ...                | NA                 | 0.7359983          | Eukaryota          | NA                 | NA                 | Stramenopiles      | NA                 | NA                 | NA                 | 9879               |\n",
       "| 0018f741aa09de859e5af52f19a2b7c1 |    0               |    0               |    0               |     0              |     0              |     0              |    0               |    0               |    0               |    0               | ...                | NA                 | 1.0000000          | Eukaryota          | NA                 | NA                 | NA                 | NA                 | NA                 | NA                 | 5051               |\n",
       "| 00236daa14e0b5a92c512b565be61b51 |    0               |    0               |    0               |     0              |     0              |     0              |    0               |    0               |    0               |    0               | ...                | Chlorophyceae      | 0.8520327          | Eukaryota          | NA                 | NA                 | Archaeplastida     | NA                 | Chlorophyta        | NA                 | 9634               |\n",
       "| 003695728276f773ed0c54ca361198e2 |    0               |    0               |    0               |     0              |     0              |     0              |    0               |    0               |    0               |    0               | ...                | NA                 | 1.0000000          | Eukaryota          | NA                 | NA                 | NA                 | NA                 | NA                 | NA                 | 5130               |\n",
       "| 0038284216ef9b1f951a5f4e5d4a3aea |    0               |    0               |    0               |     0              |     0              |     0              |    0               |    0               |    0               |    0               | ...                | Pyramimonadales    | 0.9981855          | Eukaryota          | Pyramimonadales_XX | Halosphaera        | Archaeplastida     | Pyramimonadales_X  | Chlorophyta        | Halosphaera_sp.    | 2268               |\n",
       "\n"
      ],
      "text/plain": [
       "                                 CCE_P1706_100.mzxml CCE_P1706_102.mzxml\n",
       "000167ce642266c21c62843e4a010266    0                   0               \n",
       "001567b7cb76eecd73f31a288fcfcb69    0                   0               \n",
       "0018f741aa09de859e5af52f19a2b7c1    0                   0               \n",
       "00236daa14e0b5a92c512b565be61b51    0                   0               \n",
       "003695728276f773ed0c54ca361198e2    0                   0               \n",
       "0038284216ef9b1f951a5f4e5d4a3aea    0                   0               \n",
       "                                 CCE_P1706_104.mzxml CCE_P1706_106.mzxml\n",
       "000167ce642266c21c62843e4a010266    0                    0              \n",
       "001567b7cb76eecd73f31a288fcfcb69    0                    0              \n",
       "0018f741aa09de859e5af52f19a2b7c1    0                    0              \n",
       "00236daa14e0b5a92c512b565be61b51    0                    0              \n",
       "003695728276f773ed0c54ca361198e2    0                    0              \n",
       "0038284216ef9b1f951a5f4e5d4a3aea    0                    0              \n",
       "                                 CCE_P1706_108.mzxml CCE_P1706_110.mzxml\n",
       "000167ce642266c21c62843e4a010266     0                   0              \n",
       "001567b7cb76eecd73f31a288fcfcb69     0                   0              \n",
       "0018f741aa09de859e5af52f19a2b7c1     0                   0              \n",
       "00236daa14e0b5a92c512b565be61b51     0                   0              \n",
       "003695728276f773ed0c54ca361198e2     0                   0              \n",
       "0038284216ef9b1f951a5f4e5d4a3aea     0                   0              \n",
       "                                 CCE_P1706_112.mzxml CCE_P1706_114.mzxml\n",
       "000167ce642266c21c62843e4a010266    0                   0               \n",
       "001567b7cb76eecd73f31a288fcfcb69    0                   0               \n",
       "0018f741aa09de859e5af52f19a2b7c1    0                   0               \n",
       "00236daa14e0b5a92c512b565be61b51    0                   0               \n",
       "003695728276f773ed0c54ca361198e2    0                   0               \n",
       "0038284216ef9b1f951a5f4e5d4a3aea    0                   0               \n",
       "                                 CCE_P1706_116.mzxml CCE_P1706_118.mzxml ...\n",
       "000167ce642266c21c62843e4a010266    0                   0                ...\n",
       "001567b7cb76eecd73f31a288fcfcb69    0                   0                ...\n",
       "0018f741aa09de859e5af52f19a2b7c1    0                   0                ...\n",
       "00236daa14e0b5a92c512b565be61b51    0                   0                ...\n",
       "003695728276f773ed0c54ca361198e2    0                   0                ...\n",
       "0038284216ef9b1f951a5f4e5d4a3aea    0                   0                ...\n",
       "                                 Class18SV9      Confidence Domain18SV9\n",
       "000167ce642266c21c62843e4a010266 Acantharea      0.9991760  Eukaryota  \n",
       "001567b7cb76eecd73f31a288fcfcb69 NA              0.7359983  Eukaryota  \n",
       "0018f741aa09de859e5af52f19a2b7c1 NA              1.0000000  Eukaryota  \n",
       "00236daa14e0b5a92c512b565be61b51 Chlorophyceae   0.8520327  Eukaryota  \n",
       "003695728276f773ed0c54ca361198e2 NA              1.0000000  Eukaryota  \n",
       "0038284216ef9b1f951a5f4e5d4a3aea Pyramimonadales 0.9981855  Eukaryota  \n",
       "                                 Family18SV9        Genus18SV9    \n",
       "000167ce642266c21c62843e4a010266 Acantharea_XX      Acantharea_XXX\n",
       "001567b7cb76eecd73f31a288fcfcb69 NA                 NA            \n",
       "0018f741aa09de859e5af52f19a2b7c1 NA                 NA            \n",
       "00236daa14e0b5a92c512b565be61b51 NA                 NA            \n",
       "003695728276f773ed0c54ca361198e2 NA                 NA            \n",
       "0038284216ef9b1f951a5f4e5d4a3aea Pyramimonadales_XX Halosphaera   \n",
       "                                 Kingdom18SV9   Order18SV9        Phylum18SV9\n",
       "000167ce642266c21c62843e4a010266 Rhizaria       Acantharea_X      Radiolaria \n",
       "001567b7cb76eecd73f31a288fcfcb69 Stramenopiles  NA                NA         \n",
       "0018f741aa09de859e5af52f19a2b7c1 NA             NA                NA         \n",
       "00236daa14e0b5a92c512b565be61b51 Archaeplastida NA                Chlorophyta\n",
       "003695728276f773ed0c54ca361198e2 NA             NA                NA         \n",
       "0038284216ef9b1f951a5f4e5d4a3aea Archaeplastida Pyramimonadales_X Chlorophyta\n",
       "                                 Species18SV9       X   \n",
       "000167ce642266c21c62843e4a010266 Acantharea_XXX_sp. 6114\n",
       "001567b7cb76eecd73f31a288fcfcb69 NA                 9879\n",
       "0018f741aa09de859e5af52f19a2b7c1 NA                 5051\n",
       "00236daa14e0b5a92c512b565be61b51 NA                 9634\n",
       "003695728276f773ed0c54ca361198e2 NA                 5130\n",
       "0038284216ef9b1f951a5f4e5d4a3aea Halosphaera_sp.    2268"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>10286</li>\n",
       "\t<li>81</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10286\n",
       "\\item 81\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10286\n",
       "2. 81\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10286    81"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(asv18s)\n",
    "dim(asv18s)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c01c17be",
   "metadata": {},
   "source": [
    "<p style='text-align: center;'> Dimension for ASV 18SV9 = 10286 rows and 81 columns </p>\n",
    "<p style='text-align: justify;'>More features observed with ASV18SV9 (10286 rows) and <b> there are 10 metadata columns:</b> Class18SV9, Confidence, Domain18SV9, Family18SV9, Genus18SV9, Kingdom18SV9, Order18SV9, Phylum18SV9, Species18SV9, X </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "2cca0eda",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Removing the last 4 metadata rows from the ASV files and making all CCE column data as numerci data \n",
    "asv16s_temp <- asv16s[-c((nrow(asv16s)-3):nrow(asv16s)),] \n",
    "asv16s_temp2 <- unfactor(asv16s_temp[,1:71])\n",
    "asv16s_f <- cbind(asv16s_temp2, asv16s_temp[,72:ncol(asv16s_temp)])\n",
    "\n",
    "asv18s_temp <- asv18s[-c((nrow(asv18s)-3):nrow(asv18s)),]\n",
    "asv18s_temp2 <- unfactor(asv18s_temp[,1:71])\n",
    "asv18s_f <- cbind(asv18s_temp2, asv18s_temp[,72:ncol(asv18s_temp)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "01d7cfc1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>7534</li>\n",
       "\t<li>79</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 7534\n",
       "\\item 79\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 7534\n",
       "2. 79\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 7534   79"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>10282</li>\n",
       "\t<li>81</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10282\n",
       "\\item 81\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10282\n",
       "2. 81\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10282    81"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Checking the dimensions of new asv tables\n",
    "dim(asv16s_f)\n",
    "dim(asv18s_f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "db843f17",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>CCE_P1706_3.mzxml</th><th scope=col>CCE_P1706_4.mzxml</th><th scope=col>CCE_P1706_5.mzxml</th><th scope=col>CCE_P1706_6.mzxml</th><th scope=col>CCE_P1706_7.mzxml</th><th scope=col>CCE_P1706_8.mzxml</th><th scope=col>CCE_P1706_9.mzxml</th><th scope=col>CCE_P1706_10.mzxml</th><th scope=col>CCE_P1706_11.mzxml</th><th scope=col>CCE_P1706_12.mzxml</th><th scope=col>...</th><th scope=col>CCE_P1706_221.mzxml</th><th scope=col>CCE_P1706_222.mzxml</th><th scope=col>CCE_P1706_223.mzxml</th><th scope=col>CCE_P1706_224.mzxml</th><th scope=col>CCE_P1706_225.mzxml</th><th scope=col>CCE_P1706_226.mzxml</th><th scope=col>CCE_P1706_227.mzxml</th><th scope=col>CCE_P1706_228.mzxml</th><th scope=col>CCE_P1706_229.mzxml</th><th scope=col>CCE_P1706_230.mzxml</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>2492_150.13_0</th><td>0.000002930</td><td>0.000003690</td><td>0.000002740</td><td>0.000002070</td><td>0.000002440</td><td>0.000002820</td><td>0.000003100</td><td>0.000003410</td><td>0.000002880</td><td>0.000002520</td><td>...        </td><td>0.000003170</td><td>0.000002800</td><td>0.00001180 </td><td>0.000002830</td><td>0.000236714</td><td>0.000290522</td><td>0.000211238</td><td>0.000274079</td><td>0.000278415</td><td>0.000238976</td></tr>\n",
       "\t<tr><th scope=row>129_151.03_0.001</th><td>0.001334518</td><td>0.002481587</td><td>0.000511703</td><td>0.001005175</td><td>0.000644174</td><td>0.000825979</td><td>0.001663834</td><td>0.002546505</td><td>0.001957699</td><td>0.000847911</td><td>...        </td><td>0.001099565</td><td>0.000564127</td><td>0.00100717 </td><td>0.001223748</td><td>0.001730357</td><td>0.000188482</td><td>0.001106029</td><td>0.001532877</td><td>0.000374890</td><td>0.000320765</td></tr>\n",
       "\t<tr><th scope=row>1112_151.03_0</th><td>0.000002930</td><td>0.000003690</td><td>0.000002740</td><td>0.000002070</td><td>0.000002440</td><td>0.000002820</td><td>0.000003100</td><td>0.000002550</td><td>0.000002880</td><td>0.000002520</td><td>...        </td><td>0.000003170</td><td>0.000002800</td><td>0.00000247 </td><td>0.000002830</td><td>0.000001730</td><td>0.000002040</td><td>0.000001860</td><td>0.000001910</td><td>0.000001850</td><td>0.000001930</td></tr>\n",
       "\t<tr><th scope=row>3075_151.07_0</th><td>0.000002930</td><td>0.000003690</td><td>0.000002740</td><td>0.000002070</td><td>0.000002440</td><td>0.000002820</td><td>0.000003100</td><td>0.000002550</td><td>0.000002880</td><td>0.000002520</td><td>...        </td><td>0.000003170</td><td>0.000002800</td><td>0.00000247 </td><td>0.000002830</td><td>0.000001730</td><td>0.000002040</td><td>0.000001860</td><td>0.000001910</td><td>0.000001850</td><td>0.000001930</td></tr>\n",
       "\t<tr><th scope=row>542_152.06_0</th><td>0.000012900</td><td>0.000007100</td><td>0.000022900</td><td>0.000002070</td><td>0.000002440</td><td>0.000002820</td><td>0.000003100</td><td>0.000002550</td><td>0.000002880</td><td>0.000002520</td><td>...        </td><td>0.000027100</td><td>0.000010500</td><td>0.00003350 </td><td>0.000025900</td><td>0.000014400</td><td>0.000010400</td><td>0.000019900</td><td>0.000008700</td><td>0.000009800</td><td>0.000007410</td></tr>\n",
       "\t<tr><th scope=row>31563_153_0</th><td>0.000002930</td><td>0.000003690</td><td>0.000002740</td><td>0.000002070</td><td>0.000002440</td><td>0.000002820</td><td>0.000003100</td><td>0.000002550</td><td>0.000002880</td><td>0.000002520</td><td>...        </td><td>0.000003170</td><td>0.000002800</td><td>0.00000247 </td><td>0.000002830</td><td>0.000001730</td><td>0.000002040</td><td>0.000001860</td><td>0.000001910</td><td>0.000001850</td><td>0.000001930</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & CCE\\_P1706\\_3.mzxml & CCE\\_P1706\\_4.mzxml & CCE\\_P1706\\_5.mzxml & CCE\\_P1706\\_6.mzxml & CCE\\_P1706\\_7.mzxml & CCE\\_P1706\\_8.mzxml & CCE\\_P1706\\_9.mzxml & CCE\\_P1706\\_10.mzxml & CCE\\_P1706\\_11.mzxml & CCE\\_P1706\\_12.mzxml & ... & CCE\\_P1706\\_221.mzxml & CCE\\_P1706\\_222.mzxml & CCE\\_P1706\\_223.mzxml & CCE\\_P1706\\_224.mzxml & CCE\\_P1706\\_225.mzxml & CCE\\_P1706\\_226.mzxml & CCE\\_P1706\\_227.mzxml & CCE\\_P1706\\_228.mzxml & CCE\\_P1706\\_229.mzxml & CCE\\_P1706\\_230.mzxml\\\\\n",
       "\\hline\n",
       "\t2492\\_150.13\\_0 & 0.000002930 & 0.000003690 & 0.000002740 & 0.000002070 & 0.000002440 & 0.000002820 & 0.000003100 & 0.000003410 & 0.000002880 & 0.000002520 & ...         & 0.000003170 & 0.000002800 & 0.00001180  & 0.000002830 & 0.000236714 & 0.000290522 & 0.000211238 & 0.000274079 & 0.000278415 & 0.000238976\\\\\n",
       "\t129\\_151.03\\_0.001 & 0.001334518 & 0.002481587 & 0.000511703 & 0.001005175 & 0.000644174 & 0.000825979 & 0.001663834 & 0.002546505 & 0.001957699 & 0.000847911 & ...         & 0.001099565 & 0.000564127 & 0.00100717  & 0.001223748 & 0.001730357 & 0.000188482 & 0.001106029 & 0.001532877 & 0.000374890 & 0.000320765\\\\\n",
       "\t1112\\_151.03\\_0 & 0.000002930 & 0.000003690 & 0.000002740 & 0.000002070 & 0.000002440 & 0.000002820 & 0.000003100 & 0.000002550 & 0.000002880 & 0.000002520 & ...         & 0.000003170 & 0.000002800 & 0.00000247  & 0.000002830 & 0.000001730 & 0.000002040 & 0.000001860 & 0.000001910 & 0.000001850 & 0.000001930\\\\\n",
       "\t3075\\_151.07\\_0 & 0.000002930 & 0.000003690 & 0.000002740 & 0.000002070 & 0.000002440 & 0.000002820 & 0.000003100 & 0.000002550 & 0.000002880 & 0.000002520 & ...         & 0.000003170 & 0.000002800 & 0.00000247  & 0.000002830 & 0.000001730 & 0.000002040 & 0.000001860 & 0.000001910 & 0.000001850 & 0.000001930\\\\\n",
       "\t542\\_152.06\\_0 & 0.000012900 & 0.000007100 & 0.000022900 & 0.000002070 & 0.000002440 & 0.000002820 & 0.000003100 & 0.000002550 & 0.000002880 & 0.000002520 & ...         & 0.000027100 & 0.000010500 & 0.00003350  & 0.000025900 & 0.000014400 & 0.000010400 & 0.000019900 & 0.000008700 & 0.000009800 & 0.000007410\\\\\n",
       "\t31563\\_153\\_0 & 0.000002930 & 0.000003690 & 0.000002740 & 0.000002070 & 0.000002440 & 0.000002820 & 0.000003100 & 0.000002550 & 0.000002880 & 0.000002520 & ...         & 0.000003170 & 0.000002800 & 0.00000247  & 0.000002830 & 0.000001730 & 0.000002040 & 0.000001860 & 0.000001910 & 0.000001850 & 0.000001930\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | CCE_P1706_3.mzxml | CCE_P1706_4.mzxml | CCE_P1706_5.mzxml | CCE_P1706_6.mzxml | CCE_P1706_7.mzxml | CCE_P1706_8.mzxml | CCE_P1706_9.mzxml | CCE_P1706_10.mzxml | CCE_P1706_11.mzxml | CCE_P1706_12.mzxml | ... | CCE_P1706_221.mzxml | CCE_P1706_222.mzxml | CCE_P1706_223.mzxml | CCE_P1706_224.mzxml | CCE_P1706_225.mzxml | CCE_P1706_226.mzxml | CCE_P1706_227.mzxml | CCE_P1706_228.mzxml | CCE_P1706_229.mzxml | CCE_P1706_230.mzxml |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 2492_150.13_0 | 0.000002930 | 0.000003690 | 0.000002740 | 0.000002070 | 0.000002440 | 0.000002820 | 0.000003100 | 0.000003410 | 0.000002880 | 0.000002520 | ...         | 0.000003170 | 0.000002800 | 0.00001180  | 0.000002830 | 0.000236714 | 0.000290522 | 0.000211238 | 0.000274079 | 0.000278415 | 0.000238976 |\n",
       "| 129_151.03_0.001 | 0.001334518 | 0.002481587 | 0.000511703 | 0.001005175 | 0.000644174 | 0.000825979 | 0.001663834 | 0.002546505 | 0.001957699 | 0.000847911 | ...         | 0.001099565 | 0.000564127 | 0.00100717  | 0.001223748 | 0.001730357 | 0.000188482 | 0.001106029 | 0.001532877 | 0.000374890 | 0.000320765 |\n",
       "| 1112_151.03_0 | 0.000002930 | 0.000003690 | 0.000002740 | 0.000002070 | 0.000002440 | 0.000002820 | 0.000003100 | 0.000002550 | 0.000002880 | 0.000002520 | ...         | 0.000003170 | 0.000002800 | 0.00000247  | 0.000002830 | 0.000001730 | 0.000002040 | 0.000001860 | 0.000001910 | 0.000001850 | 0.000001930 |\n",
       "| 3075_151.07_0 | 0.000002930 | 0.000003690 | 0.000002740 | 0.000002070 | 0.000002440 | 0.000002820 | 0.000003100 | 0.000002550 | 0.000002880 | 0.000002520 | ...         | 0.000003170 | 0.000002800 | 0.00000247  | 0.000002830 | 0.000001730 | 0.000002040 | 0.000001860 | 0.000001910 | 0.000001850 | 0.000001930 |\n",
       "| 542_152.06_0 | 0.000012900 | 0.000007100 | 0.000022900 | 0.000002070 | 0.000002440 | 0.000002820 | 0.000003100 | 0.000002550 | 0.000002880 | 0.000002520 | ...         | 0.000027100 | 0.000010500 | 0.00003350  | 0.000025900 | 0.000014400 | 0.000010400 | 0.000019900 | 0.000008700 | 0.000009800 | 0.000007410 |\n",
       "| 31563_153_0 | 0.000002930 | 0.000003690 | 0.000002740 | 0.000002070 | 0.000002440 | 0.000002820 | 0.000003100 | 0.000002550 | 0.000002880 | 0.000002520 | ...         | 0.000003170 | 0.000002800 | 0.00000247  | 0.000002830 | 0.000001730 | 0.000002040 | 0.000001860 | 0.000001910 | 0.000001850 | 0.000001930 |\n",
       "\n"
      ],
      "text/plain": [
       "                 CCE_P1706_3.mzxml CCE_P1706_4.mzxml CCE_P1706_5.mzxml\n",
       "2492_150.13_0    0.000002930       0.000003690       0.000002740      \n",
       "129_151.03_0.001 0.001334518       0.002481587       0.000511703      \n",
       "1112_151.03_0    0.000002930       0.000003690       0.000002740      \n",
       "3075_151.07_0    0.000002930       0.000003690       0.000002740      \n",
       "542_152.06_0     0.000012900       0.000007100       0.000022900      \n",
       "31563_153_0      0.000002930       0.000003690       0.000002740      \n",
       "                 CCE_P1706_6.mzxml CCE_P1706_7.mzxml CCE_P1706_8.mzxml\n",
       "2492_150.13_0    0.000002070       0.000002440       0.000002820      \n",
       "129_151.03_0.001 0.001005175       0.000644174       0.000825979      \n",
       "1112_151.03_0    0.000002070       0.000002440       0.000002820      \n",
       "3075_151.07_0    0.000002070       0.000002440       0.000002820      \n",
       "542_152.06_0     0.000002070       0.000002440       0.000002820      \n",
       "31563_153_0      0.000002070       0.000002440       0.000002820      \n",
       "                 CCE_P1706_9.mzxml CCE_P1706_10.mzxml CCE_P1706_11.mzxml\n",
       "2492_150.13_0    0.000003100       0.000003410        0.000002880       \n",
       "129_151.03_0.001 0.001663834       0.002546505        0.001957699       \n",
       "1112_151.03_0    0.000003100       0.000002550        0.000002880       \n",
       "3075_151.07_0    0.000003100       0.000002550        0.000002880       \n",
       "542_152.06_0     0.000003100       0.000002550        0.000002880       \n",
       "31563_153_0      0.000003100       0.000002550        0.000002880       \n",
       "                 CCE_P1706_12.mzxml ... CCE_P1706_221.mzxml CCE_P1706_222.mzxml\n",
       "2492_150.13_0    0.000002520        ... 0.000003170         0.000002800        \n",
       "129_151.03_0.001 0.000847911        ... 0.001099565         0.000564127        \n",
       "1112_151.03_0    0.000002520        ... 0.000003170         0.000002800        \n",
       "3075_151.07_0    0.000002520        ... 0.000003170         0.000002800        \n",
       "542_152.06_0     0.000002520        ... 0.000027100         0.000010500        \n",
       "31563_153_0      0.000002520        ... 0.000003170         0.000002800        \n",
       "                 CCE_P1706_223.mzxml CCE_P1706_224.mzxml CCE_P1706_225.mzxml\n",
       "2492_150.13_0    0.00001180          0.000002830         0.000236714        \n",
       "129_151.03_0.001 0.00100717          0.001223748         0.001730357        \n",
       "1112_151.03_0    0.00000247          0.000002830         0.000001730        \n",
       "3075_151.07_0    0.00000247          0.000002830         0.000001730        \n",
       "542_152.06_0     0.00003350          0.000025900         0.000014400        \n",
       "31563_153_0      0.00000247          0.000002830         0.000001730        \n",
       "                 CCE_P1706_226.mzxml CCE_P1706_227.mzxml CCE_P1706_228.mzxml\n",
       "2492_150.13_0    0.000290522         0.000211238         0.000274079        \n",
       "129_151.03_0.001 0.000188482         0.001106029         0.001532877        \n",
       "1112_151.03_0    0.000002040         0.000001860         0.000001910        \n",
       "3075_151.07_0    0.000002040         0.000001860         0.000001910        \n",
       "542_152.06_0     0.000010400         0.000019900         0.000008700        \n",
       "31563_153_0      0.000002040         0.000001860         0.000001910        \n",
       "                 CCE_P1706_229.mzxml CCE_P1706_230.mzxml\n",
       "2492_150.13_0    0.000278415         0.000238976        \n",
       "129_151.03_0.001 0.000374890         0.000320765        \n",
       "1112_151.03_0    0.000001850         0.000001930        \n",
       "3075_151.07_0    0.000001850         0.000001930        \n",
       "542_152.06_0     0.000009800         0.000007410        \n",
       "31563_153_0      0.000001850         0.000001930        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>10644</li>\n",
       "\t<li>228</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10644\n",
       "\\item 228\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10644\n",
       "2. 228\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10644   228"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(FT)\n",
    "dim(FT)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2caf1c2",
   "metadata": {},
   "source": [
    "Dimension of feature table = 10644 features, 230 columns. Only considering the files belonging to different cycles (Cycles 1-4), --> there are only 144 files. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d5f791b",
   "metadata": {},
   "source": [
    "## Combine the ASVs with the feature table:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "539671b8",
   "metadata": {},
   "source": [
    "Now changing the rownames of FT as combined name of row ID, m/z value and RT and **subsetting the top 2000 abundant features in feature table:**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "64e211b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "rownames(FT) <- paste(rownames(FT),round(FT[,1],digits = 2),round(FT[,3],digits = 3), sep = '_')\n",
    "FT <- FT[,-1:-2]\n",
    "\n",
    "#lets subset ft based on top 2000 abundant features\n",
    "ft_new <- FT #calling FT under a new name: ft_new\n",
    "ft_new$avg <- rowMeans(ft_new) # getting the average of each row (or each feature) under a new column \"avg\"\n",
    "ft_new <- ft_new[with(ft_new,order(-avg)),] # arranged in descending order of avg\n",
    "ft_new <- ft_new[1:2000,]  # Taken the top 2000 features\n",
    "\n",
    "ft <- ft_new[,1:(ncol(ft_new)-1)] #removing the avg column"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e943313d",
   "metadata": {},
   "source": [
    "The order of taxonomic levels: Domain, Kingdom, Phylum or division, class, order, family, genus, species. Lets arrange the ASV files in that order. And we are planning to do the analysis only until genus level and not to the species level"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "a7c7ae08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"CCE_P1706_100.mzxml\" \"CCE_P1706_102.mzxml\" \"CCE_P1706_104.mzxml\"\n",
      " [4] \"CCE_P1706_106.mzxml\" \"CCE_P1706_108.mzxml\" \"CCE_P1706_110.mzxml\"\n",
      " [7] \"CCE_P1706_112.mzxml\" \"CCE_P1706_114.mzxml\" \"CCE_P1706_116.mzxml\"\n",
      "[10] \"CCE_P1706_118.mzxml\" \"CCE_P1706_120.mzxml\" \"CCE_P1706_122.mzxml\"\n",
      "[13] \"CCE_P1706_154.mzxml\" \"CCE_P1706_156.mzxml\" \"CCE_P1706_158.mzxml\"\n",
      "[16] \"CCE_P1706_160.mzxml\" \"CCE_P1706_162.mzxml\" \"CCE_P1706_164.mzxml\"\n",
      "[19] \"CCE_P1706_166.mzxml\" \"CCE_P1706_168.mzxml\" \"CCE_P1706_170.mzxml\"\n",
      "[22] \"CCE_P1706_172.mzxml\" \"CCE_P1706_174.mzxml\" \"CCE_P1706_176.mzxml\"\n",
      "[25] \"CCE_P1706_178.mzxml\" \"CCE_P1706_180.mzxml\" \"CCE_P1706_182.mzxml\"\n",
      "[28] \"CCE_P1706_184.mzxml\" \"CCE_P1706_186.mzxml\" \"CCE_P1706_188.mzxml\"\n",
      "[31] \"CCE_P1706_208.mzxml\" \"CCE_P1706_210.mzxml\" \"CCE_P1706_212.mzxml\"\n",
      "[34] \"CCE_P1706_214.mzxml\" \"CCE_P1706_216.mzxml\" \"CCE_P1706_218.mzxml\"\n",
      "[37] \"CCE_P1706_220.mzxml\" \"CCE_P1706_222.mzxml\" \"CCE_P1706_224.mzxml\"\n",
      "[40] \"CCE_P1706_228.mzxml\" \"CCE_P1706_230.mzxml\" \"CCE_P1706_40.mzxml\" \n",
      "[43] \"CCE_P1706_42.mzxml\"  \"CCE_P1706_44.mzxml\"  \"CCE_P1706_46.mzxml\" \n",
      "[46] \"CCE_P1706_48.mzxml\"  \"CCE_P1706_50.mzxml\"  \"CCE_P1706_52.mzxml\" \n",
      "[49] \"CCE_P1706_54.mzxml\"  \"CCE_P1706_56.mzxml\"  \"CCE_P1706_58.mzxml\" \n",
      "[52] \"CCE_P1706_60.mzxml\"  \"CCE_P1706_62.mzxml\"  \"CCE_P1706_64.mzxml\" \n",
      "[55] \"CCE_P1706_66.mzxml\"  \"CCE_P1706_68.mzxml\"  \"CCE_P1706_70.mzxml\" \n",
      "[58] \"CCE_P1706_72.mzxml\"  \"CCE_P1706_74.mzxml\"  \"CCE_P1706_76.mzxml\" \n",
      "[61] \"CCE_P1706_78.mzxml\"  \"CCE_P1706_80.mzxml\"  \"CCE_P1706_82.mzxml\" \n",
      "[64] \"CCE_P1706_84.mzxml\"  \"CCE_P1706_86.mzxml\"  \"CCE_P1706_88.mzxml\" \n",
      "[67] \"CCE_P1706_90.mzxml\"  \"CCE_P1706_92.mzxml\"  \"CCE_P1706_94.mzxml\" \n",
      "[70] \"CCE_P1706_96.mzxml\"  \"CCE_P1706_98.mzxml\"  \"Class16S\"           \n",
      "[73] \"Confidence\"          \"Domain16S\"           \"Family16S\"          \n",
      "[76] \"Genus16S\"            \"Order16S\"            \"Phylum16S\"          \n",
      "[79] \"Species16S\"         \n"
     ]
    }
   ],
   "source": [
    "print(colnames(asv16s_f))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "b3247493",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"CCE_P1706_100.mzxml\" \"CCE_P1706_102.mzxml\" \"CCE_P1706_104.mzxml\"\n",
      " [4] \"CCE_P1706_106.mzxml\" \"CCE_P1706_108.mzxml\" \"CCE_P1706_110.mzxml\"\n",
      " [7] \"CCE_P1706_112.mzxml\" \"CCE_P1706_114.mzxml\" \"CCE_P1706_116.mzxml\"\n",
      "[10] \"CCE_P1706_118.mzxml\" \"CCE_P1706_120.mzxml\" \"CCE_P1706_122.mzxml\"\n",
      "[13] \"CCE_P1706_154.mzxml\" \"CCE_P1706_156.mzxml\" \"CCE_P1706_158.mzxml\"\n",
      "[16] \"CCE_P1706_160.mzxml\" \"CCE_P1706_162.mzxml\" \"CCE_P1706_164.mzxml\"\n",
      "[19] \"CCE_P1706_166.mzxml\" \"CCE_P1706_168.mzxml\" \"CCE_P1706_170.mzxml\"\n",
      "[22] \"CCE_P1706_172.mzxml\" \"CCE_P1706_174.mzxml\" \"CCE_P1706_176.mzxml\"\n",
      "[25] \"CCE_P1706_178.mzxml\" \"CCE_P1706_180.mzxml\" \"CCE_P1706_182.mzxml\"\n",
      "[28] \"CCE_P1706_184.mzxml\" \"CCE_P1706_186.mzxml\" \"CCE_P1706_188.mzxml\"\n",
      "[31] \"CCE_P1706_208.mzxml\" \"CCE_P1706_210.mzxml\" \"CCE_P1706_212.mzxml\"\n",
      "[34] \"CCE_P1706_214.mzxml\" \"CCE_P1706_216.mzxml\" \"CCE_P1706_218.mzxml\"\n",
      "[37] \"CCE_P1706_220.mzxml\" \"CCE_P1706_222.mzxml\" \"CCE_P1706_224.mzxml\"\n",
      "[40] \"CCE_P1706_228.mzxml\" \"CCE_P1706_230.mzxml\" \"CCE_P1706_40.mzxml\" \n",
      "[43] \"CCE_P1706_42.mzxml\"  \"CCE_P1706_44.mzxml\"  \"CCE_P1706_46.mzxml\" \n",
      "[46] \"CCE_P1706_48.mzxml\"  \"CCE_P1706_50.mzxml\"  \"CCE_P1706_52.mzxml\" \n",
      "[49] \"CCE_P1706_54.mzxml\"  \"CCE_P1706_56.mzxml\"  \"CCE_P1706_58.mzxml\" \n",
      "[52] \"CCE_P1706_60.mzxml\"  \"CCE_P1706_62.mzxml\"  \"CCE_P1706_64.mzxml\" \n",
      "[55] \"CCE_P1706_66.mzxml\"  \"CCE_P1706_68.mzxml\"  \"CCE_P1706_70.mzxml\" \n",
      "[58] \"CCE_P1706_72.mzxml\"  \"CCE_P1706_74.mzxml\"  \"CCE_P1706_76.mzxml\" \n",
      "[61] \"CCE_P1706_78.mzxml\"  \"CCE_P1706_80.mzxml\"  \"CCE_P1706_82.mzxml\" \n",
      "[64] \"CCE_P1706_84.mzxml\"  \"CCE_P1706_86.mzxml\"  \"CCE_P1706_88.mzxml\" \n",
      "[67] \"CCE_P1706_90.mzxml\"  \"CCE_P1706_92.mzxml\"  \"CCE_P1706_94.mzxml\" \n",
      "[70] \"CCE_P1706_96.mzxml\"  \"CCE_P1706_98.mzxml\"  \"Class16S\"           \n",
      "[73] \"Confidence\"          \"Domain16S\"           \"Family16S\"          \n",
      "[76] \"Genus16S\"            \"Order16S\"            \"Phylum16S\"          \n"
     ]
    }
   ],
   "source": [
    "asv16s_new <- asv16s_f[,-79] #removing species column\n",
    "asv16s_new <- asv16s_new[ ,order(names(asv16s_new))] #order the asv16s_new table according to its colnames order\n",
    "name_16s <- colnames(asv16s_new)\n",
    "print(name_16s) #checking the colnames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "6b8f6194",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"CCE_P1706_100.mzxml\" \"CCE_P1706_102.mzxml\" \"CCE_P1706_104.mzxml\"\n",
      " [4] \"CCE_P1706_106.mzxml\" \"CCE_P1706_108.mzxml\" \"CCE_P1706_110.mzxml\"\n",
      " [7] \"CCE_P1706_112.mzxml\" \"CCE_P1706_114.mzxml\" \"CCE_P1706_116.mzxml\"\n",
      "[10] \"CCE_P1706_118.mzxml\" \"CCE_P1706_120.mzxml\" \"CCE_P1706_122.mzxml\"\n",
      "[13] \"CCE_P1706_154.mzxml\" \"CCE_P1706_156.mzxml\" \"CCE_P1706_158.mzxml\"\n",
      "[16] \"CCE_P1706_160.mzxml\" \"CCE_P1706_162.mzxml\" \"CCE_P1706_164.mzxml\"\n",
      "[19] \"CCE_P1706_166.mzxml\" \"CCE_P1706_168.mzxml\" \"CCE_P1706_170.mzxml\"\n",
      "[22] \"CCE_P1706_172.mzxml\" \"CCE_P1706_174.mzxml\" \"CCE_P1706_176.mzxml\"\n",
      "[25] \"CCE_P1706_178.mzxml\" \"CCE_P1706_180.mzxml\" \"CCE_P1706_182.mzxml\"\n",
      "[28] \"CCE_P1706_184.mzxml\" \"CCE_P1706_186.mzxml\" \"CCE_P1706_188.mzxml\"\n",
      "[31] \"CCE_P1706_208.mzxml\" \"CCE_P1706_210.mzxml\" \"CCE_P1706_212.mzxml\"\n",
      "[34] \"CCE_P1706_214.mzxml\" \"CCE_P1706_216.mzxml\" \"CCE_P1706_218.mzxml\"\n",
      "[37] \"CCE_P1706_220.mzxml\" \"CCE_P1706_222.mzxml\" \"CCE_P1706_224.mzxml\"\n",
      "[40] \"CCE_P1706_228.mzxml\" \"CCE_P1706_230.mzxml\" \"CCE_P1706_40.mzxml\" \n",
      "[43] \"CCE_P1706_42.mzxml\"  \"CCE_P1706_44.mzxml\"  \"CCE_P1706_46.mzxml\" \n",
      "[46] \"CCE_P1706_48.mzxml\"  \"CCE_P1706_50.mzxml\"  \"CCE_P1706_52.mzxml\" \n",
      "[49] \"CCE_P1706_54.mzxml\"  \"CCE_P1706_56.mzxml\"  \"CCE_P1706_58.mzxml\" \n",
      "[52] \"CCE_P1706_60.mzxml\"  \"CCE_P1706_62.mzxml\"  \"CCE_P1706_64.mzxml\" \n",
      "[55] \"CCE_P1706_66.mzxml\"  \"CCE_P1706_68.mzxml\"  \"CCE_P1706_70.mzxml\" \n",
      "[58] \"CCE_P1706_72.mzxml\"  \"CCE_P1706_74.mzxml\"  \"CCE_P1706_76.mzxml\" \n",
      "[61] \"CCE_P1706_78.mzxml\"  \"CCE_P1706_80.mzxml\"  \"CCE_P1706_82.mzxml\" \n",
      "[64] \"CCE_P1706_84.mzxml\"  \"CCE_P1706_86.mzxml\"  \"CCE_P1706_88.mzxml\" \n",
      "[67] \"CCE_P1706_90.mzxml\"  \"CCE_P1706_92.mzxml\"  \"CCE_P1706_94.mzxml\" \n",
      "[70] \"CCE_P1706_96.mzxml\"  \"CCE_P1706_98.mzxml\"  \"Domain16S\"          \n",
      "[73] \"Phylum16S\"           \"Class16S\"            \"Order16S\"           \n",
      "[76] \"Family16S\"           \"Genus16S\"            \"Confidence\"         \n"
     ]
    }
   ],
   "source": [
    "name_16s[72:78] <- c( \"Domain16S\",\"Phylum16S\",\"Class16S\" ,\"Order16S\",\"Family16S\",\"Genus16S\", \"Confidence\")\n",
    "print(name_16s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "11e3fd61",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>7534</li>\n",
       "\t<li>77</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 7534\n",
       "\\item 77\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 7534\n",
       "2. 77\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 7534   77"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Arranging asv 16s_new with the above mentioned order of column names\n",
    "asv16s_new <- asv16s_new[, name_16s]\n",
    "asv16s_new <- asv16s_new[,-78] #removing confidence column\n",
    "dim(asv16s_new)\n",
    "\n",
    "#ordering asv16s_new according to the row info from the last few columns\n",
    "ordered_asv16s <- asv16s_new[order(asv16s_new[,72], asv16s_new[,73],asv16s_new[,74], asv16s_new[,75],asv16s_new[,76], asv16s_new[,77] ),]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34c1fdcc",
   "metadata": {},
   "source": [
    "Look at the last few columns to see how they are arranged:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "1dfea949",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>CCE_P1706_100.mzxml</th><th scope=col>CCE_P1706_102.mzxml</th><th scope=col>CCE_P1706_104.mzxml</th><th scope=col>CCE_P1706_106.mzxml</th><th scope=col>CCE_P1706_108.mzxml</th><th scope=col>CCE_P1706_110.mzxml</th><th scope=col>CCE_P1706_112.mzxml</th><th scope=col>CCE_P1706_114.mzxml</th><th scope=col>CCE_P1706_116.mzxml</th><th scope=col>CCE_P1706_118.mzxml</th><th scope=col>...</th><th scope=col>CCE_P1706_92.mzxml</th><th scope=col>CCE_P1706_94.mzxml</th><th scope=col>CCE_P1706_96.mzxml</th><th scope=col>CCE_P1706_98.mzxml</th><th scope=col>Domain16S</th><th scope=col>Phylum16S</th><th scope=col>Class16S</th><th scope=col>Order16S</th><th scope=col>Family16S</th><th scope=col>Genus16S</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>2290e9c9370de78bb0c7c4db94551c3c</th><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>...                         </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>Archaea                     </td><td>Altiarchaeota               </td><td>Altiarchaeia                </td><td>NA                          </td><td>NA                          </td><td>NA                          </td></tr>\n",
       "\t<tr><th scope=row>2b5c93ce3d5388f010a868e8c97b51a0</th><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>...                         </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>Archaea                     </td><td>Crenarchaeota               </td><td>Bathyarchaeia               </td><td>NA                          </td><td>NA                          </td><td>NA                          </td></tr>\n",
       "\t<tr><th scope=row>0bf89996a9ed591dd28d94aeb89664e4</th><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>...                         </td><td>0                           </td><td>0                           </td><td>0                           </td><td>0                           </td><td>Archaea                     </td><td>Crenarchaeota               </td><td>Crenarchaeota Incertae Sedis</td><td>Aigarchaeales               </td><td>Geothermarchaeaceae         </td><td>uncultured marine archaeon  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & CCE\\_P1706\\_100.mzxml & CCE\\_P1706\\_102.mzxml & CCE\\_P1706\\_104.mzxml & CCE\\_P1706\\_106.mzxml & CCE\\_P1706\\_108.mzxml & CCE\\_P1706\\_110.mzxml & CCE\\_P1706\\_112.mzxml & CCE\\_P1706\\_114.mzxml & CCE\\_P1706\\_116.mzxml & CCE\\_P1706\\_118.mzxml & ... & CCE\\_P1706\\_92.mzxml & CCE\\_P1706\\_94.mzxml & CCE\\_P1706\\_96.mzxml & CCE\\_P1706\\_98.mzxml & Domain16S & Phylum16S & Class16S & Order16S & Family16S & Genus16S\\\\\n",
       "\\hline\n",
       "\t2290e9c9370de78bb0c7c4db94551c3c & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & ...                          & 0                            & 0                            & 0                            & 0                            & Archaea                      & Altiarchaeota                & Altiarchaeia                 & NA                           & NA                           & NA                          \\\\\n",
       "\t2b5c93ce3d5388f010a868e8c97b51a0 & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & ...                          & 0                            & 0                            & 0                            & 0                            & Archaea                      & Crenarchaeota                & Bathyarchaeia                & NA                           & NA                           & NA                          \\\\\n",
       "\t0bf89996a9ed591dd28d94aeb89664e4 & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & 0                            & ...                          & 0                            & 0                            & 0                            & 0                            & Archaea                      & Crenarchaeota                & Crenarchaeota Incertae Sedis & Aigarchaeales                & Geothermarchaeaceae          & uncultured marine archaeon  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | CCE_P1706_100.mzxml | CCE_P1706_102.mzxml | CCE_P1706_104.mzxml | CCE_P1706_106.mzxml | CCE_P1706_108.mzxml | CCE_P1706_110.mzxml | CCE_P1706_112.mzxml | CCE_P1706_114.mzxml | CCE_P1706_116.mzxml | CCE_P1706_118.mzxml | ... | CCE_P1706_92.mzxml | CCE_P1706_94.mzxml | CCE_P1706_96.mzxml | CCE_P1706_98.mzxml | Domain16S | Phylum16S | Class16S | Order16S | Family16S | Genus16S |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 2290e9c9370de78bb0c7c4db94551c3c | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | ...                          | 0                            | 0                            | 0                            | 0                            | Archaea                      | Altiarchaeota                | Altiarchaeia                 | NA                           | NA                           | NA                           |\n",
       "| 2b5c93ce3d5388f010a868e8c97b51a0 | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | ...                          | 0                            | 0                            | 0                            | 0                            | Archaea                      | Crenarchaeota                | Bathyarchaeia                | NA                           | NA                           | NA                           |\n",
       "| 0bf89996a9ed591dd28d94aeb89664e4 | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | 0                            | ...                          | 0                            | 0                            | 0                            | 0                            | Archaea                      | Crenarchaeota                | Crenarchaeota Incertae Sedis | Aigarchaeales                | Geothermarchaeaceae          | uncultured marine archaeon   |\n",
       "\n"
      ],
      "text/plain": [
       "                                 CCE_P1706_100.mzxml CCE_P1706_102.mzxml\n",
       "2290e9c9370de78bb0c7c4db94551c3c 0                   0                  \n",
       "2b5c93ce3d5388f010a868e8c97b51a0 0                   0                  \n",
       "0bf89996a9ed591dd28d94aeb89664e4 0                   0                  \n",
       "                                 CCE_P1706_104.mzxml CCE_P1706_106.mzxml\n",
       "2290e9c9370de78bb0c7c4db94551c3c 0                   0                  \n",
       "2b5c93ce3d5388f010a868e8c97b51a0 0                   0                  \n",
       "0bf89996a9ed591dd28d94aeb89664e4 0                   0                  \n",
       "                                 CCE_P1706_108.mzxml CCE_P1706_110.mzxml\n",
       "2290e9c9370de78bb0c7c4db94551c3c 0                   0                  \n",
       "2b5c93ce3d5388f010a868e8c97b51a0 0                   0                  \n",
       "0bf89996a9ed591dd28d94aeb89664e4 0                   0                  \n",
       "                                 CCE_P1706_112.mzxml CCE_P1706_114.mzxml\n",
       "2290e9c9370de78bb0c7c4db94551c3c 0                   0                  \n",
       "2b5c93ce3d5388f010a868e8c97b51a0 0                   0                  \n",
       "0bf89996a9ed591dd28d94aeb89664e4 0                   0                  \n",
       "                                 CCE_P1706_116.mzxml CCE_P1706_118.mzxml ...\n",
       "2290e9c9370de78bb0c7c4db94551c3c 0                   0                   ...\n",
       "2b5c93ce3d5388f010a868e8c97b51a0 0                   0                   ...\n",
       "0bf89996a9ed591dd28d94aeb89664e4 0                   0                   ...\n",
       "                                 CCE_P1706_92.mzxml CCE_P1706_94.mzxml\n",
       "2290e9c9370de78bb0c7c4db94551c3c 0                  0                 \n",
       "2b5c93ce3d5388f010a868e8c97b51a0 0                  0                 \n",
       "0bf89996a9ed591dd28d94aeb89664e4 0                  0                 \n",
       "                                 CCE_P1706_96.mzxml CCE_P1706_98.mzxml\n",
       "2290e9c9370de78bb0c7c4db94551c3c 0                  0                 \n",
       "2b5c93ce3d5388f010a868e8c97b51a0 0                  0                 \n",
       "0bf89996a9ed591dd28d94aeb89664e4 0                  0                 \n",
       "                                 Domain16S Phylum16S    \n",
       "2290e9c9370de78bb0c7c4db94551c3c Archaea   Altiarchaeota\n",
       "2b5c93ce3d5388f010a868e8c97b51a0 Archaea   Crenarchaeota\n",
       "0bf89996a9ed591dd28d94aeb89664e4 Archaea   Crenarchaeota\n",
       "                                 Class16S                     Order16S     \n",
       "2290e9c9370de78bb0c7c4db94551c3c Altiarchaeia                 NA           \n",
       "2b5c93ce3d5388f010a868e8c97b51a0 Bathyarchaeia                NA           \n",
       "0bf89996a9ed591dd28d94aeb89664e4 Crenarchaeota Incertae Sedis Aigarchaeales\n",
       "                                 Family16S           Genus16S                  \n",
       "2290e9c9370de78bb0c7c4db94551c3c NA                  NA                        \n",
       "2b5c93ce3d5388f010a868e8c97b51a0 NA                  NA                        \n",
       "0bf89996a9ed591dd28d94aeb89664e4 Geothermarchaeaceae uncultured marine archaeon"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ordered_asv16s, n=3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "428cdaae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"The total number of genus levels in ASV 16S:438\"\n"
     ]
    }
   ],
   "source": [
    "genus_levels <-levels(as.factor(ordered_asv16s$Genus16S)) # getting the genus levels\n",
    "print(paste0('The total number of genus levels in ASV 16S:',length(genus_levels))) # there are 438 genus levels"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "308a4e38",
   "metadata": {},
   "source": [
    "<b>Combining the feature abundances according to each genus level in ASV 16S: </b>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "15a2df44",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "33.7271037961242"
      ],
      "text/latex": [
       "33.7271037961242"
      ],
      "text/markdown": [
       "33.7271037961242"
      ],
      "text/plain": [
       "[1] 33.7271"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "my_genusSet <- c()\n",
    "\n",
    "for (i in 1: length(genus_levels)){\n",
    "  \n",
    "  #Filtering all features of ordered_asv16s with a particular gene name in Genus16S column  \n",
    "  each_gene <- ordered_asv16s %>% filter(Genus16S == genus_levels[i])\n",
    "  \n",
    "  # If it is just one feature, subset it, else, get a column sum in such a way that one row for each gene\n",
    "  if(nrow(each_gene) == 1){\n",
    "    subset_sum <- each_gene \n",
    "  } else {\n",
    "    subset_sum <- as.data.frame(t(colSums(as.matrix(sapply(each_gene[,1:71],as.numeric)))))\n",
    "    subset_sum <- cbind(subset_sum,each_gene[1,72:77])\n",
    "  }\n",
    "  my_genusSet <- rbind(my_genusSet,subset_sum)\n",
    "}\n",
    "\n",
    "NArows <- nrow(ordered_asv16s[ordered_asv16s$Genus16S== \"NA\",]) #no.of rows with genus_level as NA\n",
    "NApercent <- (NArows/nrow(ordered_asv16s))*100 #33.73% NAs\n",
    "NApercent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6f261cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(my_genusSet,\"ASV16s_GenusLevels_34percent_NAs.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "689c10cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>CCE_P1706_100.mzxml</th><th scope=col>CCE_P1706_102.mzxml</th><th scope=col>CCE_P1706_104.mzxml</th><th scope=col>CCE_P1706_106.mzxml</th><th scope=col>CCE_P1706_108.mzxml</th><th scope=col>CCE_P1706_110.mzxml</th><th scope=col>CCE_P1706_112.mzxml</th><th scope=col>CCE_P1706_114.mzxml</th><th scope=col>CCE_P1706_116.mzxml</th><th scope=col>CCE_P1706_118.mzxml</th><th scope=col>...</th><th scope=col>CCE_P1706_92.mzxml</th><th scope=col>CCE_P1706_94.mzxml</th><th scope=col>CCE_P1706_96.mzxml</th><th scope=col>CCE_P1706_98.mzxml</th><th scope=col>Domain16S</th><th scope=col>Phylum16S</th><th scope=col>Class16S</th><th scope=col>Order16S</th><th scope=col>Family16S</th><th scope=col>Genus16S</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>0711eb9dfdaec37d1a640cec68d811c3</th><td>0                                 </td><td> 9                                </td><td> 0                                </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td> 20                               </td><td>  7                               </td><td>  0                               </td><td>0                                 </td><td>...                               </td><td> 22                               </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td>Bacteria                          </td><td>Proteobacteria                    </td><td>Gammaproteobacteria               </td><td>Francisellales                    </td><td>Francisellaceae                   </td><td>[Caedibacter] taeniospiralis group</td></tr>\n",
       "\t<tr><th scope=row>480938aa486ad1d47a4846ccbefeb82e</th><td>0                                 </td><td>77                                </td><td>80                                </td><td>58                                </td><td>0                                 </td><td>0                                 </td><td>169                               </td><td>192                               </td><td>103                               </td><td>0                                 </td><td>...                               </td><td>140                               </td><td>24                                </td><td>0                                 </td><td>0                                 </td><td>Bacteria                          </td><td>Verrucomicrobia                   </td><td>Verrucomicrobiae                  </td><td>Opitutales                        </td><td>Puniceicoccaceae                  </td><td>A714019                           </td></tr>\n",
       "\t<tr><th scope=row>08c731f2101b546847bea9fff028196d</th><td>0                                 </td><td> 0                                </td><td> 0                                </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td>  0                               </td><td>  0                               </td><td>  0                               </td><td>0                                 </td><td>...                               </td><td>  0                               </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td>Bacteria                          </td><td>Acidobacteria                     </td><td>Holophagae                        </td><td>Acanthopleuribacterales           </td><td>Acanthopleuribacteraceae          </td><td>Acanthopleuribacter               </td></tr>\n",
       "\t<tr><th scope=row>a085e7a88e2ce90054e3be1a1999d416</th><td>0                                 </td><td> 0                                </td><td> 0                                </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td>  0                               </td><td>  0                               </td><td>  0                               </td><td>0                                 </td><td>...                               </td><td>  0                               </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td>Bacteria                          </td><td>Proteobacteria                    </td><td>Gammaproteobacteria               </td><td>Betaproteobacteriales             </td><td>Burkholderiaceae                  </td><td>Achromobacter                     </td></tr>\n",
       "\t<tr><th scope=row>9eaf3126b1a59d53885da5e5b61ab98f</th><td>0                                 </td><td> 0                                </td><td> 0                                </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td>  0                               </td><td>  0                               </td><td>  0                               </td><td>0                                 </td><td>...                               </td><td>  0                               </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td>Bacteria                          </td><td>Proteobacteria                    </td><td>Gammaproteobacteria               </td><td>Pseudomonadales                   </td><td>Moraxellaceae                     </td><td>Acinetobacter                     </td></tr>\n",
       "\t<tr><th scope=row>60f1c2b419383905edf4fd8e7c3f22be</th><td>0                                 </td><td> 0                                </td><td> 0                                </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td>  0                               </td><td>  0                               </td><td>  0                               </td><td>0                                 </td><td>...                               </td><td>  0                               </td><td> 0                                </td><td>0                                 </td><td>0                                 </td><td>Bacteria                          </td><td>Proteobacteria                    </td><td>Gammaproteobacteria               </td><td>Alteromonadales                   </td><td>Alteromonadaceae                  </td><td>Aestuariibacter                   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & CCE\\_P1706\\_100.mzxml & CCE\\_P1706\\_102.mzxml & CCE\\_P1706\\_104.mzxml & CCE\\_P1706\\_106.mzxml & CCE\\_P1706\\_108.mzxml & CCE\\_P1706\\_110.mzxml & CCE\\_P1706\\_112.mzxml & CCE\\_P1706\\_114.mzxml & CCE\\_P1706\\_116.mzxml & CCE\\_P1706\\_118.mzxml & ... & CCE\\_P1706\\_92.mzxml & CCE\\_P1706\\_94.mzxml & CCE\\_P1706\\_96.mzxml & CCE\\_P1706\\_98.mzxml & Domain16S & Phylum16S & Class16S & Order16S & Family16S & Genus16S\\\\\n",
       "\\hline\n",
       "\t0711eb9dfdaec37d1a640cec68d811c3 & 0                                      &  9                                     &  0                                     &  0                                     & 0                                      & 0                                      &  20                                    &   7                                    &   0                                    & 0                                      & ...                                    &  22                                    &  0                                     & 0                                      & 0                                      & Bacteria                               & Proteobacteria                         & Gammaproteobacteria                    & Francisellales                         & Francisellaceae                        & {[}Caedibacter{]} taeniospiralis group\\\\\n",
       "\t480938aa486ad1d47a4846ccbefeb82e & 0                                  & 77                                 & 80                                 & 58                                 & 0                                  & 0                                  & 169                                & 192                                & 103                                & 0                                  & ...                                & 140                                & 24                                 & 0                                  & 0                                  & Bacteria                           & Verrucomicrobia                    & Verrucomicrobiae                   & Opitutales                         & Puniceicoccaceae                   & A714019                           \\\\\n",
       "\t08c731f2101b546847bea9fff028196d & 0                                  &  0                                 &  0                                 &  0                                 & 0                                  & 0                                  &   0                                &   0                                &   0                                & 0                                  & ...                                &   0                                &  0                                 & 0                                  & 0                                  & Bacteria                           & Acidobacteria                      & Holophagae                         & Acanthopleuribacterales            & Acanthopleuribacteraceae           & Acanthopleuribacter               \\\\\n",
       "\ta085e7a88e2ce90054e3be1a1999d416 & 0                                  &  0                                 &  0                                 &  0                                 & 0                                  & 0                                  &   0                                &   0                                &   0                                & 0                                  & ...                                &   0                                &  0                                 & 0                                  & 0                                  & Bacteria                           & Proteobacteria                     & Gammaproteobacteria                & Betaproteobacteriales              & Burkholderiaceae                   & Achromobacter                     \\\\\n",
       "\t9eaf3126b1a59d53885da5e5b61ab98f & 0                                  &  0                                 &  0                                 &  0                                 & 0                                  & 0                                  &   0                                &   0                                &   0                                & 0                                  & ...                                &   0                                &  0                                 & 0                                  & 0                                  & Bacteria                           & Proteobacteria                     & Gammaproteobacteria                & Pseudomonadales                    & Moraxellaceae                      & Acinetobacter                     \\\\\n",
       "\t60f1c2b419383905edf4fd8e7c3f22be & 0                                  &  0                                 &  0                                 &  0                                 & 0                                  & 0                                  &   0                                &   0                                &   0                                & 0                                  & ...                                &   0                                &  0                                 & 0                                  & 0                                  & Bacteria                           & Proteobacteria                     & Gammaproteobacteria                & Alteromonadales                    & Alteromonadaceae                   & Aestuariibacter                   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | CCE_P1706_100.mzxml | CCE_P1706_102.mzxml | CCE_P1706_104.mzxml | CCE_P1706_106.mzxml | CCE_P1706_108.mzxml | CCE_P1706_110.mzxml | CCE_P1706_112.mzxml | CCE_P1706_114.mzxml | CCE_P1706_116.mzxml | CCE_P1706_118.mzxml | ... | CCE_P1706_92.mzxml | CCE_P1706_94.mzxml | CCE_P1706_96.mzxml | CCE_P1706_98.mzxml | Domain16S | Phylum16S | Class16S | Order16S | Family16S | Genus16S |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 0711eb9dfdaec37d1a640cec68d811c3 | 0                                  |  9                                 |  0                                 |  0                                 | 0                                  | 0                                  |  20                                |   7                                |   0                                | 0                                  | ...                                |  22                                |  0                                 | 0                                  | 0                                  | Bacteria                           | Proteobacteria                     | Gammaproteobacteria                | Francisellales                     | Francisellaceae                    | [Caedibacter] taeniospiralis group |\n",
       "| 480938aa486ad1d47a4846ccbefeb82e | 0                                  | 77                                 | 80                                 | 58                                 | 0                                  | 0                                  | 169                                | 192                                | 103                                | 0                                  | ...                                | 140                                | 24                                 | 0                                  | 0                                  | Bacteria                           | Verrucomicrobia                    | Verrucomicrobiae                   | Opitutales                         | Puniceicoccaceae                   | A714019                            |\n",
       "| 08c731f2101b546847bea9fff028196d | 0                                  |  0                                 |  0                                 |  0                                 | 0                                  | 0                                  |   0                                |   0                                |   0                                | 0                                  | ...                                |   0                                |  0                                 | 0                                  | 0                                  | Bacteria                           | Acidobacteria                      | Holophagae                         | Acanthopleuribacterales            | Acanthopleuribacteraceae           | Acanthopleuribacter                |\n",
       "| a085e7a88e2ce90054e3be1a1999d416 | 0                                  |  0                                 |  0                                 |  0                                 | 0                                  | 0                                  |   0                                |   0                                |   0                                | 0                                  | ...                                |   0                                |  0                                 | 0                                  | 0                                  | Bacteria                           | Proteobacteria                     | Gammaproteobacteria                | Betaproteobacteriales              | Burkholderiaceae                   | Achromobacter                      |\n",
       "| 9eaf3126b1a59d53885da5e5b61ab98f | 0                                  |  0                                 |  0                                 |  0                                 | 0                                  | 0                                  |   0                                |   0                                |   0                                | 0                                  | ...                                |   0                                |  0                                 | 0                                  | 0                                  | Bacteria                           | Proteobacteria                     | Gammaproteobacteria                | Pseudomonadales                    | Moraxellaceae                      | Acinetobacter                      |\n",
       "| 60f1c2b419383905edf4fd8e7c3f22be | 0                                  |  0                                 |  0                                 |  0                                 | 0                                  | 0                                  |   0                                |   0                                |   0                                | 0                                  | ...                                |   0                                |  0                                 | 0                                  | 0                                  | Bacteria                           | Proteobacteria                     | Gammaproteobacteria                | Alteromonadales                    | Alteromonadaceae                   | Aestuariibacter                    |\n",
       "\n"
      ],
      "text/plain": [
       "                                 CCE_P1706_100.mzxml CCE_P1706_102.mzxml\n",
       "0711eb9dfdaec37d1a640cec68d811c3 0                    9                 \n",
       "480938aa486ad1d47a4846ccbefeb82e 0                   77                 \n",
       "08c731f2101b546847bea9fff028196d 0                    0                 \n",
       "a085e7a88e2ce90054e3be1a1999d416 0                    0                 \n",
       "9eaf3126b1a59d53885da5e5b61ab98f 0                    0                 \n",
       "60f1c2b419383905edf4fd8e7c3f22be 0                    0                 \n",
       "                                 CCE_P1706_104.mzxml CCE_P1706_106.mzxml\n",
       "0711eb9dfdaec37d1a640cec68d811c3  0                   0                 \n",
       "480938aa486ad1d47a4846ccbefeb82e 80                  58                 \n",
       "08c731f2101b546847bea9fff028196d  0                   0                 \n",
       "a085e7a88e2ce90054e3be1a1999d416  0                   0                 \n",
       "9eaf3126b1a59d53885da5e5b61ab98f  0                   0                 \n",
       "60f1c2b419383905edf4fd8e7c3f22be  0                   0                 \n",
       "                                 CCE_P1706_108.mzxml CCE_P1706_110.mzxml\n",
       "0711eb9dfdaec37d1a640cec68d811c3 0                   0                  \n",
       "480938aa486ad1d47a4846ccbefeb82e 0                   0                  \n",
       "08c731f2101b546847bea9fff028196d 0                   0                  \n",
       "a085e7a88e2ce90054e3be1a1999d416 0                   0                  \n",
       "9eaf3126b1a59d53885da5e5b61ab98f 0                   0                  \n",
       "60f1c2b419383905edf4fd8e7c3f22be 0                   0                  \n",
       "                                 CCE_P1706_112.mzxml CCE_P1706_114.mzxml\n",
       "0711eb9dfdaec37d1a640cec68d811c3  20                   7                \n",
       "480938aa486ad1d47a4846ccbefeb82e 169                 192                \n",
       "08c731f2101b546847bea9fff028196d   0                   0                \n",
       "a085e7a88e2ce90054e3be1a1999d416   0                   0                \n",
       "9eaf3126b1a59d53885da5e5b61ab98f   0                   0                \n",
       "60f1c2b419383905edf4fd8e7c3f22be   0                   0                \n",
       "                                 CCE_P1706_116.mzxml CCE_P1706_118.mzxml ...\n",
       "0711eb9dfdaec37d1a640cec68d811c3   0                 0                   ...\n",
       "480938aa486ad1d47a4846ccbefeb82e 103                 0                   ...\n",
       "08c731f2101b546847bea9fff028196d   0                 0                   ...\n",
       "a085e7a88e2ce90054e3be1a1999d416   0                 0                   ...\n",
       "9eaf3126b1a59d53885da5e5b61ab98f   0                 0                   ...\n",
       "60f1c2b419383905edf4fd8e7c3f22be   0                 0                   ...\n",
       "                                 CCE_P1706_92.mzxml CCE_P1706_94.mzxml\n",
       "0711eb9dfdaec37d1a640cec68d811c3  22                 0                \n",
       "480938aa486ad1d47a4846ccbefeb82e 140                24                \n",
       "08c731f2101b546847bea9fff028196d   0                 0                \n",
       "a085e7a88e2ce90054e3be1a1999d416   0                 0                \n",
       "9eaf3126b1a59d53885da5e5b61ab98f   0                 0                \n",
       "60f1c2b419383905edf4fd8e7c3f22be   0                 0                \n",
       "                                 CCE_P1706_96.mzxml CCE_P1706_98.mzxml\n",
       "0711eb9dfdaec37d1a640cec68d811c3 0                  0                 \n",
       "480938aa486ad1d47a4846ccbefeb82e 0                  0                 \n",
       "08c731f2101b546847bea9fff028196d 0                  0                 \n",
       "a085e7a88e2ce90054e3be1a1999d416 0                  0                 \n",
       "9eaf3126b1a59d53885da5e5b61ab98f 0                  0                 \n",
       "60f1c2b419383905edf4fd8e7c3f22be 0                  0                 \n",
       "                                 Domain16S Phylum16S       Class16S           \n",
       "0711eb9dfdaec37d1a640cec68d811c3 Bacteria  Proteobacteria  Gammaproteobacteria\n",
       "480938aa486ad1d47a4846ccbefeb82e Bacteria  Verrucomicrobia Verrucomicrobiae   \n",
       "08c731f2101b546847bea9fff028196d Bacteria  Acidobacteria   Holophagae         \n",
       "a085e7a88e2ce90054e3be1a1999d416 Bacteria  Proteobacteria  Gammaproteobacteria\n",
       "9eaf3126b1a59d53885da5e5b61ab98f Bacteria  Proteobacteria  Gammaproteobacteria\n",
       "60f1c2b419383905edf4fd8e7c3f22be Bacteria  Proteobacteria  Gammaproteobacteria\n",
       "                                 Order16S               \n",
       "0711eb9dfdaec37d1a640cec68d811c3 Francisellales         \n",
       "480938aa486ad1d47a4846ccbefeb82e Opitutales             \n",
       "08c731f2101b546847bea9fff028196d Acanthopleuribacterales\n",
       "a085e7a88e2ce90054e3be1a1999d416 Betaproteobacteriales  \n",
       "9eaf3126b1a59d53885da5e5b61ab98f Pseudomonadales        \n",
       "60f1c2b419383905edf4fd8e7c3f22be Alteromonadales        \n",
       "                                 Family16S               \n",
       "0711eb9dfdaec37d1a640cec68d811c3 Francisellaceae         \n",
       "480938aa486ad1d47a4846ccbefeb82e Puniceicoccaceae        \n",
       "08c731f2101b546847bea9fff028196d Acanthopleuribacteraceae\n",
       "a085e7a88e2ce90054e3be1a1999d416 Burkholderiaceae        \n",
       "9eaf3126b1a59d53885da5e5b61ab98f Moraxellaceae           \n",
       "60f1c2b419383905edf4fd8e7c3f22be Alteromonadaceae        \n",
       "                                 Genus16S                          \n",
       "0711eb9dfdaec37d1a640cec68d811c3 [Caedibacter] taeniospiralis group\n",
       "480938aa486ad1d47a4846ccbefeb82e A714019                           \n",
       "08c731f2101b546847bea9fff028196d Acanthopleuribacter               \n",
       "a085e7a88e2ce90054e3be1a1999d416 Achromobacter                     \n",
       "9eaf3126b1a59d53885da5e5b61ab98f Acinetobacter                     \n",
       "60f1c2b419383905edf4fd8e7c3f22be Aestuariibacter                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(my_genusSet)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "75818ab5",
   "metadata": {},
   "source": [
    "**Repeating the same steps for ASV_18SV9:**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "c0b2cd81",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>10282</li>\n",
       "\t<li>78</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 10282\n",
       "\\item 78\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 10282\n",
       "2. 78\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 10282    78"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"The total number of genus levels in ASV 18S:603\"\n"
     ]
    }
   ],
   "source": [
    "asv18s_new <- asv18s_f[,-80:-81] # removing species & X columns\n",
    "asv18s_new <- asv18s_new[ ,order(names(asv18s_new))] #ordering it by the column names\n",
    "name_18s <- colnames(asv18s_new)\n",
    "\n",
    "#Rearranging the column names:\n",
    "name_18s[72:79] <- c( \"Domain18SV9\",\"Kingdom18SV9\",\"Phylum18SV9\",\"Class18SV9\" ,\"Order18SV9\",\"Family18SV9\",\"Genus18SV9\", \"Confidence\")\n",
    "asv18s_new <- asv18s_new[, name_18s] # rearranging asv 18s_new table according to the ordered column names \n",
    "asv18s_new <- asv18s_new[,-79] #removing confidence column\n",
    "dim(asv18s_new)\n",
    "\n",
    "#ordering asv18s_new according to the row info from the last few columns\n",
    "ordered_asv18s <- asv18s_new[order( asv18s_new[,72], asv18s_new[,73],asv18s_new[,74], asv18s_new[,75],asv18s_new[,76], asv18s_new[,77],asv18s_new[,78] ),]\n",
    "\n",
    "genus_levels_18s <-levels(as.factor(ordered_asv18s$Genus18SV9)) # getting the genus levels\n",
    "print(paste0('The total number of genus levels in ASV 18S:',length(genus_levels_18s))) # there are 603 genus levels"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f396fef",
   "metadata": {},
   "source": [
    "<b>Combining the feature abundances according to each genus level in ASV 18SV9: </b>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "2987b65b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "59.9688776502626"
      ],
      "text/latex": [
       "59.9688776502626"
      ],
      "text/markdown": [
       "59.9688776502626"
      ],
      "text/plain": [
       "[1] 59.96888"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "my_genusSet2 <- c()\n",
    "\n",
    "for (i in 1: length(genus_levels_18s)){\n",
    "  \n",
    "  each_gene <- ordered_asv18s %>% filter(Genus18SV9 == genus_levels_18s[i])\n",
    "  \n",
    "  if(nrow(each_gene) == 1){\n",
    "    subset_sum <- each_gene\n",
    "  } else {\n",
    "    subset_sum <- as.data.frame(t(colSums(as.matrix(sapply(each_gene[,1:71],as.numeric)))))\n",
    "    subset_sum <- cbind(subset_sum,each_gene[1,72:78])\n",
    "  }\n",
    "  my_genusSet2 <- rbind(my_genusSet2,subset_sum)\n",
    "}\n",
    "\n",
    "NArows_18sv9 <- nrow(ordered_asv18s[ordered_asv18s$Genus18SV9== \"NA\",])\n",
    "NApercent_18sv9 <- (NArows_18sv9/nrow(ordered_asv18s))*100 #59.96% NAs\n",
    "NApercent_18sv9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a63c04d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(my_genusSet2,\"ASV18s_GenusLevels_60percent_NAs.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8f7e4eb",
   "metadata": {},
   "source": [
    "After preparing the ASVs, lets change their rownames and then combine them with ft. For the rownames, we are combinig the taxonomic names and making it as the rownames. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "110f4987",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#combining the taxonomic levels as rownames\n",
    "rownames(my_genusSet) <- paste(my_genusSet[,72],my_genusSet[,73],my_genusSet[,74],my_genusSet[,75],my_genusSet[,76],my_genusSet[,77], sep = '_')\n",
    "Final_asv16s <- my_genusSet[,1:71] #taking only feature columns and excluding the taxonomic columns as that info is in rownames now\n",
    "\n",
    "#Similarly for ASV 18SV9:\n",
    "rownames(my_genusSet2) <- paste(my_genusSet2[,72],my_genusSet2[,73],my_genusSet2[,74],my_genusSet2[,75],my_genusSet2[,76],my_genusSet2[,77],my_genusSet2[,78], sep = '_')\n",
    "Final_asv18sv9 <- my_genusSet2[,1:71]\n",
    "\n",
    "#Checking if both asv files have same column names or same CCE file names:\n",
    "identical(colnames(Final_asv16s),colnames(Final_asv18sv9)) #should return TRUE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "1c8a7318",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"CCE_P1706_100.mzxml\" \"CCE_P1706_102.mzxml\" \"CCE_P1706_104.mzxml\"\n",
      " [4] \"CCE_P1706_106.mzxml\" \"CCE_P1706_108.mzxml\" \"CCE_P1706_110.mzxml\"\n",
      " [7] \"CCE_P1706_112.mzxml\" \"CCE_P1706_114.mzxml\" \"CCE_P1706_116.mzxml\"\n",
      "[10] \"CCE_P1706_118.mzxml\" \"CCE_P1706_120.mzxml\" \"CCE_P1706_122.mzxml\"\n",
      "[13] \"CCE_P1706_154.mzxml\" \"CCE_P1706_156.mzxml\" \"CCE_P1706_158.mzxml\"\n",
      "[16] \"CCE_P1706_160.mzxml\" \"CCE_P1706_162.mzxml\" \"CCE_P1706_164.mzxml\"\n",
      "[19] \"CCE_P1706_166.mzxml\" \"CCE_P1706_168.mzxml\" \"CCE_P1706_170.mzxml\"\n",
      "[22] \"CCE_P1706_172.mzxml\" \"CCE_P1706_174.mzxml\" \"CCE_P1706_176.mzxml\"\n",
      "[25] \"CCE_P1706_178.mzxml\" \"CCE_P1706_180.mzxml\" \"CCE_P1706_182.mzxml\"\n",
      "[28] \"CCE_P1706_184.mzxml\" \"CCE_P1706_186.mzxml\" \"CCE_P1706_188.mzxml\"\n",
      "[31] \"CCE_P1706_208.mzxml\" \"CCE_P1706_210.mzxml\" \"CCE_P1706_212.mzxml\"\n",
      "[34] \"CCE_P1706_214.mzxml\" \"CCE_P1706_216.mzxml\" \"CCE_P1706_218.mzxml\"\n",
      "[37] \"CCE_P1706_220.mzxml\" \"CCE_P1706_222.mzxml\" \"CCE_P1706_224.mzxml\"\n",
      "[40] \"CCE_P1706_228.mzxml\" \"CCE_P1706_230.mzxml\" \"CCE_P1706_40.mzxml\" \n",
      "[43] \"CCE_P1706_42.mzxml\"  \"CCE_P1706_44.mzxml\"  \"CCE_P1706_46.mzxml\" \n",
      "[46] \"CCE_P1706_48.mzxml\"  \"CCE_P1706_50.mzxml\"  \"CCE_P1706_52.mzxml\" \n",
      "[49] \"CCE_P1706_54.mzxml\"  \"CCE_P1706_56.mzxml\"  \"CCE_P1706_58.mzxml\" \n",
      "[52] \"CCE_P1706_60.mzxml\"  \"CCE_P1706_62.mzxml\"  \"CCE_P1706_64.mzxml\" \n",
      "[55] \"CCE_P1706_66.mzxml\"  \"CCE_P1706_68.mzxml\"  \"CCE_P1706_70.mzxml\" \n",
      "[58] \"CCE_P1706_72.mzxml\"  \"CCE_P1706_74.mzxml\"  \"CCE_P1706_76.mzxml\" \n",
      "[61] \"CCE_P1706_78.mzxml\"  \"CCE_P1706_80.mzxml\"  \"CCE_P1706_82.mzxml\" \n",
      "[64] \"CCE_P1706_84.mzxml\"  \"CCE_P1706_86.mzxml\"  \"CCE_P1706_88.mzxml\" \n",
      "[67] \"CCE_P1706_90.mzxml\"  \"CCE_P1706_92.mzxml\"  \"CCE_P1706_94.mzxml\" \n",
      "[70] \"CCE_P1706_96.mzxml\"  \"CCE_P1706_98.mzxml\" \n"
     ]
    }
   ],
   "source": [
    "#Picking only the CCE files (from the original ft) that corresponds to the asv files to perform correlation:\n",
    "Corr_ft<- ft[,which(colnames(ft)%in%colnames(Final_asv16s))]\n",
    "Corr_ft <- Corr_ft[,order(colnames(Corr_ft))]  #ordering the ft by its column names\n",
    "print(colnames(Corr_ft)) #only 71 CCE files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "6fccb155",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "identical(colnames(Final_asv16s),colnames(Corr_ft)) #should return TRUE\n",
    "identical(colnames(Final_asv18sv9),colnames(Corr_ft)) #should return TRUE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "8379a0ec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>1041</li>\n",
       "\t<li>71</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1041\n",
       "\\item 71\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1041\n",
       "2. 71\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1041   71"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>780</li>\n",
       "\t<li>71</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 780\n",
       "\\item 71\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 780\n",
       "2. 71\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 780  71"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Final_asv <- rbind(Final_asv16s,Final_asv18sv9) #combine both asv files\n",
    "Final_asv1 <- as.data.frame(Final_asv,stringsAsFactors=F) #converting the table from Matrix form to dataframe \n",
    "Final_asv1 <- as.data.frame(sapply(Final_asv1, as.numeric)) #applying as.numeric to all columns/rows of Final_asv\n",
    "rownames(Final_asv1)<- rownames(Final_asv) # The rownames disappear with the previous step. Hence we define it again\n",
    "dim(Final_asv1)\n",
    "\n",
    "OhneZero_ASV <- Final_asv1[rowSums(Final_asv1 != 0) > 0,] #excluding all the rows that has only zeros\n",
    "dim(OhneZero_ASV)\n",
    "\n",
    "Target <- rbind(OhneZero_ASV,Corr_ft)\n",
    "\n",
    "#Sampling only the colnames and then adding it to the table\n",
    "set.seed(873465)# Seed for reproducibility\n",
    "Decoy_asv1 <- OhneZero_ASV\n",
    "colnames(Decoy_asv1) <- sample(colnames(OhneZero_ASV))\n",
    "\n",
    "Decoy_1 <- rbind(Decoy_asv1,Corr_ft)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "7eb28d4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>CCE_P1706_100.mzxml</th><th scope=col>CCE_P1706_102.mzxml</th><th scope=col>CCE_P1706_104.mzxml</th><th scope=col>CCE_P1706_106.mzxml</th><th scope=col>CCE_P1706_108.mzxml</th><th scope=col>CCE_P1706_110.mzxml</th><th scope=col>CCE_P1706_112.mzxml</th><th scope=col>CCE_P1706_114.mzxml</th><th scope=col>CCE_P1706_116.mzxml</th><th scope=col>CCE_P1706_118.mzxml</th><th scope=col>...</th><th scope=col>CCE_P1706_80.mzxml</th><th scope=col>CCE_P1706_82.mzxml</th><th scope=col>CCE_P1706_84.mzxml</th><th scope=col>CCE_P1706_86.mzxml</th><th scope=col>CCE_P1706_88.mzxml</th><th scope=col>CCE_P1706_90.mzxml</th><th scope=col>CCE_P1706_92.mzxml</th><th scope=col>CCE_P1706_94.mzxml</th><th scope=col>CCE_P1706_96.mzxml</th><th scope=col>CCE_P1706_98.mzxml</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group</th><td>0  </td><td> 9 </td><td> 0 </td><td> 0 </td><td>0  </td><td>0  </td><td> 20</td><td>  7</td><td>  0</td><td>0  </td><td>...</td><td> 17</td><td> 0 </td><td>0  </td><td>0  </td><td> 0 </td><td>12 </td><td> 22</td><td> 0 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019</th><td>0  </td><td>77 </td><td>80 </td><td>58 </td><td>0  </td><td>0  </td><td>169</td><td>192</td><td>103</td><td>0  </td><td>...</td><td>158</td><td> 0 </td><td>0  </td><td>0  </td><td>54 </td><td>57 </td><td>140</td><td>24 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter</th><td>0  </td><td> 0 </td><td> 0 </td><td> 0 </td><td>0  </td><td>0  </td><td>  0</td><td>  0</td><td>  0</td><td>0  </td><td>...</td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td><td> 0 </td><td> 4 </td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter</th><td>0  </td><td> 0 </td><td> 0 </td><td> 0 </td><td>0  </td><td>0  </td><td>  0</td><td>  0</td><td>  0</td><td>0  </td><td>...</td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td><td> 0 </td><td> 0 </td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter</th><td>0  </td><td> 0 </td><td> 0 </td><td> 0 </td><td>0  </td><td>0  </td><td>  0</td><td>  0</td><td>  0</td><td>0  </td><td>...</td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td><td> 0 </td><td> 0 </td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax</th><td>0  </td><td>30 </td><td> 0 </td><td>20 </td><td>0  </td><td>0  </td><td>  0</td><td> 14</td><td>  4</td><td>0  </td><td>...</td><td> 38</td><td>12 </td><td>0  </td><td>0  </td><td> 0 </td><td> 3 </td><td>  0</td><td> 9 </td><td>0  </td><td>0  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & CCE\\_P1706\\_100.mzxml & CCE\\_P1706\\_102.mzxml & CCE\\_P1706\\_104.mzxml & CCE\\_P1706\\_106.mzxml & CCE\\_P1706\\_108.mzxml & CCE\\_P1706\\_110.mzxml & CCE\\_P1706\\_112.mzxml & CCE\\_P1706\\_114.mzxml & CCE\\_P1706\\_116.mzxml & CCE\\_P1706\\_118.mzxml & ... & CCE\\_P1706\\_80.mzxml & CCE\\_P1706\\_82.mzxml & CCE\\_P1706\\_84.mzxml & CCE\\_P1706\\_86.mzxml & CCE\\_P1706\\_88.mzxml & CCE\\_P1706\\_90.mzxml & CCE\\_P1706\\_92.mzxml & CCE\\_P1706\\_94.mzxml & CCE\\_P1706\\_96.mzxml & CCE\\_P1706\\_98.mzxml\\\\\n",
       "\\hline\n",
       "\tBacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Francisellales\\_Francisellaceae\\_{[}Caedibacter{]} taeniospiralis group & 0   &  9  &  0  &  0  & 0   & 0   &  20 &   7 &   0 & 0   & ... &  17 &  0  & 0   & 0   &  0  & 12  &  22 &  0  & 0   & 0  \\\\\n",
       "\tBacteria\\_Verrucomicrobia\\_Verrucomicrobiae\\_Opitutales\\_Puniceicoccaceae\\_A714019 & 0   & 77  & 80  & 58  & 0   & 0   & 169 & 192 & 103 & 0   & ... & 158 &  0  & 0   & 0   & 54  & 57  & 140 & 24  & 0   & 0  \\\\\n",
       "\tBacteria\\_Acidobacteria\\_Holophagae\\_Acanthopleuribacterales\\_Acanthopleuribacteraceae\\_Acanthopleuribacter & 0   &  0  &  0  &  0  & 0   & 0   &   0 &   0 &   0 & 0   & ... &   0 &  0  & 0   & 0   &  0  &  4  &   0 &  0  & 0   & 0  \\\\\n",
       "\tBacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Betaproteobacteriales\\_Burkholderiaceae\\_Achromobacter & 0   &  0  &  0  &  0  & 0   & 0   &   0 &   0 &   0 & 0   & ... &   0 &  0  & 0   & 0   &  0  &  0  &   0 &  0  & 0   & 0  \\\\\n",
       "\tBacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Pseudomonadales\\_Moraxellaceae\\_Acinetobacter & 0   &  0  &  0  &  0  & 0   & 0   &   0 &   0 &   0 & 0   & ... &   0 &  0  & 0   & 0   &  0  &  0  &   0 &  0  & 0   & 0  \\\\\n",
       "\tBacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Oceanospirillales\\_Alcanivoracaceae\\_Alcanivorax & 0   & 30  &  0  & 20  & 0   & 0   &   0 &  14 &   4 & 0   & ... &  38 & 12  & 0   & 0   &  0  &  3  &   0 &  9  & 0   & 0  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | CCE_P1706_100.mzxml | CCE_P1706_102.mzxml | CCE_P1706_104.mzxml | CCE_P1706_106.mzxml | CCE_P1706_108.mzxml | CCE_P1706_110.mzxml | CCE_P1706_112.mzxml | CCE_P1706_114.mzxml | CCE_P1706_116.mzxml | CCE_P1706_118.mzxml | ... | CCE_P1706_80.mzxml | CCE_P1706_82.mzxml | CCE_P1706_84.mzxml | CCE_P1706_86.mzxml | CCE_P1706_88.mzxml | CCE_P1706_90.mzxml | CCE_P1706_92.mzxml | CCE_P1706_94.mzxml | CCE_P1706_96.mzxml | CCE_P1706_98.mzxml |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group | 0   |  9  |  0  |  0  | 0   | 0   |  20 |   7 |   0 | 0   | ... |  17 |  0  | 0   | 0   |  0  | 12  |  22 |  0  | 0   | 0   |\n",
       "| Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019 | 0   | 77  | 80  | 58  | 0   | 0   | 169 | 192 | 103 | 0   | ... | 158 |  0  | 0   | 0   | 54  | 57  | 140 | 24  | 0   | 0   |\n",
       "| Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter | 0   |  0  |  0  |  0  | 0   | 0   |   0 |   0 |   0 | 0   | ... |   0 |  0  | 0   | 0   |  0  |  4  |   0 |  0  | 0   | 0   |\n",
       "| Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter | 0   |  0  |  0  |  0  | 0   | 0   |   0 |   0 |   0 | 0   | ... |   0 |  0  | 0   | 0   |  0  |  0  |   0 |  0  | 0   | 0   |\n",
       "| Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter | 0   |  0  |  0  |  0  | 0   | 0   |   0 |   0 |   0 | 0   | ... |   0 |  0  | 0   | 0   |  0  |  0  |   0 |  0  | 0   | 0   |\n",
       "| Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax | 0   | 30  |  0  | 20  | 0   | 0   |   0 |  14 |   4 | 0   | ... |  38 | 12  | 0   | 0   |  0  |  3  |   0 |  9  | 0   | 0   |\n",
       "\n"
      ],
      "text/plain": [
       "                                                                                                              CCE_P1706_100.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              CCE_P1706_102.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  9                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 77                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    30                 \n",
       "                                                                                                              CCE_P1706_104.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 80                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     0                 \n",
       "                                                                                                              CCE_P1706_106.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 58                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    20                 \n",
       "                                                                                                              CCE_P1706_108.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              CCE_P1706_110.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              CCE_P1706_112.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  20                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 169                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                      0                \n",
       "                                                                                                              CCE_P1706_114.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group   7                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 192                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     14                \n",
       "                                                                                                              CCE_P1706_116.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group   0                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 103                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                      4                \n",
       "                                                                                                              CCE_P1706_118.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              ...\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group ...\n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 ...\n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        ...\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              ...\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       ...\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    ...\n",
       "                                                                                                              CCE_P1706_80.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  17               \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 158               \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     38               \n",
       "                                                                                                              CCE_P1706_82.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                  0                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    12                \n",
       "                                                                                                              CCE_P1706_84.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                 \n",
       "                                                                                                              CCE_P1706_86.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                 \n",
       "                                                                                                              CCE_P1706_88.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 54                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     0                \n",
       "                                                                                                              CCE_P1706_90.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 12                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 57                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         4                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     3                \n",
       "                                                                                                              CCE_P1706_92.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  22               \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 140               \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                      0               \n",
       "                                                                                                              CCE_P1706_94.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 24                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     9                \n",
       "                                                                                                              CCE_P1706_96.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                 \n",
       "                                                                                                              CCE_P1706_98.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>2780</li>\n",
       "\t<li>71</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2780\n",
       "\\item 71\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2780\n",
       "2. 71\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2780   71"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(Target)\n",
    "dim(Target)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "954668ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>CCE_P1706_228.mzxml</th><th scope=col>CCE_P1706_46.mzxml</th><th scope=col>CCE_P1706_158.mzxml</th><th scope=col>CCE_P1706_162.mzxml</th><th scope=col>CCE_P1706_188.mzxml</th><th scope=col>CCE_P1706_174.mzxml</th><th scope=col>CCE_P1706_216.mzxml</th><th scope=col>CCE_P1706_60.mzxml</th><th scope=col>CCE_P1706_94.mzxml</th><th scope=col>CCE_P1706_172.mzxml</th><th scope=col>...</th><th scope=col>CCE_P1706_114.mzxml</th><th scope=col>CCE_P1706_218.mzxml</th><th scope=col>CCE_P1706_186.mzxml</th><th scope=col>CCE_P1706_208.mzxml</th><th scope=col>CCE_P1706_220.mzxml</th><th scope=col>CCE_P1706_100.mzxml</th><th scope=col>CCE_P1706_106.mzxml</th><th scope=col>CCE_P1706_164.mzxml</th><th scope=col>CCE_P1706_122.mzxml</th><th scope=col>CCE_P1706_156.mzxml</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group</th><td>0  </td><td> 9 </td><td> 0 </td><td> 0 </td><td>0  </td><td>0  </td><td> 20</td><td>  7</td><td>  0</td><td>0  </td><td>...</td><td> 17</td><td> 0 </td><td>0  </td><td>0  </td><td> 0 </td><td>12 </td><td> 22</td><td> 0 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019</th><td>0  </td><td>77 </td><td>80 </td><td>58 </td><td>0  </td><td>0  </td><td>169</td><td>192</td><td>103</td><td>0  </td><td>...</td><td>158</td><td> 0 </td><td>0  </td><td>0  </td><td>54 </td><td>57 </td><td>140</td><td>24 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter</th><td>0  </td><td> 0 </td><td> 0 </td><td> 0 </td><td>0  </td><td>0  </td><td>  0</td><td>  0</td><td>  0</td><td>0  </td><td>...</td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td><td> 0 </td><td> 4 </td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter</th><td>0  </td><td> 0 </td><td> 0 </td><td> 0 </td><td>0  </td><td>0  </td><td>  0</td><td>  0</td><td>  0</td><td>0  </td><td>...</td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td><td> 0 </td><td> 0 </td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter</th><td>0  </td><td> 0 </td><td> 0 </td><td> 0 </td><td>0  </td><td>0  </td><td>  0</td><td>  0</td><td>  0</td><td>0  </td><td>...</td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td><td> 0 </td><td> 0 </td><td>  0</td><td> 0 </td><td>0  </td><td>0  </td></tr>\n",
       "\t<tr><th scope=row>Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax</th><td>0  </td><td>30 </td><td> 0 </td><td>20 </td><td>0  </td><td>0  </td><td>  0</td><td> 14</td><td>  4</td><td>0  </td><td>...</td><td> 38</td><td>12 </td><td>0  </td><td>0  </td><td> 0 </td><td> 3 </td><td>  0</td><td> 9 </td><td>0  </td><td>0  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & CCE\\_P1706\\_228.mzxml & CCE\\_P1706\\_46.mzxml & CCE\\_P1706\\_158.mzxml & CCE\\_P1706\\_162.mzxml & CCE\\_P1706\\_188.mzxml & CCE\\_P1706\\_174.mzxml & CCE\\_P1706\\_216.mzxml & CCE\\_P1706\\_60.mzxml & CCE\\_P1706\\_94.mzxml & CCE\\_P1706\\_172.mzxml & ... & CCE\\_P1706\\_114.mzxml & CCE\\_P1706\\_218.mzxml & CCE\\_P1706\\_186.mzxml & CCE\\_P1706\\_208.mzxml & CCE\\_P1706\\_220.mzxml & CCE\\_P1706\\_100.mzxml & CCE\\_P1706\\_106.mzxml & CCE\\_P1706\\_164.mzxml & CCE\\_P1706\\_122.mzxml & CCE\\_P1706\\_156.mzxml\\\\\n",
       "\\hline\n",
       "\tBacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Francisellales\\_Francisellaceae\\_{[}Caedibacter{]} taeniospiralis group & 0   &  9  &  0  &  0  & 0   & 0   &  20 &   7 &   0 & 0   & ... &  17 &  0  & 0   & 0   &  0  & 12  &  22 &  0  & 0   & 0  \\\\\n",
       "\tBacteria\\_Verrucomicrobia\\_Verrucomicrobiae\\_Opitutales\\_Puniceicoccaceae\\_A714019 & 0   & 77  & 80  & 58  & 0   & 0   & 169 & 192 & 103 & 0   & ... & 158 &  0  & 0   & 0   & 54  & 57  & 140 & 24  & 0   & 0  \\\\\n",
       "\tBacteria\\_Acidobacteria\\_Holophagae\\_Acanthopleuribacterales\\_Acanthopleuribacteraceae\\_Acanthopleuribacter & 0   &  0  &  0  &  0  & 0   & 0   &   0 &   0 &   0 & 0   & ... &   0 &  0  & 0   & 0   &  0  &  4  &   0 &  0  & 0   & 0  \\\\\n",
       "\tBacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Betaproteobacteriales\\_Burkholderiaceae\\_Achromobacter & 0   &  0  &  0  &  0  & 0   & 0   &   0 &   0 &   0 & 0   & ... &   0 &  0  & 0   & 0   &  0  &  0  &   0 &  0  & 0   & 0  \\\\\n",
       "\tBacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Pseudomonadales\\_Moraxellaceae\\_Acinetobacter & 0   &  0  &  0  &  0  & 0   & 0   &   0 &   0 &   0 & 0   & ... &   0 &  0  & 0   & 0   &  0  &  0  &   0 &  0  & 0   & 0  \\\\\n",
       "\tBacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Oceanospirillales\\_Alcanivoracaceae\\_Alcanivorax & 0   & 30  &  0  & 20  & 0   & 0   &   0 &  14 &   4 & 0   & ... &  38 & 12  & 0   & 0   &  0  &  3  &   0 &  9  & 0   & 0  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | CCE_P1706_228.mzxml | CCE_P1706_46.mzxml | CCE_P1706_158.mzxml | CCE_P1706_162.mzxml | CCE_P1706_188.mzxml | CCE_P1706_174.mzxml | CCE_P1706_216.mzxml | CCE_P1706_60.mzxml | CCE_P1706_94.mzxml | CCE_P1706_172.mzxml | ... | CCE_P1706_114.mzxml | CCE_P1706_218.mzxml | CCE_P1706_186.mzxml | CCE_P1706_208.mzxml | CCE_P1706_220.mzxml | CCE_P1706_100.mzxml | CCE_P1706_106.mzxml | CCE_P1706_164.mzxml | CCE_P1706_122.mzxml | CCE_P1706_156.mzxml |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group | 0   |  9  |  0  |  0  | 0   | 0   |  20 |   7 |   0 | 0   | ... |  17 |  0  | 0   | 0   |  0  | 12  |  22 |  0  | 0   | 0   |\n",
       "| Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019 | 0   | 77  | 80  | 58  | 0   | 0   | 169 | 192 | 103 | 0   | ... | 158 |  0  | 0   | 0   | 54  | 57  | 140 | 24  | 0   | 0   |\n",
       "| Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter | 0   |  0  |  0  |  0  | 0   | 0   |   0 |   0 |   0 | 0   | ... |   0 |  0  | 0   | 0   |  0  |  4  |   0 |  0  | 0   | 0   |\n",
       "| Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter | 0   |  0  |  0  |  0  | 0   | 0   |   0 |   0 |   0 | 0   | ... |   0 |  0  | 0   | 0   |  0  |  0  |   0 |  0  | 0   | 0   |\n",
       "| Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter | 0   |  0  |  0  |  0  | 0   | 0   |   0 |   0 |   0 | 0   | ... |   0 |  0  | 0   | 0   |  0  |  0  |   0 |  0  | 0   | 0   |\n",
       "| Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax | 0   | 30  |  0  | 20  | 0   | 0   |   0 |  14 |   4 | 0   | ... |  38 | 12  | 0   | 0   |  0  |  3  |   0 |  9  | 0   | 0   |\n",
       "\n"
      ],
      "text/plain": [
       "                                                                                                              CCE_P1706_228.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              CCE_P1706_46.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  9                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 77                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    30                \n",
       "                                                                                                              CCE_P1706_158.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 80                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     0                 \n",
       "                                                                                                              CCE_P1706_162.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 58                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    20                 \n",
       "                                                                                                              CCE_P1706_188.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              CCE_P1706_174.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              CCE_P1706_216.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  20                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 169                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                      0                \n",
       "                                                                                                              CCE_P1706_60.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group   7               \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 192               \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     14               \n",
       "                                                                                                              CCE_P1706_94.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group   0               \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 103               \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0               \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                      4               \n",
       "                                                                                                              CCE_P1706_172.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              ...\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group ...\n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 ...\n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        ...\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              ...\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       ...\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    ...\n",
       "                                                                                                              CCE_P1706_114.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  17                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 158                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     38                \n",
       "                                                                                                              CCE_P1706_218.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                  0                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    12                 \n",
       "                                                                                                              CCE_P1706_186.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              CCE_P1706_208.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              CCE_P1706_220.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 54                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     0                 \n",
       "                                                                                                              CCE_P1706_100.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 12                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 57                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         4                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     3                 \n",
       "                                                                                                              CCE_P1706_106.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  22                \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 140                \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter          0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter                0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                         0                \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                      0                \n",
       "                                                                                                              CCE_P1706_164.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group  0                 \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 24                 \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter         0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter               0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                        0                 \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                     9                 \n",
       "                                                                                                              CCE_P1706_122.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  \n",
       "                                                                                                              CCE_P1706_156.mzxml\n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group 0                  \n",
       "Bacteria_Verrucomicrobia_Verrucomicrobiae_Opitutales_Puniceicoccaceae_A714019                                 0                  \n",
       "Bacteria_Acidobacteria_Holophagae_Acanthopleuribacterales_Acanthopleuribacteraceae_Acanthopleuribacter        0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Betaproteobacteriales_Burkholderiaceae_Achromobacter              0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Pseudomonadales_Moraxellaceae_Acinetobacter                       0                  \n",
       "Bacteria_Proteobacteria_Gammaproteobacteria_Oceanospirillales_Alcanivoracaceae_Alcanivorax                    0                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>2780</li>\n",
       "\t<li>71</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 2780\n",
       "\\item 71\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 2780\n",
       "2. 71\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 2780   71"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(Decoy_1)\n",
    "dim(Decoy_1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93e6e14f",
   "metadata": {},
   "source": [
    "## Correlation Analysis:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "7dee799a",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset <- list(Target, Decoy_1)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a344723e",
   "metadata": {},
   "source": [
    "<b>Took around 20-25 mins to run the next cell:</b> The below cell gets the correlation scores table for both Target and Decoy."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "0ab346c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "final <- list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "id": "da722c6e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using SampleNames as id variables\n",
      "Using SampleNames as id variables\n",
      "Using SampleNames as id variables\n",
      "Using SampleNames as id variables\n"
     ]
    }
   ],
   "source": [
    "\n",
    "for(val in 1:length(dataset)) {\n",
    "\n",
    "data_D <- dataset[[val]] \n",
    "\n",
    "nonfeatures <- c(1:(length(rownames(data_D)[grepl(\"\\\\b[A-Z][a-z]*\\\\d*\", rownames(data_D))])))\n",
    "\n",
    "# Transpose your data table and put it into the format that will be used for the correlation analysis---------------------\n",
    "data_D_t <- t(data_D)\n",
    "data_D_t <- as.data.frame(data_D_t,stringsAsFactors=F)\n",
    "data_D_t <- as.data.frame(sapply(data_D_t, as.numeric))\n",
    "rownames(data_D_t)<- colnames(data_D)\n",
    "\n",
    "tab2_D <- data_D_t\n",
    "tab2_D$SampleNames <- rownames(tab2_D)\n",
    "tab2_D <- tab2_D[,c(which(colnames(tab2_D)==\"SampleNames\"),1:(ncol(tab2_D)-1))]\n",
    "\n",
    "# Calculate the correlation of each feature with metadata:------------------------------------------------------------------\n",
    "ct <- list()\n",
    "\n",
    "for (i in 2:length(nonfeatures)){\n",
    "  ct[[i]] <- t(sapply((length(nonfeatures)+1):ncol(tab2_D), function(x) unlist(cor.test(scale(tab2_D[,i])[,1], scale(tab2_D[,x])[,1])[c(\"estimate\", \"p.value\")])))\n",
    "}\n",
    "\n",
    "ctm <- list()\n",
    "for (i in 2:length(ct)){\n",
    "  \n",
    "  ph <- matrix(0,nrow=length(nonfeatures)-1,ncol=2)\n",
    "  \n",
    "  ct1 <- rbind(c(\"cor\",\" p_value\"), ph, ct[[i]])\n",
    "  tab3 <- rbind(t(ct1),  as.matrix(tab2_D))\n",
    "  rownames(tab3) <- NULL\n",
    "  tab3[1:5, 1:5]\n",
    "  #write.csv(tab3, \"features_quantification_matrix_edited_with_correlation.csv\", row.names=FALSE)\n",
    "  \n",
    "  # Tranpose the table for molecular networking mapping in Cytoscape\n",
    "  new = t(tab3)\n",
    "  colnames(new) = new[1,]\n",
    "  new = new[-1,]\n",
    "  new = cbind(c(rep(0,length(nonfeatures)-1),1:(nrow(new)-length(nonfeatures)+1)), rownames(new),new)\n",
    "  rownames(new) <- NULL\n",
    "  colnames(new)[1:2] <- c(\"shared name\", \"IDs\")\n",
    "  new[1:(length(nonfeatures)-1),1] <- \"\"\n",
    "  #new[1:5,1:5]\n",
    "  \n",
    "  # Features passing Bonferronii method\n",
    "  # Prepare the new table\n",
    "  new2 <- cbind(new[,1:4], c(rep(0,(length(nonfeatures)-1)), p.adjust(as.numeric(ct1[-c(1:length(nonfeatures)),2]), method = \"bonferroni\")), new[,-c(1:4)])\n",
    "  colnames(new2)[5] <- \"p_value_corrected\"\n",
    "  #new2[1:20,1:10]\n",
    "  ctm[[i]] <- new2\n",
    "}\n",
    "                     \n",
    "names_ctm <- colnames(tab2_D)[nonfeatures] \n",
    "names(ctm) <- names_ctm\n",
    "x <- ctm[[2]][,2]\n",
    "x2 <- ctm[[2]][,2]# change here if you want to change to feature numbers instead of names ctm[[2]][,1]\n",
    "\n",
    "for (i in 2:length(ctm)){\n",
    "  x <- cbind(x,ctm[[i]][,which(colnames(ctm[[i]])==\"p_value_corrected\")])\n",
    "  x2 <- cbind(x2,ctm[[i]][,which(colnames(ctm[[i]])==\"cor\")])\n",
    "}\n",
    "\n",
    "colnames(x) <- names(ctm)\n",
    "x <- x[-nonfeatures[-(length(nonfeatures))],]\n",
    "colnames(x2) <- names(ctm)\n",
    "x2 <- x2[-nonfeatures[-(length(nonfeatures))],]\n",
    "\n",
    "# create p-value and correlation matrices for heatmap\n",
    "correl <- apply(x2[,-1], 2, as.numeric)\n",
    "rownames(correl) <- x2[,1]\n",
    "\n",
    "# create output tables\n",
    "y <- as.data.frame(x,stringsAsFactors = F)\n",
    "y2 <- as.data.frame(x2,stringsAsFactors = F)\n",
    "y$SampleNames <- as.factor(y$SampleNames)\n",
    "y2$SampleNames <- as.factor(y2$SampleNames)\n",
    "y[,-1] <- sapply(y[,-1], as.numeric)\n",
    "y2[,-1] <- sapply(y2[,-1], as.numeric)\n",
    "\n",
    "y <- melt(y)\n",
    "y2 <- melt(y2)\n",
    "\n",
    "y3 <- cbind(y[,c(1:3)],y2[,3])\n",
    "y3 <- cbind(y3, (y2[,3])^2)\n",
    "colnames(y3)[c(1,3:5)] <- c(\"Feature\",\"pvalue\",\"correlation_coef\",\"r2\")\n",
    "\n",
    "dim(y3)\n",
    "\n",
    "final[[val]] <- y3                    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "id": "cdfa1387",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "2"
      ],
      "text/latex": [
       "2"
      ],
      "text/markdown": [
       "2"
      ],
      "text/plain": [
       "[1] 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(final)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "id": "19af90de",
   "metadata": {},
   "outputs": [],
   "source": [
    "hist_Target <- final[[1]]\n",
    "hist_Decoy <- final[[2]]\n",
    "\n",
    "write.table(hist_Target,\"20220915_CorrelationTable_Metadata_Target.csv\",row.names = F,quote = F,sep=\";\")\n",
    "write.table(hist_Decoy,\"20220915_CorrelationTable_Metadata_Decoy_colnames_shuffled.csv\",row.names = F,quote = F,sep=\";\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "41b61572",
   "metadata": {},
   "outputs": [],
   "source": [
    "hist_Target <- read.csv(\"20220915_CorrelationTable_Metadata_Target.csv\",sep=\";\")\n",
    "hist_Decoy <- read.csv(\"20220915_CorrelationTable_Metadata_Decoy_colnames_shuffled.csv\",sep=\";\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c282bfc6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>Feature</th><th scope=col>variable</th><th scope=col>pvalue</th><th scope=col>correlation_coef</th><th scope=col>r2</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>Eukaryota_Alveolata_Dinoflagellata_Dinophyceae_Dinophyceae_X_Tovelliaceae_Woloszynskia                       </td><td>Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group</td><td>1                                                                                                            </td><td>-0.1832473                                                                                                   </td><td>0.03357959                                                                                                   </td></tr>\n",
       "\t<tr><td>4969_312.14_0.001_0_0.001                                                                                    </td><td>Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group</td><td>1                                                                                                            </td><td>-0.1579660                                                                                                   </td><td>0.02495327                                                                                                   </td></tr>\n",
       "\t<tr><td>34346_542.32_0_0_0                                                                                           </td><td>Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group</td><td>1                                                                                                            </td><td>-0.2800746                                                                                                   </td><td>0.07844179                                                                                                   </td></tr>\n",
       "\t<tr><td>17812_354.19_0_0_0                                                                                           </td><td>Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group</td><td>1                                                                                                            </td><td>-0.1146186                                                                                                   </td><td>0.01313742                                                                                                   </td></tr>\n",
       "\t<tr><td>37207_641.42_0_0_0                                                                                           </td><td>Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group</td><td>1                                                                                                            </td><td>-0.2622471                                                                                                   </td><td>0.06877356                                                                                                   </td></tr>\n",
       "\t<tr><td>35342_568.34_0_0_0                                                                                           </td><td>Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group</td><td>1                                                                                                            </td><td>-0.2771756                                                                                                   </td><td>0.07682634                                                                                                   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllll}\n",
       " Feature & variable & pvalue & correlation\\_coef & r2\\\\\n",
       "\\hline\n",
       "\t Eukaryota\\_Alveolata\\_Dinoflagellata\\_Dinophyceae\\_Dinophyceae\\_X\\_Tovelliaceae\\_Woloszynskia                        & Bacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Francisellales\\_Francisellaceae\\_{[}Caedibacter{]} taeniospiralis group & 1                                                                                                                           & -0.1832473                                                                                                                  & 0.03357959                                                                                                                 \\\\\n",
       "\t 4969\\_312.14\\_0.001\\_0\\_0.001                                                                                           & Bacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Francisellales\\_Francisellaceae\\_{[}Caedibacter{]} taeniospiralis group & 1                                                                                                                           & -0.1579660                                                                                                                  & 0.02495327                                                                                                                 \\\\\n",
       "\t 34346\\_542.32\\_0\\_0\\_0                                                                                                  & Bacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Francisellales\\_Francisellaceae\\_{[}Caedibacter{]} taeniospiralis group & 1                                                                                                                           & -0.2800746                                                                                                                  & 0.07844179                                                                                                                 \\\\\n",
       "\t 17812\\_354.19\\_0\\_0\\_0                                                                                                  & Bacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Francisellales\\_Francisellaceae\\_{[}Caedibacter{]} taeniospiralis group & 1                                                                                                                           & -0.1146186                                                                                                                  & 0.01313742                                                                                                                 \\\\\n",
       "\t 37207\\_641.42\\_0\\_0\\_0                                                                                                  & Bacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Francisellales\\_Francisellaceae\\_{[}Caedibacter{]} taeniospiralis group & 1                                                                                                                           & -0.2622471                                                                                                                  & 0.06877356                                                                                                                 \\\\\n",
       "\t 35342\\_568.34\\_0\\_0\\_0                                                                                                  & Bacteria\\_Proteobacteria\\_Gammaproteobacteria\\_Francisellales\\_Francisellaceae\\_{[}Caedibacter{]} taeniospiralis group & 1                                                                                                                           & -0.2771756                                                                                                                  & 0.07682634                                                                                                                 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| Feature | variable | pvalue | correlation_coef | r2 |\n",
       "|---|---|---|---|---|\n",
       "| Eukaryota_Alveolata_Dinoflagellata_Dinophyceae_Dinophyceae_X_Tovelliaceae_Woloszynskia                        | Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group | 1                                                                                                             | -0.1832473                                                                                                    | 0.03357959                                                                                                    |\n",
       "| 4969_312.14_0.001_0_0.001                                                                                     | Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group | 1                                                                                                             | -0.1579660                                                                                                    | 0.02495327                                                                                                    |\n",
       "| 34346_542.32_0_0_0                                                                                            | Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group | 1                                                                                                             | -0.2800746                                                                                                    | 0.07844179                                                                                                    |\n",
       "| 17812_354.19_0_0_0                                                                                            | Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group | 1                                                                                                             | -0.1146186                                                                                                    | 0.01313742                                                                                                    |\n",
       "| 37207_641.42_0_0_0                                                                                            | Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group | 1                                                                                                             | -0.2622471                                                                                                    | 0.06877356                                                                                                    |\n",
       "| 35342_568.34_0_0_0                                                                                            | Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group | 1                                                                                                             | -0.2771756                                                                                                    | 0.07682634                                                                                                    |\n",
       "\n"
      ],
      "text/plain": [
       "  Feature                                                                               \n",
       "1 Eukaryota_Alveolata_Dinoflagellata_Dinophyceae_Dinophyceae_X_Tovelliaceae_Woloszynskia\n",
       "2 4969_312.14_0.001_0_0.001                                                             \n",
       "3 34346_542.32_0_0_0                                                                    \n",
       "4 17812_354.19_0_0_0                                                                    \n",
       "5 37207_641.42_0_0_0                                                                    \n",
       "6 35342_568.34_0_0_0                                                                    \n",
       "  variable                                                                                                     \n",
       "1 Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group\n",
       "2 Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group\n",
       "3 Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group\n",
       "4 Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group\n",
       "5 Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group\n",
       "6 Bacteria_Proteobacteria_Gammaproteobacteria_Francisellales_Francisellaceae_[Caedibacter] taeniospiralis group\n",
       "  pvalue correlation_coef r2        \n",
       "1 1      -0.1832473       0.03357959\n",
       "2 1      -0.1579660       0.02495327\n",
       "3 1      -0.2800746       0.07844179\n",
       "4 1      -0.1146186       0.01313742\n",
       "5 1      -0.2622471       0.06877356\n",
       "6 1      -0.2771756       0.07682634"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(hist_Target)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3e23aff6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlgAAAJYCAMAAACJuGjuAAAAMFBMVEUAAABNTU1oaGh8fHyM\njIyampqnp6eysrK9vb3Hx8fQ0NDZ2dnh4eHp6enw8PD////QFLu4AAAACXBIWXMAABJ0AAAS\ndAHeZh94AAAbTUlEQVR4nO2diXqzKhQAMXuzmPd/27origrIMaIz371/01SBgxNEMKK+AAKo\nXxcA9gligQiIBSIgFoiAWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBYIAJigQiIBSIgFoiA\nWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBYIAJigQiIBSIgFoiAWCACYoEIiAUiIBaIgFgg\nAmKBCIgFIiAWiIBYIAJigQiIBSIgFoiAWOujjlDpEjGqpuaqV8pYlVeBrIfcE6XanHolsS2Y\n6uJZkDrVvECnh2ci3RKN5uFfxnn06pzgZ2K9klU+t/dcBSexTAVbLlaT6rlM5OaVilai0TwE\nxepV5wQ/E0vyY9XhpNR7PFfbgi0Xq97vUafyntnBMj3Te4JVe7Iu+CpiTW8kitW5b2oPp33n\nU82OzF/x38KOwI8+q/ZZrNlipff8PHD5+zZNQLHR85o3r89ql0/22/nR2fNzKs4bf5fs9en2\nqdN7nNTplbUAiTq/9Oy19AxivbJSXD/dv44UbDSsQWmqMupl/6a3RCXFNm2q+b/Ff0m5RSba\nuQ69X/Ay1Tb5Nr22MG1B9DzMNfHMQzc1ORMFMWY7z4pifZIq9rN2/KpOh7oUe7yqTdo9T8UO\n9VbqVb5b/v65Ne81dNMbaJL9Uu6SfGYLNh7WoDRlGXtlr5N9dVNNSrFK6i1ug4J3Um1edNJr\nCtMpyFCsQYJl6IaT2VRBjGHMs6JY2ccgaxPSrNiPThEvddWUcSTNr/WeKt8t65qc0+/3pvmS\n6dGtgRItvUFNtKlf5wo2EZapNH+Dste/Jt1yZHldm3SaHZ7DimhSHSSftIXpFqTNo8rfmGAT\nusZUQYxhzCMkll7FTbx5M5Fmn8Dmve8z+/lIs5NRGVXW+UjyH0m7Z15zedfko6WUvZv3g0/v\n4kebdy89w6kweRVHIpkpmDGs6pWpNIOylwc9La1tdv6UR6w4zWRbJO9C6JOx4EWqzQtTeoOC\ndApqSDAv3tUQ31RBzGHMs6JYeYU3J+66iNe8mci5FZ+kS1kJ+UGq93z2ki7/fWk/2g166RnE\nytNL5wtmDMv4Tl3GXtmzXysxLt2dX2ULcG53SE/3j7Hgz26Rx9LTC9L5bSTB1BDfVEGmsp1i\nRbHu5RvXpr6qH0W5i49y2QUx/jnb4O92VloN6j+a/brpmTrvg12NBTOG1f6il6bIslf2tgoS\nfee/4s3bfMFT7Q8j6Q2r5dtL4GP663hkvf1Gw5hhzavCW13Cz+BP9SulH5z2979Tx9RpsbRX\nNmIZCzYVlrE0/bK3R6Rfjqz7XJx07AreSX6Y3kRBxhK0EKv7ajyMadYU65v+lRccZ+1Pzecj\nMbZYxa/5p/x0fbxtxOqmZyeWqWBTYRlL0y97oqVTbZTkxVOq9ymaLPgg+c675oI0vxkTtBDL\neETGdh5hVbFynletFi6zfazir6fq/VmxLrN9LMuCTYVlLI2hj/Uc7JzlcW/EOne7NqMF7yff\nfddckCZ/Y4KG+KYKYgzDghXFOjWdhfoDlI5eFareka9+zrdY81eFlgWzCKvXUPTK/ldegP61\n7WAVbvb/p3hz7qpQz3aYnqkgafPuWIIGN6YKYsrWhhXFykp8/hRd5XwcLr84yn82Q3zl8Mpw\nHKtI6Fxs/ExmxeqnZyOWuWATYRlLMzqOVVy61qk2g0R/3S0eEwUfJt9eCusFqfOo9hhJ0OTG\nVEEM2dqw5qmw7iPn6hfjKcWLczeK4vOitJH34u1qUDsfEX1Ni9VLz0askYKNh2UszaDs1a+l\npf1wi9cv44B3r+D95Muty3f1gtR51HuYEzS5MVUQQ7Y2rNrHKrox5+pepEtT8mvSGUd6X4tJ\nq740+dvJ9f2pB9QNqddo6VmJNVKw0bCMpRmUvZyAuzx7qT5O7f1Y+TRcs8VIwdvku+lV72oF\nqfNoVTQlaHRjoiCmbC2QECsAaTU0HiOzZbc+ODGzsRhV2fl4n/UJwCiwLjtirU/bcdSnctZG\nadjts5WyW+ATnmsWIql609zBsvje3WX41PxWym7B8cT6pvf8gjy5/vgz71XzGym7BQcUC3YC\nYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBYIAJigQiIBSIgFoiAWCACYoEIiAUiIBaIgFggAmKB\nCIgFIiAWiIBYIAJigQiIBSIgFoiAWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBYIAJigQiI\nBSIgFoiAWCACYoEI/mK97uVKVpfbK2B5YCf4ipWeOg8KNy6WBYfGV6ybSv7exavPM9n+Ugyw\nNr5iJerdvH7Hu1IXSOErlrZQxiGWswInaLFAhAV9rOeneEUfCwx4n8Q6q/Opk3HVdzgyC8ax\nbsU4VnK5M44FA+h2gwiI5Y7U2pG7gikddxQfx3mY0nEHsSxgSscdxLKAAVJ3FL2seZjScUfR\nZs1Di+UOYlnAlI47iGUBUzruIJYFTOm4g1gWUEHuIJYFVJA7iGUBUzruIJYFTOm4g1gWMKXj\nDmJZwACpO4hlgdCUjurimcVmUcwWzrNCi7WzI5ArRZs1ywpTOjs7AuqLWBasMKWzsyOAWFas\nMKWzsyOAWFasUD07OwKIZQViuYJYVqwwpbOzI4BYVqwwpbOzI4BYVqwwpbOzI9CIxRDpFAyQ\nuqJos2xY4Vs6O6t/xLKCFssVxLKCKR1XEMsKpnRcQSwrmNJxBbGsYOTdFcSyArFcQSwrllfO\n7DjhzuofsaxALFcQywr/AVLr29p3Vv+IZYVv5bwSxNpZYGHxrpz0os7FCCmnQjCwoHL+lPr7\nHlos7m8YZ0nVfM7qkh5ZrL2FFpJlVXNXyROxwMDCqnmf5s8HO6t9xLJicdVcEQsMMKXjCmJZ\ngViuIJYViOUKYlmBWK4glhWI5QpiWYFYbrRPx0KsSRDLDUWLZQdiudETi9nCMRDLjX6Ltavg\nQoJYbiCWJYjlBmJZglhuIJYliOUGYlmCWG4MxOK60AxiuTFssXYVXjgQyw3EsgSx3EAsSxDL\nDcSyBLHcQCxLEMsNxLIEsdxALEsQyw2DWIxkmUAsN0wt1q4CDAViuYFYliCWG4hlCWK5gViW\nIJYbiGUJYrmBWJYglhuIZQliuYFYlvhXyjGX7kUsS3wr5ahL9xrFYvB9iG+NHHPp3v4X7JsW\na0cxBsK3Ro65EKZZK8Qy4Fsjx1y6d1wsToY9aLFcmGixdhRlEBb0sQ64dC9iWeNdH4dcuhex\nrFkwjnXApXsRyxpG3l2YEovuuwZiuTDZYu0ozgAwpeMCYlnDlI4LiGUNUzouIJY1DJC6gFjW\nMKXjAmJZQ4vlwrRYDDh0YErHhZkWa0eRLoYpHRcQyxqmdFxALGsYeXcBsaxBLBcQyxqmdFxA\nLGuY0nFhTiwGHBqY0nFhtsXaUawLYYDUhXmxaLMqhKZ0VBfPLLaGGvtWodKNgxxaLGuU1alw\nL9EuhSkdaxDLBaZ0rEEsF5jSsQaxXGDk3RrEcgGxrEEsF5jSscZSrN0MryyDKR1rbFusvcS7\nDKZ0rEEsFxggtcZaLE6GX76l44B9i7WXiJdAi2UNYrnAlI41iOUCUzrWOIhFL2vBUWdKZ6LF\n2kvM/jDybg1iuYBY1iCWC0tr4JGo00M2i62AWC5418D7opLH986UDmIZ8a2Bd2HUTV3T7+ei\nJtusvVQyYrngWwPXfOzqVo6MpuokkcXWQCwXlk3pqEvnl9BZbA0nsQ4/krVMrL/yHMiUjvHN\nQ+N/KrzWw+3plSkdxOrjfaNf0jT2arrB2k0VI5YL/vHfap2SyfYKsY4JI+/WIJYL3fhP9490\nFvFSXOchljX6jaBKwq19VPGoP2N/OPiAQzf69O8q4dY+KthZrJ3E7Us/+tf9FNqtfVQwYrlh\niP6dZO3WzB0LC7OIEMRyYxj982xxx8KyLGIEsdzoRZ/es+bq9Ewzuy5CWUQKYrmhRf/KO++3\n8ntd4S5q9lHBHmId+sJQG8fKGqtHPQM4M0/jmUW8eLVY+wjdC20c6/KUziJeEMsNbRxLPot4\nQSw3tNDTYmI5uYU1bB+1i1hudEP/lLfCKJUEHXvfR+0ilhvd0M/lzXvpLdxQQz+LePET67jX\nhfokdP9F8CzixbPF2kfwHnQDT1TZuUoRawhiudEN/KbO+fM9Xufpe9iXZBEviOWGFnj9aKJw\n84SDLKIFsdzQA//Ln0x0DnhnwzCLWEEsN7jn3RJfsY56XYhYlni3WPsI3xnEsgSx3NDCvjfL\nTYhlES2I5UY37LvMarv7qFnEckMfIA18PTjMIl4Qyw3jlI5cFvHiL9Yxrwu7QV+UyB1Z+6jX\nBS3WPirAEf22mfPMI9sXZxEviOWGfiqk8z4KYrmBWJYglhv+QR9q6d78s4ZYLvgGfbCleyf9\nmRXriNeFesjPS14HF4tb3g+2dO8ysXZRBY4M78fK3rP4MsXBFsJELFe6IT/Uubgr+aGu8/vZ\nj6zuoVYRyxV9Sif9Vl8Am92PFguxJuk3PLZiHWzpXsRypRvyqWqx3tNr45Qca+lexHLF0Md6\n2t3lcKilexHLFS3kC9/SGQGxXBmOY6nLn2QWcYJYrjClY8VSsY439s6UjhWLW6w9VIITvgEz\npYNYk/jeNsMAKWJN4isWUzqINYkh4NfZ4rlrtFiOYh2t+24KN7WYhGZKx7XF2kM1OGAM1+bT\nxZQOYk1hCvdhtXgAUzqINYG5836XyiJWEMsVk1insN+030ONIpYrTOlYEUKsY10XMqVjQfnl\nr8UtVvwV4cDIAOnsIOmhpnRm/EEsA75iHWqAFLHc0YK9J/m6cq/E4ka/mSkde0NjALHc6QZ7\nr1qht8VaOrRYiDWJseHhWzo6ocTaQeNtTTfUpGmx+JaORrAWK/6qsKYbat4KZT/4lk4fxHJH\nC7VuhYKu0bSD2kQsd/RQi7V0Qi85Hn9thhPrOL0spnQsCNhixV8ZljClYwFiuaMHyoPXjCCW\nO8PO+5cHr/VBLHe6gfLgtREQyx19gJQHrxlBLHf6DQ8PXjMQUqyjDDh0w+TBa0bq2/wCtVix\nV4clhj4WUzo6s6IglgEtTB68ZgKxfBiOY/HgtR6hxTpGN8s7xvSatWzVrOK+hxvCt1ix14gN\n3RgvDnc1pEk5UVgmgliI1cN+nFPnlnfw00d5ezxiIVaf/nCDLUm54yc5fRALsYZ0Y0wv9kv3\n1i6l5zNiIdYQ/VRo/42ttnU7nRELsQb4itVOVH/UGbEc3z7AgIN3hLemcp4z9RR7JUq0WLHX\nyTz+Eb6bb7V+roiFWD3qCAUb59grEbF80MUS0Sv2SkQsHxBrhvamGcRyAbFmsBAFsQwg1gyI\n5QdizSAk1u5HshBrBqkWK+5amacVS+whfHFXIWL5gVgzIJYfK8QXdxUilh+INQNi+YFYM4iJ\ntfPrQsSaQa7Firte5kCsGRDLD8SaAbH8QKwZEMsPxJoBsfxArBkQyw/EmqR7N1ZwsXY94IBY\nk1iJ4t1ixVwzcyDWJIjlC2JNIizWjk+GiDWJdIsVc91Mg1iTIJYviDUJYvmCWJMgli+INQli\n+YJYkyCWL4g1CWL5gliTIJYv/pEdYoVVcbF2O0TqG9chVljVp6BlWqx4a2ca37gOscKqpSiI\nZcA3rkOsV4hY/vjGdYgVVhHLH1qsCVYRa6fd9wV9rP2vsLpOixVv/UzhHdURVlhFLH8WjGPt\nf4VVxPKHkfdR+qNYYmLtspeFWKNYi7K4xYq1hqZgSmcUxFoCUzqjINYSmNIZBbGWwADpKIi1\nBKZ0RllRrB1eF9JijbJmixVrHY3DlM4oiLUEpnRGQawlMKUzCmItgZH3EYYTOqJi7a77jlgj\nOIgSpMWKs5bGYUpnBMRaBlM6IyDWMpjSGQGxlsEAqRFT1x2xXBCa0hFb/HAlnIxALAO0WEZ+\nIFacn8BRmNIx8pMWK8aKGoUpHSOItRSmdIwg1lIYeTeCWEtBLCO/EWtP3XemdIz8qMWKsapG\nYEpniDIPjyKWC0zpDHE1ArEMMEA65Idi7aeXJTSlEyKLn/HLFiu6yhqDFmsIYgWAKZ0hiBUA\npnSG/FSsvfSymNIZ8tsWK7rqMsPIe5+izUCspSBWH3cjEMsAYvVBrCAgVp+fi7WP7jti6ZTT\nhL9tsaKqsDH8R96tvy8RVT35GIFYBnyDeCAWYk3hHcQ7mb5ZJkAWv2ATYu2hl+Ufwnt6IidE\nFj9gE2LFVWVmFoTw6MxDC2WxNvUdfoi1GK4Ku/gZgVgGEKvLZsSKv5eFWF02I1ZMlWYGsbog\nVjAQq6X9cg5iLQaxWnyNEBEr9l4WYrVsSqx4qs0MYrUgVkAQq6b79WfEWgxi1fgbgVgGEKtm\nc2LF3X1HrJrNiRVLxZlBrBrECgpi1WxQrJhPhohVoj8SayNixVF1ZhCrZIkRiGUAsUo2Kla8\nJ0PEyuk/G3IzYsVQeWYQK2fZoUcsA4iVs2GxYj0ZItbwRLgpsbZffWYQyzgwiVhLQSzEEgGx\nNi9WnN0sxDItQ7EpsTZfg0YQK9ChRywdxIpBrAhPhkcXy2m5+p+Jte06NHJ4scIdelGxomuz\nECsOsbZdiwYOLZbrwoSIZc+xxYqpxYrsZHhgsdyXoPipWJutRzNHFkvg0CNWDWIhlgiHFctn\npYAfixVVL+u4Yom0KaJibbQmzRxULL/HI/9crIjarKOKJXXoZcXaZF2aQay4xIqmzTqkWL4P\nG92AWBusTTMHFMv/CWubECuSNuuIYkkfevHUY+BwYi15+sdGxIqizTqeWGsceunUI1DrYGIt\ne0jDZsTaVJ2aOZpYKx560dQ332YdSqyl36XfkFgbqlUzBxJr+RcINyXWxhutw4gV4us4mxLr\nu+3z4RHEUirQze1bE2vLau1frIA3Xm1PrO2qtXexgt4fs0Wxyvb4l1VsZt9iBZ5t3qZYpVy/\nq2Qz/gV63S/Fh+Vye0llsZDgs83bFWt77ZZvYdKTajmLZLEANeivb+DQy6e+Jbl8C3JTyd+7\nePV5JuomkYUXQ6U2dejlUy+C34BfviVI1Lt5/VaJRBZ2qB7bP/TiqZdqdavkB6L55qgVdVhu\nPbCpP8IG8ZRCO8ie+zm0WHBEFvSxnp/i1WwfC46Id6t37rScpzRkkWAPLBjHuhXjWMnlPjOO\nBUfk99elsEsQC0RALBABsUAExAIREAtEQCwQAbFABMQCEX4h1o+m7MGaEAc5QBpbzXNv+UQV\nEGLFk09UASFWPPlEFRBixZNPVAEhVjz5RBUQYsWTT1QBIVY8+UQVEGLFk09UASFWPPlEFRBi\nxZNPVAEhVjz5RBUQYsWTT1QBcXcDiIBYIAJigQiIBSIgFoiAWCACYoEIiAUiIBaIgFggAmKB\nCIgFIiAWiIBYIAJigQiIBSKsK9ZDz+6WqOQW/hnx/WTDPeliIhOpYNaJpiDo0VlVrLdeH+Ua\nBKfQufSTfUscikHZhYJZJ5o65amMHVlTrHeiFf2lknf+XuD1BwbJvtUlbA6mTISCWSeaMuGw\nR2dFsR7qrBX9pp7Zv3/qHjabQbKP0DmYMhEKZp1oynTDHp0VxVI3ff25i8pXeQr+CRwk+1CP\nsDmYMhEKZp1ockIfnRXFeptXOQzdXRgke1HPa9YNlc1EKJh1oskJfXTWvSr8kVgF0wtXL81k\nRbHCR6PnNZKxa2KLi+OU2y/EUurv+01vQU8hvxNLIho9r5GMXRNbXByn3H4hVkkadCjgd2KV\nhI1Gz2syY/vEFhdnNofOsItWzCTssajzGUs26DEfZBI4mPGMKiQGsoIenR+KVV53fEJdSNX5\njCUb9FAMMgkczHhGFeJiLQ3oh6fCezFS8gy9oPQg2UTl8xJhj/kgE6Fg1ommIujR+aFYa428\n3/LKScsBP7FMVht5l4imItaR929b9PLnSebCuZNskU+aFG+EbUv6mUgFs040JUGPzi/FSov5\n8+CZdJJt8zkFvjw3ZiIQzDrRfDvpBwqI+7FABMQCERALREAsEAGxQATEAhEQC0RALBABsUAE\nxAIREAtEQCwQAbFABMQCERALREAsEAGxQATEAhEQC0RALBABsUAExAIREAtEQCwQAbFABMQC\nERALREAsEAGxQATEAhEQC0RALBABsVygtqzZSVUNHrhe8+xv12KfepnI5+q3gF8/p6fpTS/S\nq8wzI0Owc7FOqr+dh1hlItVCgcmCsnVSCyJWvvyJzFpgy9mdWBbvux7UcvuzuqUqPbs3Eb3s\nAj6hXRXPYt8miGWbeP6v+qbuTZakWMGSCs6Gi+ZCVsU3ldy/VWU/z0qdn9WZz7RxyTM7l5QP\nBlYqPRXP5L8lzcJ9j5NKHt8mkfzJ/U1a2WbnsrXItiqfYVwl0aRU716VqM6qSq3Jot75cynL\nr2HIpleuzR6/zRbMDaWK5dYe5RF7lHX+mBHrrppHphf736qFkK/FBpf6OedVIjd1etZpFZsl\naf2qeBp6lUSd0qXzh25WXbG6OxePb++ZZcqmV67NHr/NFsyNrKbTzKdTecQS9c5XnT3NnAqL\nFdr+it+K/fMVPsrlGFT+Mnsn61I9m+2vuXPFQg1/+d+uuSZ/1Q5/TRJNStruWlZ1Gfo7P3pL\nehmz6STMqVAeVSzNUTcFqlkQxKKPVR3twphLtYCMyl/mfqT5Wa3e/n3L1LoUm72+ZW+r3uHc\nJNGkNNi9yap+Odi5V1pjNp2EEUueTv+6OGupy/vdvm/cOOfzvJ87R7v7ozMq0THjeapPt9/h\nDvo7vd17WfX2aN80ltNcLsSSp3e07nmPJfnMiXXuH6JZscoVKD3E6meFWHEwOFrP22m2j3XN\nLrSen1GxBtsXtdUzbVwsbfdBVgvEMm2wPTZcNBcGR2vw22Dj6qcmltbHeurbl8MNRZfnPOj8\nXPpG9XYfZNXtY11GxDJm00kYseTRxTqV12BFi2UYm27Fen3fWh+rc1VYXIhlF2qXOpGrutQj\n74/8yuw2uCrsJq7t3svqU2022LknijGbTsKIJY8u1l/ZC3kVi+4NR8rr43FTzXb1W+e2A1O+\nzDtqZSLVQoFJs9lgHEtLvN09f6eTVZlaJ4t2574oxnGsNmHEkqd3KixG3vML+NdpQqx8ZOr8\nempDAvlY96v87ZEpcP20iXxu7d0N+XVnNSSeNIPneuLN7sU7bVZlalUW+s4DUQzZdBJGrNgY\nXVeU2rKGqupSDJCnl9E7GKgta45QVaplZstqSs/9nqtg2Jd140RefCscDtYj65ydfnlTJmIB\nTIFYIAJigQiIBSIgFoiAWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBYIAJigQiIBSIgFoiA\nWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBYIAJigQj/UVh3bxd/s2sAAAAASUVORK5CYII=",
      "text/plain": [
       "Plot with title \"Histogram of hist_Target$correlation_coef\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlgAAAJYCAMAAACJuGjuAAAAMFBMVEUAAABNTU1oaGh8fHyM\njIyampqnp6eysrK9vb3Hx8fQ0NDZ2dnh4eHp6enw8PD////QFLu4AAAACXBIWXMAABJ0AAAS\ndAHeZh94AAAazklEQVR4nO2diYKquhIAgws6Lvj/fzussgXoDjQKVr13jw5COgllCAmCewEY\n4D6dAdgniAUmIBaYgFhgAmKBCYgFJiAWmIBYYAJigQmIBSYgFpiAWGACYoEJiAUmIBaYgFhg\nAmKBCYgFJiAWmIBYYAJigQmIBSYgFpiAWGACYoEJiAUmIBaYgFhgAmKBCYgFJiAWmIBYYAJi\ngQmIBSYgFpiAWGACYoEJiAUmIBaYgFhgAmKtjvuJOjcppHvXXfnOeSvzbBG7xyVyro7UyYk4\nY67kGD/nZ+dwnZfGeL79ny1DuzLH+JxY92iVb+4l00ElljdjruY2JzvHIo14Thqv8XwbitWp\nzDE+J5blF6vBwbnHcFRxxhpiuXt4bq5VGo/pdccYy7dhxR7kGV9HrPGVTBEd+8a2aC1Mbmnd\nRuG5Sbf+y/8/sxvwoW+qIsSqLVZyyY4Ep7/XuwXIV7qdswa2OsI807+O18aWz0N+5Pg7pe8P\nRRcn+/B6cIe07bhG7thpQlrpecS6p7k4P5ufDmTMW6RDeTBM4shFVY8ridPFx6oIjQzUGkZZ\nClky+f+j0a0a5a4roBmwyk5dKXW+31nt1cMtK7ivyRnJiDesgDXFekZVD7i1/8puhzvlW9zL\nVeotD/kG1Vr5gahcIa3zuH9waqbX0yT9o9gkek5mzF+kW9HcVNvkoas/4l4GzlWf7JZ/GhVi\nFQxv1Sj3+00rYKfq0oV9sXoJFgX3HMzGMtIIO1A1ftYUK63ktE1I0oxfG5k8VZVTlCR6/1lt\n6bLN0s7JMXm94pYvqR7NOihopderizr181TGBoqUpHu6kc+olelbNwP3KnOnfN+kkc7vlIa3\nqstdv2kFLLLTrJQ632VWvQm+C95iLCONsANV48dKrJpyQfFv1kyUe6b8KP0mu2uSHoyKcqXd\njyh7ieots7rLDkDPVkrp0qwnfHjkL3XsTnqeQ2F0z/dFNJGxXpFab4t9mhRKpn9Ej1zMQy8D\nZcafRdrPYo/ljdjYVu9yv9+0ApbZ6VVKI6ueBLOKPXtKN5YRX1gRa4qVyfI+dFeZPGfNREac\nf5dO5aHjr97y1km6+PfeeqlX6KTnEStLL5nOWK9IrbdpPsv9fnpnOjlcnr0MXIojTFwuvRct\nQHpoG92qLnf1phWwk0efWAMJJp7SjWVkLOwoa4p1KRac3zVWvuQ5z7/MZRfX93G6wl98dK06\nbL+8t2umN3RWOJ2xXpFab+vyReMZSIq2MXqX4y/fKp7OdtL6oBWw3rZfKa9OAk/fp0Nl6203\nEHaaVc8K4yqPz95Hr3qH+bZM98ehYeq4WK13ErG8GRsqUnHAbH1zRjOQtwe3Rtcm7T5XSQiy\n3chwvwJ8ldIrt1qs5jtvWAmrivVK/opTjmPro/c3JPK2WPmf2ff8cL4+JGI105OJ5cvYUJEK\nRyL/zvBk4JaleiyONnm75VznOzSa7epNK2C51FspnhYr6n06ULb+dr6wItYVK+N2btXDabKP\nlX9ajR5NinWa7GMJMzZUpMKRU7Pvd2x2UroZyPbNrTxXSCNc3mKNbtUTqxWwXOqtlLqh9Cbo\nKd1YRnxhRawp1uHdXai+QsngWaHr7PnydbrFmj4rFGbMX6T7qVjprzi7/MsbudHzu/I4e6ky\nl/73nN6qJ1YrYFulRqUk76VDCXrcGMuIL6yINcXKDgrP93lSdnqUvb4H+d7f74pmUsd85Vs0\nKVY3PYlY/ox1i/Tm1srnvfnHtZ+BcoihUPU9SPQ3sVVPrHbAYmm7Uqp8l1sMJOhzYywjnrAi\nVj0UVn3kTP58RCV/c2yWI//GuNbIe764HJDPRkTv42J10nMCsQYy1tmu4j2SnpMrePcOXVfd\n9ayHXQ3hHhuBxrbqi9UKWCxtV0qVb1d/Fz0J+twYy4gnrIh1+1h5N+ZYXo10euf9HDXGkR7n\nfNqqK022ODo/ntWAuif1ilZ6IrEGMtberjAiro6S+fza6Vb9EdV/dAqUH9fff1wP9fVYI1v1\nxWoFLJe2KqXKd62iL0GvG2PZ94SVYCLWAiSzLiL4Lq7tssh3zpb5tkK6ovvxOLYnALfMI2r3\n2RDrE9Rdx3lXas7FtZid0Mzr+hZmscKNxbBJNpj3FSyzr96dx7JiXRbL1yL8oliv5JKdkkfn\nj7ZXC9Z9djHV32LZWoafFAv2AWKBCYgFJiAWmIBYYAJigQmIBSYgFpiAWGACYoEJiAUmIBaY\ngFhgAmKBCYgFJiAWmIBYYAJigQmIBSYgFpiAWGACYoEJiAUmIBaYgFhgAmKBCYgFJiAWmIBY\nYAJigQmIBSYgFpiAWGACYoEJiAUmIBaYgFhgAmKBCYgFJiAWmIBYYAJigQmIBSYgFpiAWGAC\nYoEJiAUmIBaYgFghmD0+cj9YPV/T/hGen8TxhZxihQra4T5wtFlTIFYIjjZrCsTSk7VWiDUB\nYulxtFjTIJYexBKAWHoQSwBi6SnF4rxwDMTS42izpkEsPYglALH0IJYAxNKDWAIQSw9iCUAs\nPYglALH0IJYAxNKDWAIQSw9iCUAsPW+xGHsfBrG0VBfN0GaNglha3AuxBCCWFsQSgVhaEEsE\nYmlBLBGIpQWxRCCWFsQSgVhaWmIxkjUEYmlxtFkSEEsLYolALC2IJQKxtCCWCMTSglgiEEsL\nYokIr5j75ZTf/eoU361CfCWIJSK0YpJD485qR5MQXwpiiQitmNhFf4/83fMWudgixJeCWCJC\nKyZyj/f7h4ssQnwpiCUitGJacxnjExs7q3vEEkGLpaUjFrOFfmb0sW7P/N2v97F2VrylCK6W\nY+Os8JCYhPhOEEvEjHGsOB/Hik6X3x7H2lnxloKRdy2IJQKxtCCWCKZ0tPTE4rzQB1M6Wvot\n1s4KuAxM6WhBLBEMkGpBLBFM6WhBLBG0WDra95pBrEGY0tHh02pXBVwKpnR0IJYQpnR0IJYQ\nRt51IJYQxNKBWEKY0tGBWEKY0tGBWEKY0tGBWEIYINWBWEKMpnRck8AQX4lXrJ2VcRFosXQM\ntFi7KuMiMKWjA7GEMKWjA7GEMKWjA7GEMPKuA7GEIJYOxBLClI4OxBLClI4OxBLClI4OxBLC\nAKkOxBLCr3R0DInFpE4HWiwdgy3Wrkq5AEzp6EAsIUzp6EAsIUzpaPD/XBWxPDDyrsGvFWJ5\nQCwNiCVmZn1cD86dbqYhvgnEEjNvHKvswY+eFO6pyhFLzCyxYhen54PP2F0tQnwhiCVmlliR\ny8cZEnewCPGFIJaYWWJV8xg/M6UzJhaTOi1miXWuxPqVKZ3RFmtH5VyAcLFOl+vN/aVvk/hn\npnQQS0y4WO8fozoX/cqUDmKJCa6Nx+N6PZ3yLnw86tWeKhyxxDDyrgGxxCCWBsQSg1gaEEsM\nYmlALDGIpQGxxCCWBsQSg1gaEEsMYmkYF4vZwgaIpWGixdpRSWeDWBoQSwxiaUAsMYilAbHE\nIJYGxBKDWBoQSwxiaUAsMYilAbHEIJYGxBKDWBoQSwxiiXFD95pBLA+IJcbRYilALDGIpQGx\nxCCWBsQSIxCLC2feIJYYSYu1l7LOB7HEIJYGxBKDWBoQSwxiaUAsMYilAbHEIJYGxBKDWBoQ\nSwxiaTCqCdfEJsTqIJYGWiwxiKUBscQglgbEEiMSazcH/rkglhhZi7WX0s4FscQglgbEEoNY\nGhBLDGJpQCwxiKUBscQglgbEEoNYGpr1cLg8rUNsGcTS0KwH55yFW3upasTS0KyH5O9s4dZe\nqhqxNHTr4X45LO3WXqoasTR46uERpe3W6OPDZ4fYJEKxmC3M6dfC7ZhfRHU0DLFNpC3WXso7\nj04tJJe0uTrcktSuk1GIzYJYGlq1cM867/Gj+GCx+tlLRSOWhtY4VtpYXavH8I4/mT40xJZB\nLA2tcazTzTrElkEsDa1xLPsQWwaxNLRqIYmz49/UU+lnhdgs+TACYolp1sIzynvszkWLjr3v\no6I9/iDWCM1aOLpz1lYl8XJDDd0Q2wWxdLQnobtvFg+xXRBLR7MWIld0rhLE6oNYOpq1ELvj\nPX25H11sFWK7IJaOVi0cy5stLDdP2AuxWRRiMQ396u71v1Om1YJXNvRDbBVNi7WPEs+Da96F\nIJYOxBKCWDoQSwhi6WjVQXZZsvheaffLKV/1FN8VITYLYulo1sFFcRO+5FCvPHEWuY9qRiwd\n7QFS+flg7KK/4orA5y0aH/faRzUjlg7vlI6AyD3e7x/jFwXuo5oRS0ezDk5Ofr2Mkxu5j2pG\nLB3ty2aOE/3wGlosxBql3fDIO+9pH+tWXLVFHwuxPISK9Z5XzDiMHkL3Uc2IpSO8Du5xPo4V\nnS6MY3UXMA29xl7fRyXrWqx9lHkO7Rq4nbLv2mnZ283so5IRS0f/eqx0mejHFEzpINYIzRq4\numN+VfLVnSe3Y0oHsUbpXvNe/gBscjumdBBrlO4AulQsBkgRa5RmDRzKFuvhDtPbMaWDWGN4\n+lg3yVUOtFiINUqrBk7yX+kwpYNYo/THsdzpT7IhUzqINQZTOiKy8T3E0sCUjohBf4Y++PnZ\nQsQSoRZrF6WeQ/BlM781pYNYWkLF+rEpHcTS4in//Si479qPTekglhZf+RPBJPSPDZAilhZv\n+QWHwokpHed0HbZvB7G0+Mp/FTw8gBYLsUbxd94vk9v92JQOYmnxiXWQ/NL+t6Z0EEsLUzoi\nEEsLI+8iEEvLwADpkidze6hixNISLhZTOog1Qqv8lyh7rtw9Elzox5TOlFi7GL8Lp1n6Szk2\n9RA8S4cpnckWaw/lDsc7gM6vdLoglpZm6aN3i8WvdDoglpZm6bPR9PSFX+n0QSwtrdJXo+mC\nZzQxpYNYo7RLnz9LR/bIcaZ0EGsMpnREIJYWpnREIJaWdum58doAiKWl33l/CW+8VnCNJq+y\n2UMFI5aWZuk1N157PU4uupaP32FKB7G6tAdI5TdeexTjEu6cvJ6n8XGvPVRwkFg/PV3YHUCX\ninXOxq7iYmQ0GR+p30PtBrZYeyh6IM2i62+8Vs5WM6WDWF08fSzRlE7h0l+x4t6ndJxDLC2t\noituvHbOelcFyXnvUzoT/iCWh/44luzGa0nUuMhm/GeI269dxNITXvS40imamLPefu0ilp5m\n0U+CqxpmhtgmiKVHfr3eIiG2CWLp6Q43GIfYJoilp1n05CR/dG9giG2CWHrah0KTOw9tv3YR\nSw9iCUAsPVzoJyBcrN+dhkYsATNarO0XPpB6+Nw8xHZBLD1tsUz02n7dIpYexBKAWHoQSwBi\n6UEsAYilB7EEIJYexBKAWHpqsUzuP9oMsV0QSw9iCUAsPYy8C0AsPYglYI5YvzpbiFgCZrVY\n2y9+EIglALH0IJYAxNKDWJNUv4NGLA2INcmkP4jlAbEmQawQEGsSxAoBsSZBrBAQa5K5Yv3m\nECliTTK7xdp6BQSBWJMgVgiINQlihWBUaLNrcD4AYoVAizUJYoWAWJMgVgiINQlihYBYkyBW\nCIg1CWKFgFiTIFYIiDUJYoWAWJPMF2vzQ3kBINYkC7RYW6+CABBrEsQKAbEmQawQEGsSxAoB\nsSaof6ODWBoQawKBP4jlAbEmQKwwEGsCxAoDsSZArDAQa4JlxPq9sXfEmmChFmvblRAAYk2A\nWGEg1gSIFQZiTYBYYSDWBIgVBmJNgFhhINYEiBUGYk2AWGEg1gRLifVrQ6SINcFiLda2q0EN\nYk2AWGEg1gSIFQZiTYBYYSDWBIgVRnhx75dTflu1U3y3CvENIFYYocVNDo1b9h1NQnwDzZ9S\nzBXrtwYcQgsbu+jvkb973iIXW4T4BkT+SFusLVeEmtDCRu7xfv9wkUWIbwCxQgktbKtdH2/k\nt1yfiBUKLdYoiBXKjD7W7Zm/o4+FWB6CC3tsnBUeEpMQXwBihTJjHCvOx7Gi02XH41iIFQoj\n76MgViiINQpihcKUzijLivVLY+9M6YyycIu15apQwpTOKIgVCgOkoyBWKEzpjIJYodBijYJY\noTClMwpihcKUzgjty/wQSwNTOiMI/VGI9TsjWYy8j7C8WBuuDCWINQJihcOUzgiIFQ5TOiMg\nVjhM6YyAWOEwQDqChVi/cl5oNKXjmgSG+DwmLdZ2q0MFLdYIiBUOUzojIFY4TOmMgFjhMKUz\nAmKFw8j7CDZibfl0Rg5iDdK9tmG5FmurNaKBKZ1BxKIglgemdAZBrDkwpTMIYs2BAdJBEGsO\nRlM6S4T4NIZi/cB5IS3WIJYt1lbrRA5TOoMg1hyY0hkEsebAlM4giDUHRt4HQaw5INYA/Qmd\nRcXa/XkhYg2gECWoxdpmrchBrAEQax6INQBizSN85F38e4ltVqG5WDvvZYWW7opYc1usbdaL\nlODSPaLxi2UWCPFREGse4aV7jE/kLBHikyDWPGaU7tqYhzYK8UFWEGvXvSzOCr34hkeXb7G2\nWDNSEMuLShTE8oBYXlYSa8cHQ8TyslaLtcW6kYFYXhBrLojlBbHmglheVhNrt70sxOrj/IMN\nNi3W5mpHCGL10YqCWB4Qq8+6Yu30YIhYfVZusTZXPyIQq8/aYu2yzUKsPqu3WJurIQGI1SVv\nQBBrLojVRS/KfLF2eDBErC6fEGtrdSQAsbog1iIgVpfPiLW7gyFitSlmcz7QYm2qlgQgVpsQ\nI5YRa2eNFmK1+ZxYr30dDxGrSXVZw4fE2lJVTYFYTcKMWFCs/RwQEavJx8V67abVQqya+vq+\nj4q1jzYLsWpCjVi6xdrFARGxKprXI39YrNcO2i3Eqgg3wkSsrasVnvl9PVau/fuJrxBr24fE\n0Izv67Fy3Z/lfIlYlVxb9Cs0x3t6rJz4UaqfEOtVD3BtSq/QvO7gIU2NG11+v1jF/6R35/wC\nQvM38Vi50Spw8OUEStHayYHbKVos+EVm9LGkj5WDXyS41ZM/Vg5+kRnjWNLHysEv8u0nF7BR\nEAtMQCwwAbHABMQCExALTEAsMAGxwATEAhM+IdaHpuxBzBI7eYE0vjXm3uJsqkCItZ04myoQ\nYm0nzqYKhFjbibOpAiHWduJsqkCItZ04myoQYm0nzqYKhFjbibOpAiHWduJsqkCItZ04myoQ\nYm0nzqYKhFjbibOpAnF1A5iAWGACYoEJiAUmIBaYgFhgAmKBCYgFJiAWmIBYYAJigQmIBSYg\nFpiAWGACYoEJiAUmrCvWtR0ujlwUL3+P+G6yy93pYiSIVWHWKU3OontnVbEe7foonkFwWDpK\nN9mHxa7o5d2oMOuUpkp5LLCSNcV6RK2s3130yJYt/PyBXrIPd1o2gi+IUWHWKU2R8LJ7Z0Wx\nru7Yynrsbum/f+6ybJhestelI/iCGBVmndIU6S67d1YUy8Xt58+dXPaUp8W/gb1kr+66bARf\nEKPCrFOajKX3zopiPfxPOVy6u9BL9uRu57QbahvEqDDrlCZj6b2z7lnhh8TKGX9w9dwgK4q1\nfGnasQYCaxObnR1VtE+I5dzf65XEix5CPieWRWnasQYCaxObnR1VtE+IVZAsOhTwObEKli1N\nO9ZoYHlis7MzGaEx7NLKZrTsvqjiDCW76D7vBVm4MMOBSiwGshbdOx8UqzjveC51IlXFGUp2\n0V3RC7JwYYYDlZiLNbdAHzwUXvKRktvSD5TuJRu5bF5i2X3eC2JUmHVKU7Lo3vmgWGuNvMdZ\n5STFgJ9ZkNVG3i1KU7LVkfdXnfXi9WBz4txINo+TRPmCZduSbhCrwqxTmoJF984nxUry+fPF\ngzSSreMcFj499wYxKMw6pXk10l+oQFyPBSYgFpiAWGACYoEJiAUmIBaYgFhgAmKBCYgFJiAW\nmIBYYAJigQmIBSYgFpiAWGACYoEJiAUmIBaYgFhgAmKBCYgFJiAWmIBYYAJigQmIBSYgFpiA\nWGACYoEJiAUmIBaYgFhgAmKBCYilgMqSs5e66t1xvaJ3r87i7soH9ZP4nuew5/d172988y0M\nIjnb3DJyEfYu1qFXwPJ5fy56qgKUzwmMZuSskaFFxMqefmLzKLAF2J9YU8uLJc+j8r6tRxcn\nLjnqm4hODha8Qbtzuu/GqvyqWFnTobqjdbaZeyX6JstSrMWSWp5vzpuGtJJjF11eZXXfjs4d\nb+Vxr7dmzs2ds5frwUXlPYjjyB2LNiBdmN2YuHpiTfaa3bj/nVJvzSzZ5OBO1Usj3SI/6WEr\nvwVxmaEiE42Nn6ci9y08YeqEzR4NvQzfmzMdzuXPW7sW++xa1Pp1TKzCmlN9N/P8IchRUr3L\nFp6K++dnzxmN3eFWpeRZs8hAXL000s3iXVx1c/amWM2N87u3d8zyhakTRqxVSOs6SX06FPss\nco9Mh8PYoTB/c8s2S3tOt2z19O052/l/5UMZ/qpW7Zx1ZtJTMHcuPeuvWWTg/dJIt/DoL1vZ\nvcNn/3Y3vnae6OUN00iYQ+EauLxtqRoD9+4/TYh1yp9Mk2QHr7x1yvtQp/IxMsesI5bbku/x\nR5yqlR3lvGsWGaheGul2zldrsXobd/LqDdNIGLHWoN5f2X+pAqfHo17eX/O9snPNo1NzjeKY\neskeK1Meo9ztUB1se2u2n+/XS/d5uxw7YvU2Huzmt9Z8J4xYa9DZX5eoHKkaFutZd1SGxcob\njEt1Wu+KjlmAWMeODYi1FXr76xYfxvtYf0Vfure87Uj2mK1D1fdxrQivUbFa6Z7Tc7rbcxmx\nfCt8Id+cNw29/dX7q71m1n26V72XnGOvS5P1Yx7u+MiPhMVwQ/65d822UY1037noitXbuJNX\nb5hGwoi1Bm2xDsVZWN5idUenyz5PMfKen2+lPalT9k96vhV3TsKypKI8ibM7VSPv3jXbYjXS\nLfJ0fz2qPtazXK23cUcUb5hGwoi1Bm2x/op+yD1/6l7UXbM5V3is33uHjdJ2ojgnLJ8TGL0G\n1uwcA+t0y5OJdoYaHa96464o3gw1MoxYK9A5FOYj79kp/P3gF+tYDUZe0z19Llq17FyyHOiu\nHwmYVOOWz7i+usGzZrdz9U43X3LOspMfy4oMFat1Nu6J4s1QnWHE2jK3xsGUypJDXU1wdPXT\nTKksOT9RV65Gv+Xyj3kejxeY029j6/kXEb67ovz0az0QC2AUxAITEAtMQCwwAbHABMQCExAL\nTEAsMAGxwATEAhMQC0xALDABscAExAITEAtMQCwwAbHABMQCExALTEAsMAGxwATEAhMQC0xA\nLDABscAExAITEAtM+AeEynTRnW148gAAAABJRU5ErkJggg==",
      "text/plain": [
       "Plot with title \"Histogram of hist_Decoy$correlation_coef\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p1 <- hist(hist_Target$correlation_coef,breaks = seq(-1,1, length.out = 201))\n",
    "p2 <- hist(hist_Decoy$correlation_coef,breaks=seq(-1,1, length.out = 201))\n",
    "\n",
    "target_histogram <- data.frame(\"Breaks\"=(p1$breaks[-1]),\"Counts\"=(p1$counts))\n",
    "decoy_histogram <- data.frame(\"Breaks\"=(p2$breaks[-1]),\"Counts\"=(p2$counts))\n",
    "\n",
    "target_histogram$decoy <- 0\n",
    "decoy_histogram$decoy <- 1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca8aec25",
   "metadata": {},
   "source": [
    "Run the below code as it is with commented lines to visualise the plot. **To save the plot as pdf or png**, rerun the same below code after uncommenting ('Remove #') the respective line:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "49e2c7b4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>null device:</strong> 1"
      ],
      "text/latex": [
       "\\textbf{null device:} 1"
      ],
      "text/markdown": [
       "**null device:** 1"
      ],
      "text/plain": [
       "null device \n",
       "          1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAlgAAAJYCAMAAACJuGjuAAAAPFBMVEUAAAAAAEBNTU1oaGh8\nfHyMjIyampqnp6eysrK9vb2/j8+/v//Hx8fQ0NDZ2dnh4eHp6enw8PD/v7////+oISYOAAAA\nCXBIWXMAABJ0AAASdAHeZh94AAAcZ0lEQVR4nO2di7qqKBhA2ZbdTtfp/d91VDQR8JLyJ+pa\n882uTPwR1kEEU/UGEEDNnQFYJ4gFIiAWiIBYIAJigQiIBSIgFoiAWCACYoEIiAUiIBaIgFgg\nAmKBCIgFIiAWiIBYIAJigQiIBSIgFoiAWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBYIAJi\ngQiIBSIgFoiAWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBYIAJigQiLF+t+2CmVpNcxaZXy\n7f6h47sQ2TglSh3qV1+gQcEP3+Tv5yxdrFSVJPfvE/vq756o1u+CZOOUr3aoX0eKVeUzVuLO\nXS97VfP4OnVXnX4l1jfZ2JWrVK8jxfqyRf05ceeuj6yhSM6v9/t5zso5/Tr56KPQlGyMMrd9\nM7ESd+56eGQV+izf7k763e2QH2Fuemn29rlTR+PN+3VMVHJ8Vl8Xr9f8SLbLF5atjvGds8H3\nLWufDma71JsNb1D79XXM2rD9zVrWTPcJXefzdcqby3F9TEkWLdZRqbO1qDoo6XYj9yWrLePN\nM9Hf38uvzTTZQlcsZ4NH54jXmw0jaBXACVStc3y7y6rM1qE/yatV8n2LikWLlVXfs7nk04nW\nVarfXo03VT0k5dfZ3+z4tX8VeqRufXs3mHP4IhtG0FaxqnXUzV2WOKE/yQ/FXr32rtszs2ix\nlN3PuGVLss7O61RXUO5M/UZL9NLVUabflV406rl89WwwuRXVaUTuy4YvqP2arZM8CkF276bx\nRjozdP2SZ/6l00XEusQ6VP9wj7pJKXWo32RNSSla6qT3idWywVe3WFYqf9Dma1puWPfRPsua\n6czQRqP26crFxLrEUmVVvJ8fS16NL+rjSWKmf16Pe+UTy7NBN3JfNvxBnUDOjnWmK19O5dEx\nOrcWLdbOHjWqq8eqLqNWKuql1527zKnvDrH6suEN+pVYnnTVx2PlntXNm5tFi+WcjplNRaNJ\nqt4knvq7Zi+7w/nR12L5mptB2fAFHSJWZ7pPgtdVn4JGdlq4aLHu9T/Ue9E5Sd0ukf66epNW\nnS5j6a5c5hOrbYMND/qy4Qtqv+79faz2dI0c3JpnEzEQWXa+5DPknXc1Uu9JnF6xenPVs3lX\n/Q+8oZLRYr0+S9s22KzHnmz4gg45K+xMV+Zz9+k8JhIFPJ5li/X6jP6UjUY9aVeMMzli1cNF\n97fRWBwzGZLP8acYpCxTtGywKVZfNjxBndfPOmfPMk+6Mp+Zw/tn0Yc/hi3aqSxbrOKfuGan\nD0bVZ+val8+bW/m9McB9r5QoKjA/qJgNhH+DTbH6suEJ6rzePSPvXemqfFad98i6WEsXK+9e\nJHnf+9b4bE7tNd/oKbm0MSX3yCopOTyeerwotbpn3g1aYvVkwxfUec3nBe11utKV+dT9q31k\n4+4rEAviBLFABMQCERALREAsEAGxQATEAhEQC0RALBABsUAExAIREAtEQCwQAbFABMQCERAL\nREAsEAGxQATEAhEQC0RALBABsUAExAIREAtEQCwQAbFABMQCERALREAsEAGxQATEAhEQC0RA\nLBABsUAExAIREAtEQCwQAbFABMQCERALREAsEAGxQATEAhEQC0RALBABsUAExAqPWg1TCiFY\ncULFasoUseJiNWWKWHGxmjJFrLhYTZkiVlyspkwRKy5WU6aIJcTIE+7VlGmEYoUaDZkX9R9i\nzZE2nhBCZGKN+XfRkWJZ/9gQS4hMrP9GNFp1Cq9HwzYYQ6khlgiZD5PF+qdBLAliKKIx5FpJ\nilW2ZOqtX7K/qlxcfzkviCWCsFiq/KOqN6Vgxv9zg1giyIr1+avcT/WbeUEsESqxvj4mDe5j\nFZtGrIhDiFCJ9XWbNeyssLLKEqtaP4ZSQywRAojV8XXnobB/Iz8BsUSYUSxarEhCiPADsZSp\nEmeF8YUQQVYsPVKlzI/mONbYCfCgIJYIImJ1ThV2JHRTCRfr1FYTsVoROStU9irG5y6v3FSI\ntWWx/tM4YrlmdR76fKkQC7E8YjX6U/UbpeoKVS0Wqs8kUP2qh74+c491QmPDxpfNcE2tm2v2\ntqX9hSDG8sX6sh89WKxGxRn/2wc+5xBorqeq00llfijXtaM01nTzYK1pTQyMLAQxFimW0hfN\naLG+bLOGiqWab6yWyojpHgGV1WK9fR+sZf41zfhmrMpAxApLZdQsYtkdnMliFS/KuyZi/Zaf\niGXPC8qJ5U5MmvH9fSzEkiCQWG3DDc2GwhHrrTwyWeuEOhS2BEEsGcKI1f5d3Sb5WixLLHOF\n4GI1gzfiIFZ4hMUyFTLeGAt8zYm9rlv5XWI5a9ohnSTKeP0WxPIjJlbj0NgyjvVu1vUnXWNd\nVb4dIlZzYtL40zWO9bYuxv8OxPJjifXVUFaIHY6i0KZkArH8WGJ9dek7Yk1MG0+I8MwrVgSX\nzOQgVnjmbrGiALHCg1hvxJJgklirYUoBTkgbT4jwTBEL3lNq/X5KC6vT410qxIwg1kTGltdr\nZ7SYe5EQs4JYExlbXkeVXB/Fu+ctUUeJELOCWBMZW16JenzeP1QiEWJWEGsiY8urccbQffqw\nyCpBrInQYvlxxYpkNHwpTOhj3Z7Fu630sWizvmJ0ae2Ns8LdSyTEnCDWRCaMYx2LcawkPW1j\nHAuxvoKRdz+INRHE8oNYE2FKx49PLM4Lv4ApHT8+sWizvoApHT+INREGSP0g1kSY0vGDWBOh\nxfKR9RsRaxpM6fhwjEKsb2FKxwdiTYYpHR+INRlG3n0g1mQQywdiTYYpHR+INRmmdHwg1mSY\n0vGBWJNhgNQHYk1GaEon1B0AZqJNrEXuzDzQYvloE4tfgQ2GKR0fiDUZpnR8INZkmNLxgViT\nYeTdB2JNBrF8INZkmNLxgViTYUrHB2JNhikdH4g1GQZIfSDWZPiVjo8OsZjUGQYtlo8OsWiz\nhsGUjg/EmgxTOj4QazJM6bjkP1dFrIkw8u7SZhRifQFiuSBWACYW03mnVHoTDfF7ECsA08ax\nyh5850khYm2SSWId1TE7H3we1VkixGwgVgAmiZWoYpzhpXYSIWYDsQIwSaxqemNlUzo9YjGp\nM4RJYh0qsdY1pdMjFm3WEMaLlZ7ON3XN3r6OK5vSQawAjBfr82NUpZJ1TekgVgBGF9LjcT6n\nadGFP3Z6hVibhJF3F8QKAGK5IFYAEMsFsQKAWC6IFQDEckGsACCWC2IFALFcECsAiOXSKxbT\nhf0glkuvWPxutR/EckGsACCWC2IFALFcECsAiOWCWAFALBfECgBiuSBWABDLBbECgFguiBUA\nxHJBrAAgloXS95pBrIkglkWfUYg1DMSyMOTpuk3W3NmMHsSyMOS5INZ4EMtioFhcONMDYlk0\nxFIdXa25Mxo5iGXREOuCWGNBLAvECgNiWSBWGBDLArHCgFgWiBUGxLJArDAglgVihUGofJSJ\nTAghLLHap6Tnzmjk0GJZVPJk5GJdLog1CsSyqOS5XBBrCohlMVishR3ifw1iWQwWizarE8Sy\nQKwwIJaFK1br9X5zZzVqEMvCFav1sqy5sxo1iGWBWGFALAvECgNiWSBWGMzS2Z2e0iHiB7HC\nYJZOdv4j4dayKgCxwmCWzut6kHBrWRWAWGGwS+d+2oV2a1kVgFhh8JTOI8narc7Hh08OETFf\niMV0YQdu0dz2xUVUe8EQMeMTq/3nhXPnNl6sonmdsuZqd3tldqVCISLHJ1b7haRz5zZeGkVz\nzzvvx4f+IlihLav0ESsMjXGsrLE6V4/h7X4y/dgQ8YNYYWiMY6U36RDxg1hhaIxjyYeIH8QK\nQ6NoXsf8+Nf3VPpJISKnPgFErGmYRfNMih67UknQsfcllX5ljUKsiZhFs1eHvK16HcMNNdgh\nYqey5oJYE2lOQttvgoeInTax2n63Ond+48UsmkTpztULsWyx2n5TMXd+48UsmqPa37OX+14d\npULEDmKFolE0+/JmC+HmCZ0QkfOtWExDt9EsmGuaaxXwygY3RNx0iNXyK7C5cxwrXPPeoEOs\nlqtn5s5xrCBWA8QKBWI1QKxQNAomvyx58L3S7qe0WDU93r8IETmIFQqzYE5f3ITvtatX7jmL\nXFLZd4rlvZB07hzHSnOAdPj54FElV31F4POWdI97LansO8XyzuzMneNY8U7pDCBRj8/7R/dF\ngUsqe8QKhVkwqRp+vYwabuSSyh6xQtG8bGbf0w+vocVCrE6sn9gP7rxnfaybvmqLPhZ4GCvW\nZ14xZ9d5CF1S2SNWKMYXzP1YjGMl6Wkr41jey7KYhvbDyHuD6pqFFrG8FznMnec4aRbLLc3/\nAaZhbzezpJLXuniNar3IYe48x4l7PVb+W9UhZq14SqdLLHdmZ+48x4lZLGe1L65KPqtDb7pV\nT+kg1nTsa97LH4D1plv1lA5iTcceQB8q1qoHSLvFsuei585znDRvCqJbrIfa9adb85ROt1j2\ncNbceY4TTx/rNuQqhw23WIg1hEaxpMN/pbPqKR3Emo47jqXS65CEa57SQazpMKVjUI1+9onV\nnNmZO9dxwpSOQaVLn1jNmR1mC30glsE4sWizfIy+bGaNUzqIFY6xYq1ySgexwuEplPt+wH3X\nVjml84VYCrG68RXKa8Ak9CoHSL8Qy5zZWc4O/hBvoQw4FPZM6Sj1zXE1FsqTvGFiXRCrC1+h\nnAc8PGC9LdYAoxCrH3/n/dSbbpVTOogVDp9YuyG/tF/jlA5ihYMpHQPECgcj7wZfilVNGS5n\nB39IywBpyJO55ZT7l2JV46TL2cEfMl6stU7pfCWWQqwWmnf0S/Lnyt2TARf6rXdK5yuxLtXQ\n19xZj47mHf302NRjwLN01julM0IsniTg4h1A3+qvdMaIpRDLS/N3hVWLtdFf6YwR64JYXswS\nyUfTs5ft/koHscLh3rshY8AzmpjSQaxOmiVSPEtn2CPHmdJBrC6Y0jEYJ5ZCLA9M6RiME+uC\nWB648ZoBYoVj/I3XNOek9yqb5RQ6YoVj7I3X3o9UJefy8TsrmtJRiBWGsTdee+hxCXV4vZ9p\n97jXcgpdtd4uuVusZV3Z/xvG3njtkI9dHfXI6Kt7pH45RT5WrOL6mbkzHxnTbrxWzlavaEoH\nsQIx9sZr2qWrXnEVUzr6glDECsTYG68d8t6V5nVYxZRO5yMpesWij2Ux9sZrr8S4yKb7Z4gL\nKfJpYl0Wspc/Y3x5HCudkp4564UUOWIFxSyPdMBVDRNDRAxiBWX49XpBQkTMVLHoZTWwhxuE\nQ0TMVLH+LWQ/f4RZGq90+KN7R4aIGMQKytg7+o0METGIFRTEqpguFt0sAy70q5gq1p/6R6NV\ng1gVU8W6IJbJF79RnRoichArKE2xRPRaSGkjVlAQqyKIWHTfKxCrIohYtFkViFWBWEFBrIpc\nLIVYoUCsisGPpOgWi16WphZL5P6jZojICSQWbZYGsSoQKyiMvFcgVlAQq2K6WH9/iPUBsSqm\ni3W5INYHxKpArKAgVkUwsRhxyEEsjb4vXxCxmIvOQSzNF8/sRawhIJYGsQKDWJqwYtHLQqyS\nsGLRZiFWSSCx/hCrBLE0gcS6VGJt/mCIWBr91MFwYm2+zUIszaj7cCNWO0L7L3YNjhSIFRha\nLA1iBQaxNOHFWkhTLQViacKLtfEBeMTSiIi15TYLsTQiYm25zUIsDWIFBrE0iBUYxNIEE+sP\nsQoQSxNMrAtiFSCWZtQTMPvF2u55IWJpRhnVL9Z22yzE0oiJtdVGC7FyRt6/aIhYWx2AR6yc\nkXdcQ6x2ECsHsYKDWDmIFRzEypEVa5Pdd8TKCSpWeTejjY+TIlZOULEuF8RCLA1iBQexcoKO\nYyFWDmLljDQKsdpBrBxpsTZ4XohYOdJibbDNQqwcxAoOYuUgVnAQKye0WH+OWJvrZiFWTmix\n7N9UbHDKELFyECs4iJWDWMFBrBzECg5iVRM64mJtq/s+fmfvp7S4rVp6vEuF+BFZvf9CrG21\nWWN39rUzbtm3FwnxM9RfeLH+fGJtqc0au6tHlVwfxbvnLVFHiRA/o12P8WJ5LiTdVps1dlcT\n9fi8f6hEIsTPQCwBxu5qo1XvbuKjL03EEoAWC7FEmNDHuj2Ld/SxEMvD6F3dG2eFu5dIiF/x\nQ7G2c2I4YRzrWIxjJelp0eNYnyed/ESs7QzAb37kPa9rxAoPYiGWCJuf0hETy/OD6C1NGW5+\nSkdMrIvvCuXtTBlufkoHsWTY+ABp1tz+XqxNHAw3PqWj6/rHYm2izdp6i4VYQmx8SgexpNj4\nlE5WyX9/vxdrA92sjU/pZJXcq8d4sdzfrW5nnHTjI++yYrVc+o5YiwkxknywYS6x1n4w3PSU\nTlHJM4m19jZr01M6iCXHpqd0EEuO7Q6QqqKHJSyW9+eFiNWVrntKR5mMDCFMVcmiYrVflrX2\n7vuGW6y5xVp3m7XdKR3EEmW7UzqIJcp2p3QiEGvN3aytjrzrK/x+IFbbpe//Vj6zs1mxytr9\nkxbr0nGRw5pndrY6pVPV7nA9ZMRabZu11SkdxBJmq1M68Yi10oPhVgdI4xFrpW3WNn+lo6cJ\nfyVW14WkiGWx7BarrtZfiNV9kQNiNVjwlE51WUM0Yq2yl7XBKZ1GtcYg1irbrA1O6UQo1gob\nrQ2OvEco1grndjYnVj1J+O+ffiTFD8TqupAUsSIO8QWNav1Oj/FidV/k8G+Nk4aIFYlYa2uz\nEAuxRBg/8j749xIxlVg95P5jsTovy1rjpOHYnTkvVKx/M4l16ZoyXOMQ/OideSTdF8sECBGa\nxpA7Yskyfmce3RM5IUIExq1WxBJjws6cjXlooRBhmVms9osc1vjjii2dFVr1+ff3W7HaB+DX\n+OOK7YjVGHLP/xurh6hYq2mzNiSWXZ+/F6ttZmeN18BvRSxrAGsesdrGSdd4DfxmxHLrM1Kx\nVtJmIVZ8Yq2i0dqEWO7I6Fxi+Wd2XN/mLrHpbEMsT0PxN4tYF+84KWLFGqI7vj3OYF06GqNY\nyz8YbkEsT8XNKJZnAN6Xv5kLbTKrF8szzjCvWJ5xUl/+lt5mrV+sloqbUSxnnNR/srhst9Yt\nlv90cG6xnFGHllGIRR8PVy5WW439hdBDXqwFt1lrFst/OvjPvl5mBrGs4azWbC64zVq1WO01\nNrNY1qgDYsUawhu25XRwWWIt9mC4YrG6amx+sf6GibXYk8OVitVxOuhcOjqPWJehYi20D79K\nsTp67fl/QfUYn/JvsFhL7GqtUKzOzlVEYtUnh/1iLa/NWqNY3XXkXNYwn1hVH75frOV1tdYm\nVmfnyj/kPqtYf723vl3oOPy6xOrpXEUoVv7fULEW1WatSazezpV/Lmd2sXpuy7bMrtZqxOo/\nBv5rGcCaXayiFz/wiLgYt1YhVnHDmwG14h/AikCs4s3QhmsZai1dLH0XpUHHkbZxhljE6n22\noXFEjN+uCbfjjuCxckZD1V8dbeMM8YhVnSUObLiilmts3mZ/rNynqRooVnuvPSqxLqrscQ3y\na8Bt7+ZibKbmeayceRNBq6THd66iE2ucX+FKOQhj8/OrhzRZ9zptL2L/wa+sHFk9hMRq+tWp\nWXVoHH7/TnHG39y27UO5BBbMSCkaBoxM90WLBVtkQh9r6GPlYIuMbvWGP1YOtsiEcayhj5WD\nLRLbWSqsBMQCERALREAsEAGxQATEAhEQC0RALBABsUCExYo107z/NghRPwG2MQuz3GqEmD/d\nxiwstcC3EhOxiCkSE7GIKRITsYgpEhOxiCkSE7GIKRITsYgpEhOxiCkSE7GIKRITsYgpEhOx\niCkSc7FiQdwgFoiAWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBYIAJigQiIBSIgFoiAWCAC\nYoEIixPrmKjkaNxXPtxtLAYGdBfIxxTfyYJzM8C0/VyaWPq5BbvP54d0mdsB3QXyMcV3sorS\nlYkvWZhYd5U83o9EfZ5Z8FDpbwM6C34QU3ondZCkIdbU/VyYWEd1y/5e1alacK7f/iags+AH\nMaV3UsfYN8Saup8LEytV+ZOhjH/BZ3X+bUBnwQ9iSu9kjjo2nw44dT8XJla573URpOp2yPqY\nvwvoLPhBTOmdzHn4n0E5ej+XL1ZB91OpQwacRyzZnWzGbcnEtxubnJ2f4qnn6/v9OoodK6IQ\nS3onm3FbMvHtxiZn56e07O5L7PQ/CrE0cjvZjNuZieEbm5ydn1CN4yQtuytWz07AthxIxiyR\nHshqbH/qfi5MLH2u8nTOVcTK3AnYlgPJmCU/FWvqfi5ErIpTMbpyqx9Cnah80kGunp2AzoIf\nxJTeyZKGWFP3c2FiOePBx3zPX3o07ycB5xh5l97Jki2PvL939Yl3UQ6vpFgg137YAc0Fv4op\nvpOaSqwg+7k0sV7FnHvxVu9/vmAneB7uDSg8WPnznXwbsQLt59LEgoWAWCACYoEIiAUiIBaI\ngFggAmKBCIgFIiAWiIBYIAJigQiIBSIgFoiAWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiIBY\nIAJigQiIBSIgFoiAWCACYoEIiAUiIBaIgFggAmKBCIgFIiAWiLAWsR6HRB2G36PTvgHxzbfw\nW16H/HaOxd/mpto3LHxX0RlZiVhHfb/u3XPg+lZV75Rn4dfkTyY56b8DxdqtpPg9rGPPTirJ\n/u2/speBZllVHeYO6qq4Nbr+Oyoba2IVe/ashDqow7AUQmJ9uy3Eiptj9bjGV1rcWvi8K+8x\nrNRrp9LqpfgiKb/I/tyyo1ZxX+DysRe6mo3Ez1Ql5oMgj4naa4M/K9XbVAblpqr1PxtOmhtu\nPo/3tldqf2sL5O6DmSA+ViHWXj2aH6s7lCuVquIJj8VL+Xg2/UV+/Kzunm6KZSYubq9+am43\neTVWqrfpivVZv/iUuhtuiHXWSc8tgdx9MBJEyCrEah5RruUzFa75F/vX+/Nyy19e+/wBD9qj\na76yepsHMTvxuX7k1jX/fMgr11jJ2KZ5KNSbqtbPPzWCVxs2M57k/zqu+VJvIHcf6gQxskKx\n0vIpMEUrctff3/UXedW88gOK9Zi4Wgknsfk4ynueOmmsZGzTFqteX38yglcbNjOuPk808Qby\n7UOsh8GcFYplPGmv+dC9+khVLnneTntLLCex86RCO8Jnm7ZYzfWd4LZYx+xg93h0BLL3oU4Q\nI6sQK/30sW6vL8Tae5SYT6z3Ke955ae3A8WqE8TIKsQ6VT3su9Fx8Yn1SZG/PWTnW7fnZLHe\n7+biVrE861nDDbfjrtnzat2VZoIYWYVYn3GsfX6OVHVMUrs20rpT8qkqWywncV2Te6frkza2\naYu1t/pYN3c95RR/M6G7K6ndsXI3EQeRZutLDsXIez469LZOpYqvy5fii+x0LK1q9f5+VH2s\narjJSVxX3Dk/ITtaJ2vGNm2x6vU/G66DV1moD2Q7vbldSyB3H+oEMbIOsarukp4rNAd/im8r\nOfQXVTemnF/Mm4edqhoWJ7HRInjHsept2mJZ41jN4Pqvjqu5fnLTMo5l74ORIEJWItb7mmbl\nfy0/nJPPcPXbeClGrdXh48AhS3IvjjP33UcsO7F94vZsRjC26Yj1Wb/ccCN48VfHLSkG0u9t\ngdx9MBPEx1rEgshALBABsUAExAIREAtEQCwQAbFABMQCERALREAsEAGxQATEAhEQC0RALBAB\nsUAExAIREAtEQCwQAbFABMQCERALREAsEAGxQATEAhEQC0RALBABsUAExAIREAtEQCwQAbFA\nhP8B2r49jXBSWX4AAAAASUVORK5CYII=",
      "text/plain": [
       "Plot with title \"Correlation Coefficients\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#pdf('Combined_histplot.pdf') #saving the plot as pdf\n",
    "#png('Combined_histplot.png') #saving the plot as png\n",
    "plot(p2, col=rgb(1,0,0,1/4),xlim=c(-0.8,1),main = \"Correlation Coefficients\", xlab=\"Correlation coefficient scores\")  \n",
    "plot( p1, col=rgb(0,0,1,1/4), \n",
    "      add=T)\n",
    "legend(\"topright\", inset=.02,\n",
    "       c(\"Target\",\"Decoy: Column names shuffled\"), fill=c(\"#0000FF40\",\"#FF000040\"), horiz=F, cex=0.8)\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b3f6304",
   "metadata": {},
   "source": [
    "### For 100% FDR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56f5d712",
   "metadata": {},
   "outputs": [],
   "source": [
    "TD <- cbind(target_histogram[,-3],decoy_histogram$Counts) #combining both Target and decoy bins\n",
    "colnames(TD) <- c('Breaks', 'T_C','D_C')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c2ced706",
   "metadata": {},
   "source": [
    "Here, we are considering all bin values greater than or equal to 0 from TD as Positive FDR and the bins less than 0 as Negative FDR."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "4dcde913",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>Breaks</th><th scope=col>T_C</th><th scope=col>D_C</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>1.00</td><td>9   </td><td>0   </td></tr>\n",
       "\t<tr><td>0.99</td><td>1   </td><td>1   </td></tr>\n",
       "\t<tr><td>0.98</td><td>1   </td><td>0   </td></tr>\n",
       "\t<tr><td>0.97</td><td>5   </td><td>0   </td></tr>\n",
       "\t<tr><td>0.96</td><td>2   </td><td>0   </td></tr>\n",
       "\t<tr><td>0.95</td><td>4   </td><td>2   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lll}\n",
       " Breaks & T\\_C & D\\_C\\\\\n",
       "\\hline\n",
       "\t 1.00 & 9    & 0   \\\\\n",
       "\t 0.99 & 1    & 1   \\\\\n",
       "\t 0.98 & 1    & 0   \\\\\n",
       "\t 0.97 & 5    & 0   \\\\\n",
       "\t 0.96 & 2    & 0   \\\\\n",
       "\t 0.95 & 4    & 2   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| Breaks | T_C | D_C |\n",
       "|---|---|---|\n",
       "| 1.00 | 9    | 0    |\n",
       "| 0.99 | 1    | 1    |\n",
       "| 0.98 | 1    | 0    |\n",
       "| 0.97 | 5    | 0    |\n",
       "| 0.96 | 2    | 0    |\n",
       "| 0.95 | 4    | 2    |\n",
       "\n"
      ],
      "text/plain": [
       "  Breaks T_C D_C\n",
       "1 1.00   9   0  \n",
       "2 0.99   1   1  \n",
       "3 0.98   1   0  \n",
       "4 0.97   5   0  \n",
       "5 0.96   2   0  \n",
       "6 0.95   4   2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Pos_FDR <- TD %>% filter(Breaks >= 0)\n",
    "Neg_FDR <- TD %>% filter(Breaks < 0)\n",
    "Pos_FDR <- Pos_FDR %>% as.data.frame() %>% arrange(desc(Breaks)) # arranging by descending order of corr coeff \n",
    "Neg_FDR <- Neg_FDR %>% as.data.frame() %>% arrange(Breaks) # arranged from smallest to largest ==> most negative to close to 0 values\n",
    "\n",
    "head(Pos_FDR)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d71b1f8c",
   "metadata": {},
   "source": [
    "In the subset version, from the TD table, we consider bin values within the range $ 0 \\leq Bins < 0.8$ for Positive FDR and range $ -0.8 < Bins < 0$ for Negative FDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "id": "f8d86878",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>Breaks</th><th scope=col>T_C</th><th scope=col>D_C</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>0.79</td><td> 50 </td><td> 6  </td></tr>\n",
       "\t<tr><td>0.78</td><td> 72 </td><td>10  </td></tr>\n",
       "\t<tr><td>0.77</td><td> 53 </td><td> 6  </td></tr>\n",
       "\t<tr><td>0.76</td><td> 94 </td><td>24  </td></tr>\n",
       "\t<tr><td>0.75</td><td>100 </td><td>14  </td></tr>\n",
       "\t<tr><td>0.74</td><td>109 </td><td>24  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lll}\n",
       " Breaks & T\\_C & D\\_C\\\\\n",
       "\\hline\n",
       "\t 0.79 &  50  &  6  \\\\\n",
       "\t 0.78 &  72  & 10  \\\\\n",
       "\t 0.77 &  53  &  6  \\\\\n",
       "\t 0.76 &  94  & 24  \\\\\n",
       "\t 0.75 & 100  & 14  \\\\\n",
       "\t 0.74 & 109  & 24  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| Breaks | T_C | D_C |\n",
       "|---|---|---|\n",
       "| 0.79 |  50  |  6   |\n",
       "| 0.78 |  72  | 10   |\n",
       "| 0.77 |  53  |  6   |\n",
       "| 0.76 |  94  | 24   |\n",
       "| 0.75 | 100  | 14   |\n",
       "| 0.74 | 109  | 24   |\n",
       "\n"
      ],
      "text/plain": [
       "  Breaks T_C D_C\n",
       "1 0.79    50  6 \n",
       "2 0.78    72 10 \n",
       "3 0.77    53  6 \n",
       "4 0.76    94 24 \n",
       "5 0.75   100 14 \n",
       "6 0.74   109 24 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Pos_FDR_subset <- TD %>% filter(Breaks >= 0 & Breaks < 0.8)\n",
    "Neg_FDR_subset <- TD %>% filter(Breaks < 0 & Breaks > -0.8)\n",
    "Pos_FDR_subset <- Pos_FDR_subset %>% as.data.frame() %>% arrange(desc(Breaks)) # arranging by descending order of corr coeff \n",
    "Neg_FDR_subset <- Neg_FDR_subset %>% as.data.frame() %>% arrange(Breaks) # arranged from smallest to largest ==> most negative to close to 0 values\n",
    "\n",
    "head(Pos_FDR_subset)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcecf9db",
   "metadata": {},
   "source": [
    "Now we calculate the FDR values: (Keep only FDR_subset; Remove other 2 cells)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "4a4fce3a",
   "metadata": {},
   "outputs": [],
   "source": [
    "Pos_FDR$fdr <- NA\n",
    "for (i in 1:nrow(Pos_FDR)){\n",
    "  temp <- Pos_FDR %>% filter(Breaks >= Pos_FDR$Breaks[i])\n",
    "  Pos_FDR$fdr[i] <- sum(temp$D_C)/sum(temp$T_C)\n",
    "}\n",
    "\n",
    "Neg_FDR$fdr <- NA\n",
    "for (i in 1:nrow(Neg_FDR)){\n",
    "  temp <- Neg_FDR %>% filter(Breaks <= Neg_FDR$Breaks[i])\n",
    "  Neg_FDR$fdr[i] <- sum(temp$D_C)/sum(temp$T_C)\n",
    "}\n",
    "\n",
    "Pos_FDR$round_fdr <- round(Pos_FDR$fdr, digits = 2)\n",
    "Neg_FDR$round_fdr <- round(Neg_FDR$fdr, digits = 2)\n",
    "\n",
    "head(Pos_FDR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "id": "4cea9748",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>Breaks</th><th scope=col>T_C</th><th scope=col>D_C</th><th scope=col>fdr</th><th scope=col>round_fdr</th><th scope=col>round1_fdr</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>0.79     </td><td> 50      </td><td> 6       </td><td>0.1200000</td><td>0.12     </td><td>0.1      </td></tr>\n",
       "\t<tr><td>0.78     </td><td> 72      </td><td>10       </td><td>0.1311475</td><td>0.13     </td><td>0.1      </td></tr>\n",
       "\t<tr><td>0.77     </td><td> 53      </td><td> 6       </td><td>0.1257143</td><td>0.13     </td><td>0.1      </td></tr>\n",
       "\t<tr><td>0.76     </td><td> 94      </td><td>24       </td><td>0.1710037</td><td>0.17     </td><td>0.2      </td></tr>\n",
       "\t<tr><td>0.75     </td><td>100      </td><td>14       </td><td>0.1626016</td><td>0.16     </td><td>0.2      </td></tr>\n",
       "\t<tr><td>0.74     </td><td>109      </td><td>24       </td><td>0.1757322</td><td>0.18     </td><td>0.2      </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       " Breaks & T\\_C & D\\_C & fdr & round\\_fdr & round1\\_fdr\\\\\n",
       "\\hline\n",
       "\t 0.79      &  50       &  6        & 0.1200000 & 0.12      & 0.1      \\\\\n",
       "\t 0.78      &  72       & 10        & 0.1311475 & 0.13      & 0.1      \\\\\n",
       "\t 0.77      &  53       &  6        & 0.1257143 & 0.13      & 0.1      \\\\\n",
       "\t 0.76      &  94       & 24        & 0.1710037 & 0.17      & 0.2      \\\\\n",
       "\t 0.75      & 100       & 14        & 0.1626016 & 0.16      & 0.2      \\\\\n",
       "\t 0.74      & 109       & 24        & 0.1757322 & 0.18      & 0.2      \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| Breaks | T_C | D_C | fdr | round_fdr | round1_fdr |\n",
       "|---|---|---|---|---|---|\n",
       "| 0.79      |  50       |  6        | 0.1200000 | 0.12      | 0.1       |\n",
       "| 0.78      |  72       | 10        | 0.1311475 | 0.13      | 0.1       |\n",
       "| 0.77      |  53       |  6        | 0.1257143 | 0.13      | 0.1       |\n",
       "| 0.76      |  94       | 24        | 0.1710037 | 0.17      | 0.2       |\n",
       "| 0.75      | 100       | 14        | 0.1626016 | 0.16      | 0.2       |\n",
       "| 0.74      | 109       | 24        | 0.1757322 | 0.18      | 0.2       |\n",
       "\n"
      ],
      "text/plain": [
       "  Breaks T_C D_C fdr       round_fdr round1_fdr\n",
       "1 0.79    50  6  0.1200000 0.12      0.1       \n",
       "2 0.78    72 10  0.1311475 0.13      0.1       \n",
       "3 0.77    53  6  0.1257143 0.13      0.1       \n",
       "4 0.76    94 24  0.1710037 0.17      0.2       \n",
       "5 0.75   100 14  0.1626016 0.16      0.2       \n",
       "6 0.74   109 24  0.1757322 0.18      0.2       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Pos_FDR_subset$fdr <- NA\n",
    "for (i in 1:nrow(Pos_FDR_subset)){\n",
    "  temp <- Pos_FDR_subset %>% filter(Breaks >= Pos_FDR_subset$Breaks[i])\n",
    "  Pos_FDR_subset$fdr[i] <- sum(temp$D_C)/sum(temp$T_C)\n",
    "}\n",
    "\n",
    "Neg_FDR_subset$fdr <- NA\n",
    "for (i in 1:nrow(Neg_FDR_subset)){\n",
    "  temp <- Neg_FDR_subset %>% filter(Breaks <= Neg_FDR_subset$Breaks[i])\n",
    "  Neg_FDR_subset$fdr[i] <- sum(temp$D_C)/sum(temp$T_C)\n",
    "}\n",
    "\n",
    "#rounding the fdr upto 2 digits\n",
    "Pos_FDR_subset$round_fdr <- round(Pos_FDR_subset$fdr, digits = 2) \n",
    "Neg_FDR_subset$round_fdr <- round(Neg_FDR_subset$fdr, digits = 2)\n",
    "\n",
    "head(Pos_FDR_subset)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0979b314",
   "metadata": {},
   "source": [
    "Here, we subset the Positive and Negative FDR tables for $0$ to $ \\pm 0.8 $ range after calculating FDR values:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "2e1311aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>Breaks</th><th scope=col>T_C</th><th scope=col>D_C</th><th scope=col>fdr</th><th scope=col>round_fdr</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>-0.79</td><td>0    </td><td>0    </td><td>NaN  </td><td>NaN  </td></tr>\n",
       "\t<tr><td>-0.78</td><td>0    </td><td>0    </td><td>NaN  </td><td>NaN  </td></tr>\n",
       "\t<tr><td>-0.77</td><td>0    </td><td>0    </td><td>NaN  </td><td>NaN  </td></tr>\n",
       "\t<tr><td>-0.76</td><td>1    </td><td>0    </td><td>  0  </td><td>  0  </td></tr>\n",
       "\t<tr><td>-0.75</td><td>1    </td><td>0    </td><td>  0  </td><td>  0  </td></tr>\n",
       "\t<tr><td>-0.74</td><td>1    </td><td>0    </td><td>  0  </td><td>  0  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllll}\n",
       " Breaks & T\\_C & D\\_C & fdr & round\\_fdr\\\\\n",
       "\\hline\n",
       "\t -0.79 & 0     & 0     & NaN   & NaN  \\\\\n",
       "\t -0.78 & 0     & 0     & NaN   & NaN  \\\\\n",
       "\t -0.77 & 0     & 0     & NaN   & NaN  \\\\\n",
       "\t -0.76 & 1     & 0     &   0   &   0  \\\\\n",
       "\t -0.75 & 1     & 0     &   0   &   0  \\\\\n",
       "\t -0.74 & 1     & 0     &   0   &   0  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| Breaks | T_C | D_C | fdr | round_fdr |\n",
       "|---|---|---|---|---|\n",
       "| -0.79 | 0     | 0     | NaN   | NaN   |\n",
       "| -0.78 | 0     | 0     | NaN   | NaN   |\n",
       "| -0.77 | 0     | 0     | NaN   | NaN   |\n",
       "| -0.76 | 1     | 0     |   0   |   0   |\n",
       "| -0.75 | 1     | 0     |   0   |   0   |\n",
       "| -0.74 | 1     | 0     |   0   |   0   |\n",
       "\n"
      ],
      "text/plain": [
       "  Breaks T_C D_C fdr round_fdr\n",
       "1 -0.79  0   0   NaN NaN      \n",
       "2 -0.78  0   0   NaN NaN      \n",
       "3 -0.77  0   0   NaN NaN      \n",
       "4 -0.76  1   0     0   0      \n",
       "5 -0.75  1   0     0   0      \n",
       "6 -0.74  1   0     0   0      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Neg_FDR_newVal <- Neg_FDR %>% filter(Breaks > -0.8) #Specifying the limits of correlation scores for Neg_FDR \n",
    "Pos_FDR_newVal <- Pos_FDR %>% filter(Breaks < 0.8)  #Specifying the limits of correlation scores for Pos_FDR\n",
    "\n",
    "head(Neg_FDR_newVal)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bbf5137",
   "metadata": {},
   "source": [
    "Here, we try to plot the FDR plots with full range (-1 to 1 correlation scores) along with the FDR values subsetted before and after FDR calculation to observe the change in FDR calculation:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 320,
   "id": "15e69a3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 3"
      ],
      "text/latex": [
       "\\textbf{pdf:} 3"
      ],
      "text/markdown": [
       "**pdf:** 3"
      ],
      "text/plain": [
       "pdf \n",
       "  3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA8AAAAPACAMAAADNCOCpAAAAUVBMVEUAAAAAAP8A/wBNAABN\nTU1oAABoaGh8fHyMAACMjIyampqnAACnp6eysrK9vb2+vr7Hx8fQ0NDZAADZ2dnhAADh4eHp\nAADp6enw8PD/AAD////oUyNuAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3di5ai\nSrRw6chqDrQ2uvW3bTm8/4M294uiIhDBWhHzG2PvNFMUKstZRACZmhyAWmbvDQCwHAEDihEw\noBgBA4oRMKAYAQOKETCgGAEDihEwoBgBA4oRMKAYAQOKETCgGAEDihEwoBgBA4oRMKAYAQOK\nETCgGAEDihEwoBgBA4oRMKAYAQOKETCgGAEDihEwoBgBA4oRMKAYAQOKETCgGAEDihEwoBgB\nA4oRMKAYAQOKETCgGAEDihEwoBgBA4oRMKAYAXvBDAw+TdLH0/2H8+TjosO1/Xzy+Y82Nx4r\nELAX3gRcuD7fn7x5XNp8PvHst4iXiVT8zXjhfcDm9nz/+c3jbvm7gN/tl7E//ma88JRY82l2\njYvR8fD+LDUmnnjcNTLm+Po8088OQfib8cJ0wIW4HkT3Xxgv2X1273bd9efXY3HzeG2WMSQs\nFX8vXngb8PV5z/om4HwccNIe88oJWDb+XrzwNuCsHjK3X3gc6yhfFrzXY+3m80M3MT4QsGz8\nvXhhfAxr2PObo1qje5/nwNfyUFeWZyfzPACHNPzNeOGHgK9vHmfu3dLH9lB1+unQFiTgb8YL\nswM+Ze8edx0tXS/2GM+MIQ9/M16YNQd+JMVE9/H8uEqSZoMHmjeHtiAPfzNeeBvw+Ch08nQh\n1kuar3vgaGIpyMHfjBfeBpyMD0NFxlw+PK75/MAcWA3+ZrzwJuDb4elKrFvz+ZvHfTgKPZ47\nQwoC9sJrwFMHp/LBznXycYOhdqP6MaRiv13/rAPEIWAvvA/46ccEH938dupxL1di1T9GWF5W\nOZ47QwoC9sKbgNuDy4P709GlWO8Czq/HqL0WOq/22/xEsEwEDChGwIBiBAwoRsCAYgQMKEbA\ngGIEDChGwIBiBAwoRsCAYgQMKEbAgGIEDChGwIBiBAwoRsCAYgQMKEbAgGIEDChGwIBiBAwo\nRsCAYgQMKEbAgGIEDChGwIBiBAwoRsCAYgQMKEbAgGIEDChGwIBiBAwoRsCAYgQMKEbAgGIE\nDChGwIBiBAwo5iBgA3n+23sDMGVBXdsHu8Mq8Kv/9t4ATCBgzETAEhEwZiJgiQgYUIyAAcUI\nGFCMgDETc2CJCBgzEbBEBIyZCFgiAsZMBCwRAWMmApaIgAHFCBhQjIABxQgYMzEHloiAMRMB\nS0TAmImAJSJgzETAEhEwZiJgiQgYUIyAAcUIGFCMgDETc2CJCBgzEbBEBIyZCFgiAsZMBCwR\nAWMmApaIgAHFCBhQjIABxQgYMzEHloiAMRMBS0TAmImAJSJgzETAEhEwZiJgiQgYUIyAAcUI\nGFCMgDETc2CJCBgzEbBEBIyZCFgiAsZMBCwRAWMmApaIgAHFCBhQjIABxQgYMzEHloiAMRMB\nS0TAmImAJSJgzETAEhEwZiJgiQgYUIyAAcUIGFCMgDETc2CJCBgzEbBEBIyZCFgiAsZMBCwR\nAWMmApaIgOEX0xp/sf5vvExyW7227GhMuvpZViBg+OWHgI1ZXfCheJLT2idZg4DhFzP14nkJ\nuPqQmmT92h5rn2LlBjh5iMBV4FdK5sA/BDy97Pq1uUTAmElfwPXNNt63AV+LcXCU1p8/Diaq\nh8RpVExu6wXOsYnOw3UUX4jPeTsUb796TYpp9bV9cPIYLVosm8XmMHq2wQOW/3GdPKRxOx2q\nP/Eh/TL3IGCB/Au4HkKf6gzT6uuRaSa1SXnjWC1Yv2gHo+2k/cIo4HP92bldIMqGixbLHqqV\n9M82eMCKP66Th1SKf4B6nycfBCyQ/YDNT748Sf4x4Ma9+uSS55fmAUlWdBUXO0cT3fN7VH71\nWn4xS0y3q7w0913y0ZNG5bNdygdfykccy1oHi1bPPXq2/gFrvmdOHlJJTXS5V7ce1+jzsXcC\nFkjNHnh2wMl9+LDqf7fm5qHq9VrfLMPLquFvpb0vyZ+e9NotcCsfEI0WbQ55D56tf8CaP66T\nh1Sqf3Aa9/JPt/0qgNlD6Dhq+3lcT8lE8d3Np13+4L7hk6bFIPl+H2/C6Gmam92z9Q9Y88d1\n8pD6cebdJ5utApgd8K09BZS82WXnvwWcn8oJdPSYHXD/gDV/XCcPqbAHhgPzD2Id6lHx0cTn\n6+NTwNNreAm4GCuncTml/RzwxANWcDsHvtb/2jAH1kjNHHh88/Y24Ht7EKv433PAoznw01S1\nve+Qv6k7eZkDH7olX55t5Zlkl6eRksEhxDizsgrYoy/g2JzLA75vTyMdmgZv+f15Djw4Cl0d\nSc7P/UGsN0eh4/podlwum2TF/ur5KHT/4PrZ+ges+eM6eUjjllYnwaLDifPA+ugLuDrRengf\ncFbtgtNmn3IbBtztbbqbg7lqf3J3+KSX7mnenAcePrh8tsEDVvxxnTxE4CrwK30Bl0eJjh8u\n5CjaLXerx/LnkgbD3Pr/5cVUt/rmOTbmODzWdI6ay6tGa6surLo1T2wOj/Gi/QUf3bMNHrD8\nj+vkIQJXgV8pCXhT63/awTaXAZc/O9le+slpJEhWXZ6VHfb9Wd85HAacVdeZNscCCBiSNRdI\nfzzZKYLT00jFXCA7R8+T/w1XAWzkXMxPY/H7X8cXclQfHlH8IGCFQpwDy7fDpZRZkkwFPOcn\nTbAjApbIYcCxaS/eiBP2wPoQsEQOAz6bY3PrYRICVoeAJXJ5Gintqr1+GSUTsEAELJHTCznu\n3eWkjyMBA+txJRagGAFjhr9/e28BphEwZigDZg4sEQFjhn9/BCwTAeO7agRNwBI5vRJr9sVW\nBCxLuQMmYJGcXshBwDrVh7AIWCKXQ+h7NPfHowlYlGoHDJHcXsgx9+ejCVgSziEJ5vYg1tnM\n+0X0BCwJO2DBOAqNL9odMHNgiQgYXxCwZASML9oRNAFLRMD4rDuERcASETA+I2DRCBifcQxa\nNALGR5wElo2A8REBy0bA+KgfQTMHloiA8clgB0zAEhEwPiFg4QgYnwyOQWsJOEtjY5Lz+Itf\n3+zj+nXZ8s01Z/0wzvAnZusbcZoN7/n6nsDvtvb561cCxkfDQ1hKAm7eBNNE2fCr3wKOB/e/\nWfZQPOlpzha8BlxszWN0z5eCZwZcbjQB4wOFx6CPJilieSTjveW3gM33gI15zNuC1+cqtmb4\nnpzpt/cNnxmwIWB8pvAqDlO/BVc2frFvEvDsLXi9HZvr4LNftubD1wkYX6gM+Pl29To3xX4v\nqnfK16SYhtZz3nNsonPejG0Hj+qWHS9h6i/E53qxLK7err5d5P0WlG8mdMzfBJxG1ZihWKYY\npNerre9v7xj9KQaLLX0jTwIOxmgErWQOnJpjN9QdvvQP1eGjvPvdbGVyB9N88SngbtmnJYpF\nkvYL9WLpYJHxWse3MxP3n42G0Ek7YT/Vq0jb5bo7xgH3ixEwvnAc8N9P3j1LUh71rY8SDV/6\n0T2/R+ZSvtH8Pc8vZVBXk2R5lpSj23F03bKDJepFLv19prxvvEj7+D6t7nlHh7QGv5jmUj78\nWPd4KT81zcLDO/o/xctiBIz3RiNoJXvgIqljuevq55z1y778/FoOeU3X2qGaLmf1F/vHD5Yd\nLFEvcmjuq/bJt6cn6R7/JeBk8IulDuWzZCbKx8sN73gaQo8XI2C8NT4GrSbgwu0Ula//l5d+\n+SEtxr73e/1Zl9rEsLf+8vCM0PN9+dOTjB8/vt0/Oo4G55xHCz+up6Qvcxzr4GvjxX7/5hBw\nKPQGXP4G1PhNwPkpak7MOgz40Uybi//dRuejBgsno1W9Dfhpsd+/NQQcCoXHoMe7vOmAiyFw\nGvd9jx+WT0Xaf3kq4LebMLh96Q9OFYPjw9TCRxOfr4+vAT8v9uX7MYGAQ6Ex4IOpz+j0s8db\n/dIvJ6zXPp16mnkdftrf7pY9jI5N5f0cuJ82DxeZeq76Y1w9Z/3ZfXgQKxlPdfsyn+649Qex\nRosRMN5ReBlW9UI/Z8WHpAw5Lv6XJcOj0Nfyi5fmKHR1SDk/1zX2w9rBsoMlJo5CV0sPF2kf\nnz/dfroSa7gLPpcHm9P6YPMtv/eT2/6O4Z/iabEHAeOtp4CVzIHT9lBv3pzyPdQv+2N1O69O\nwZTKPWI9nSynw7EZHgjulh0s0dQ3PA9cLz5YpH18PrhtBnc392TjXXBzurfd8PbgW3fH8E8x\nWqzcaALGO08jaCUB5/djVBR2qW6fInNsB56pieqfRaiuxKpPFJ+LBqrrPm7x6ExOt2y/RFvf\nOequxMqfF+keP7hd9X4a35MOZ8HlUfHq0cdyswaj8+6O4Z9iuFi50QSMN55H0FoCDgsB4w0C\n1oCA8QYBa0DAeEPjSaTwEDDeIGANCBjTVJ4FDg8BY9pLwMyBJSJgTCNgFQgY016mwAQsEQFj\nGgGrQMCY9HoMi4AlImBM4iC0DgSMSQSsAwFjEpdx6EDAmPQaMHNgiQgYUyZG0AQsEQFjCgEr\nQcCYQsBKEDCmTBzDImCJCBhTOAitBAFjAmeBtSBgTCBgLQgYE6ZG0MyBJSJgTCBgLQgYryZH\n0AQsEQHjFQGrQcB4NXkSiYAlImC84iywGgSMF5xE0oOA8YKA9SBgvJgeQWuZA2dpbExyHn/R\nfHtJXd882dGY9N2d3XM3uttxmg3vad7L9NMzzPv69fWLBIwXqgPOouYttbPhV78FHL+5/1A8\n0+ndnd1zPwfcv6F3/27cn59h1tfr7SBgfPZmBK0k4KNJingeSbHjHPgW8PuEHl8fPH5D7/L/\nxdqTwT1p/dmStX9djIDxTHfAxlS73uxpR7U44O8Pfg243FleB5+tWvvnxQgYz7QH/Hy7/H/x\nX2qieqd8TYppaT2tPccmOufNWLd72LUYOFfLGjMcHndLl/9IxObwfo3FU5hj/ibgNKrGCIPV\nNPe3d4y2+nlr2ruLbYmrLV/wHfr9IQJXgbd0nwVOzfHR3h6mcKgOJxWfnusiy1f/wTRfHAV8\nqhdInwPulq6fLR2vZXw7M3H/2WgInbQT9H419XLdHeOAn7ZmuHC15b9/hwjYc7sF/O8n754l\nKY8C10eNhilE9/wemUueR+ae55cysKtJsjxLytHuOMJLuUB/xKi+OVo6yYbL9zvp7nlGh7TK\nNTYu5UOPdY/dasr/De/ot/plsXbh9o9DwHii/izw9Vjuyvo5aJ1B+fm1HPfWN0uHarqc1V98\nfpaXgEdL30ZLfgk46fstnuVWPkmUj5cb3vE0hB4v1i5c/3ESAsazdwErmQNXbqeo7OElhfJD\nWgyA7/f6sy69ccCP6yl5Cfjt0lND6MED4+j6ZuFuNROxDr42Wuz5j0PAeOJDwHl+L8fIkwHn\np6g5UfsuyeRpj7og4EczVy7+d6tORU0s3K/mY8DjxQgYX7ybAusIeLQLnA64GHumcd/3+GF5\neSY5Pl8fEwFPLj0d8KU/OFWMdw9TCw9W8yngp8UIGF/oDvhg6oso+9nkrX7Fl7PW6/jcz6Gb\nDb9G+Brwm6WnA46r9dWf3YcHsZLxVLcv8+mOW38Qa7TYeA58IGA8eXsMS0fAxQv/nBUfkjLk\nuPhfltQZ1Idtr+UXL81R6OpYbn6uD0v1w9wy9vt4DlzeOVp6uMrXgJ+uxBrugs/lwea0Ptjc\nrab8X3/HcKufFmtPE3MUGu9oPwidtod+8+aU76HO4FjdzqtTMqVyD1lPL8vpcGz6A8PtM3T7\n0ObOfulPAZtuqe6ebLwLbk73DlbTz3XLO4ZbPVqs3o7BxJjzwHih+zKOwv0YFS/tS3X7FJlj\nOxBNTXSqvlhdiVWfBzoXTVTXfdziPuBi2lncXw+36+Fsc2e39LeAk9P4nnQ4Cy6Pgj/Gq6mX\n6+4YbvVwsXo76oXPEVdiYYr6gMNCwBh7G7COOXBoCBgj76fABCwRAWOEgHVxGvDtVP9AxyH9\n9hsKFq8CK72fAhOwRA4DzmLT+/wrCgh4NwSsi8OAUxNd6tNhj2s0/oUnW60Ca2k/CxwchwFH\ng7PZ98FZtw1XgbUIWBmHAY/OfpuPz0LAeyFgZdgDY+jDZRzMgSVyOwe+1leMMwcWi4CVcXka\nKRkchY6zT0sS8F4IWBm354HT6jxwdDhxHlimT1NgApaIK7EwQMDaEDAGOAitDZdSYoCfJdSG\nSykxQMDacCkleh9H0MyBJeJCDvQIWB05l1KaoYWrwDoErA57YPQ+ToEJWCIupUSPgNXhUkr0\nOAitDpdSosNlHPpwJRY6BKwPAaPzOWDmwBIRMDqfp8AELBEBo0PA+hAwWl+mwAQskdMrsWZf\nbEXAeyBghRwGfCZg2TgIrZDLIfQ9+vxDhBusAisQsEJO58D3zxdQbrEKLEfACrk9iHUe/DyD\npVVgsS8XUjIHloij0GgRsEIEjBYBK0TAaBGwQgSMxrdjWAQsEQGjwUFojQgYDQLWiIDRIGCN\nCBiNb79PhzmwRASMBgFrRMBoELBGBIwGAWtEwKh9PYZFwBIRMGochFaJgFEjYJUIGDUCVomA\nUfv6tirMgSUiYNQIWCUCRo2AVSJg1AhYJQJG5fsxLAKWiIBR4SC0TgSMCgHrRMCoELBOBIzK\n12NYzIFFImBUCFgnAkaFgHUiYFQIWCcCRmnGMSwCloiAUeIgtFIEjBIBK0XAKH2fAkMkAkZp\nRsDMgSUiYJQIWCkCRomAlSJg5POOYRGwRASMnID1ImDknEXSi4CRE7BeBIyc08B6ETDyeQEz\nB5aIgJETsF4EjJlTYAKWiIBBwIoRMAhYMQIGZ5EUI2BwFkkxAgYBK0bAmBkwc2CJCBgzp8AE\nLBEBg4AVI2AQsGIEDAJWjIDBaWDFCBicRVKMgEHAihEwZgbMHFgiAgYBK0bAmHkMi4AlImAQ\nsGIEDAJWjIDBaWDFCBjzjmGRuUgEjJmngTlbLBEBY2aZ/7ELFoiAQcCKEXDw5k5u/2MMLRAB\nB4+ANSPg4M0OmOPQAhFw8GZ3ScACEXDw5nfJGFoeAg7e/CzZBctDwMGbG/B/BCwQAQdvfsCM\noeUh4OARsGYEHLrZ4+L/mAQLRMChI2DVCDh0P0XJGFoaAg7dTwGzC5aGgEP3006VgKUh4NDN\nDvi/n5aGGwQcOgJWzWnAt9PBlA7pzdYq8KvfAmYMLYzDgLPY9BIrq8DP5hdJwBI5DDg10eVe\n3XpcI5PaWAV+9mPAjKGFcRhwZO7d7buJbKwCP/t1l0rAsjgM2Jh3n2y2Cvzs14AZQ8vCHjhw\nBKyb2znw9VHdYg4sx/wh8X+/PgAOuDyNlAyOQseZlVXgVwSsm9vzwGl1Hjg6nDgPLMXPATOG\nFoUrsQJHwLrJCdgM2VkFXvyQY/v+wIyhJXEZ8ONoolOen2MTfTyExR7YnQX7U3bBkri8lDIq\n963nE5dSCkLAyjk9jVTsd9PIHLM8SzmNJMOSGhlDC+L0Qo7q0aY6gcSFHDIsmAMTsCTOL6Vs\nDlBxKaUMP8TYBcwYWpAd9sDl/zP2wDIQsHI7zIHTrLm9/SrwqyUBM4YWhKPQYSNg5TgPHLRl\no2HG0HLIuRLL8SpQWpaiYRcsBgEHbeG+lIDFIOCg/RJwPwcmYDkIOGgLA2YSLAYBB+2XXekg\nYCbBYhBw0BYGzBhaDAIO2tKAGUNLQcBBW7onZQwtBQGHbPmOlICFIOCQEbB6BByynwIezoGZ\nBEtBwCFbHjCTYCEIOGTLA2YMLQQBh+ynCglYIgIO2YqAmQTLQMAhW7EbZRIsAwGHbE2EBCwC\nAQds1TCYgEUg4ID9FvB4DswkWAYCDtiqgJkEi0DAAVsVMGNoEQg4YASsHwEHbF3ATIIlIOCA\nrduHErAEBBywdQFzFEsCAg7YygIJWAACDthvBT7PgQlYAgIO2MqAmQQLQMABI2D9CDhcPwb4\nEjBHsQQg4HCtDZhJsAAEHK7VQ2AC3h8Bh2t1wEyC90fA4SJgDxBwuFbPgQl4fwQcrh+nsK8B\nMwneHwGHa33A7IJ3R8DhImAPEHC41gfMGHp3BByuDeoj4L0RcLg2qI8x9N4IOFhbxEfAeyPg\nYP0a39QcmDH03gg4WATsAwIO1iYBM4beGQEHi4B9QMDB2iRgxtA7I+BgbZMeAe+LgIO1TXqM\nofe1VcD3w9ot+boKbIuAfbAm4FtiTHIvb90PZtPmCNiBXwOengMzht7XioBvpnLPH0W+Jt15\nq/CrjQJmF7yrFQEnZbSpSa5Fvods763CrwjYBysCrkfNxkTmcN9wi4argD0/h/cmYMbQu9og\n4Pi24faMVwF7CNgLGwS84dY8rwL2bDb0ZQy9JwIOFQF7gYBDtV13jKF3tCrgkZ23Cj/6Obt3\nc2AC3hMBh2q7gBlD74hroUNFwF4g4FBtFzBj6B0RcKgI2AvrAr4e42L2m6RbX8pBwNZtOe5l\nDL2fNQE/ku4IVvLYcqMI2D4C9sOKgLPIxNfyhxgel9hEW24UAdu3aXSMoXezIuDUJN3XEnPa\nZnvGq4A1vwf8fg7MLng/KwKOTT9ufgxi3gABW0fAflh9KeXUJ6sRsHW/j3o/BMwYejcEHCgC\n9gMBB2rbgBlD74WAA7XtPpOA98IPM4Rp6+IYQ++EgMNEwJ7gWugwLQj40xyYMfReCDhMBOwJ\nAg7T1gEzht7JVkeht0XAti3ojYAl4pfahWnzgBlD74OAw7T5DpOA90HAYdp+xMsYehcEHCYC\n9gQBB2nJgPfzHJgx9D4IOEgE7AsupQwSAfvCacC306Fa9PDt11gSsGUWAmYSvAuHV2Jl8SD3\nz7+Bh4AtI2BfOAw4NdHlXt16XCOT2lgFZrIRG2PoPTgMODL37vb98++hJWDLCNgXDgP+4Td4\nELBlBOwL9sBBWhLwtzkwk+A9uJ0DX+vfJM0ceGeLdpYELJHLnwdOBkeh48zKKjCLnYAZQ+/A\n6Q/039LqPHB0OHEeeFcE7A1+I0eICNgbcgK2dl0mXlhKjUmwey4DztLy0PMpNia5WFoFZrFU\nGrtg9xwG/IiKPWsWcSnl/gjYGw4DPppDVvzv+ChaPnIaaU+LAv46B2YMvQOnV2Jlzf+K0TQX\ncuyJgL3h+lLKyAw+2XwVmMdWwIyhnXM6hL7n+am+njL7PAkmYKuWdUbAEjkM+G6i9J4foqLg\na2yuNlaBWawFzBjaOZenka5Rf6L3ZGcVmMPejpKAXXN7IcflWP1WjsPpYW0V+M5ewIyhXZNz\nJZbjVYTMXmaGXbBjBBygZZXNmQMzhnaNgANEwP4g4ABZDJhJsGMEHCCLATMJdoyAw7NwLzkr\nYMbQjhFweKwOcxlDu0XA4bHaGGNotwg4PHZ3kgTsFAGHx+ocmIDdIuDwLExsZsBMgp0i4PDY\nDZhJsFMEHB67ATOGdoqAw2M5YMbQLhFweCzvIgnYJQIOjvXAGEM7RMDBIWCfEHBwlgY8dw7M\nGNolAg4OAfuEgINjPWDG0A4RcHCW5jU/YHbB7hBwcAjYJwQcHAcDXMbQzhBwcAjYJwQcGhfj\nW8bQzhBwaBbHNX8OTMDuEHBoXATMGNoZAg6Nk4DZBbtCwKEhYK8QcGgWj25/CZgxtCsEHBo3\naRGwIwQcGjdpMYZ2hIBDQ8BeIeDALC/rpzkwY2hHCDgwBOwXAg6Mq4AZQ7tBwIEhYL8QcGBc\nBcwY2g0CDoyzrgjYCQIODAH7hYAD46wrJsFOEHBglgf84xyYgJ0g4LCsyOrHgBlDO0HAYSFg\nzxBwWBwGzBjaBQIOCwF7hoDD4nBcS8AuEHBYXE5MmQQ7QMBhcRkVu2AHCDgsKwL+dQ5MwC4Q\ncFhcBswY2gECDsqanSIBS0TAQXEbMGNo+wg4KATsGwIOiuOkGENbR8BBcVwUu2DrCDgoBOwb\nAg7KmoB/nwMzhraPgINCwL4h4JCsGtIuCZgxtG0EHBIC9g4Bh8R5wIyhbSPgkLjfIbILtoyA\nQ0LA3iHgkOyQE2Nouwg4JKtqWjQHZhdsGQGHhIC9Q8Ah2SFgxtB2EXBICNg7BBySPQJmDG0V\nAYdkl70hu2CbCDgg++wMCdgmAg7IPgEzhraJgAOyLqWFc+DcsAu2iIADsk/AjKFtIuCA7BQw\nY2iLCDgg63aFiwNmDG0RAQdkp4AZQ1tEwAHZKyTG0PYQcED2CpgxtD0EHJDdOiJgawg4IHvN\ngRlD20PA4ViZ0YqAGUNbQ8Dh2C9gxtDWEHA4dgyYMbQtTgO+nQ6mdEhvtlaB9wjYQw4DzmLT\nS6ysAp/sWRFjaEscBpya6HKvbj2ukUltrAKfELCHHAYcmXt3+24iG6vAJ3tGxBjaEocBG/Pu\nk81WgU9WBrxmDkzAtrAHDseeATOGtsTtHPj6qG4xB94FAXvI5WmkZHAUOs6srAIf7BowY2g7\n3J4HTqvzwNHhxHlg99YmRMAScSVWMHZOiDG0FXICNkN2VhE2AvbRLgF/DZSALdg5YMbQVhBw\nMNbuAtfNgQnYDqcXcsweJROwBTsHzBjaCocB3yIC3tPeAbMLtsHlEDo7mKS6koMh9B4I2Edu\n58AXYy45Ae9idT9rA2YMbYPjg1iPxBwyAt7D/jvA/bfAQ86PQp9MdCXgHeyfz/5b4CH3p5Hu\n8fcLNQh4ewLyYQy9vT3OAx8JeAer61k9B5bwb4h35FxK6XgVwSFgLxFwKAQEzBh6ewQcCgkB\nswveHAEHYn07BCwRAQdCRjuMobdGwIGQEbCMrfAJAQdCSDrsgjdGwIFYX84Gc2AC3hwBB0JI\nwEIGAv4g4EAICZhd8MYIOBBSAmYXvC0CDsMG3RCwRAQcBjndMIbeFAGHQU7AcrbECwQcBjnZ\nyNkSLxBwGDYYuG4zB2YMvS0CDoOggNkFb4mAwyAoYHbBWyLgIGyx1yNgiQg4CKICZgy9IQIO\ngqxm2AVvh4CDICtgWVujGwEHQdY+z8jaHNUIOAhbFLPZHFjavyeqEXAQhAXMGHozBByCTYLZ\nMGDG0Jsh4BBIC5gx9GYIOATiAmYMvRUCDoG4XhhDb4WAQyAvF3lbpBQBh0BeLuLGBFoRcAg2\nCXjLOTBj6K0QcAC22d1tGrDAQYFOBBwAiQEzht4GAZx7tdQAABLoSURBVAeAgP1FwAGQGDBj\n6G0QcABEtsIueBMEHACRAQvdKm0IOAAyU2EXvAUC9t9GpWw8B5b674oyBOw/qQGzC94AAftP\nasDsgjdAwP4TGzC74PUI2H8b7egIWCIC9p/ckarcLVODgP0nNxN2wasRsPckVyL33xYtCNh7\nWwW8/RxY9j8uOhCw9yQHzC54LQL23laNWAmYXfBKBOw90QGzC16JgL0nO2B2wesQsO+EFyJ8\n88QjYN9JL4Qx9CoE7DvpAUvfPuEI2Heb7eHszIHZBa9DwL4THzC74DUI2HfiA2YXvAYBe267\n/Zu1gNkFr0DAnlMQMLvgFQjYcxp2bxq2USoC9pyGvRtvVbgcAXtORRsqNlImAvbcdm3YmwOz\nC16OgP224fTSYsDsghcjYL8pCZjDWEsRsN+UBMwueCkC9tuGYVgNmF3wQgTsNy17tj8tGyoN\nAXtNz45Nz5bKQsBeU5QFu+BFCNhrWwZsdQ6s6t8aSQjYa4oCZhe8CAF7bcsobAfMLngJAvaa\npoDZBS9BwF5TFTCnkhYgYK/pSoJB9O8I2GfaitD1740IBOwzbQFr214BCNhnmwZhfQ6cswv+\nHQH7bNMeXATMLvhXBOwzdQGzC/4VAftMX8CcSvoRAftMX8AhDaLNFi9zpwHfTgdTOqQ3W6vA\ngMoYwtkFmy0KdhhwFpteYmUVGFEZcDiDaG0Bpya63Ktbj2tkUhurwIjKgJVu9QLaAo7Mvbt9\nN5GNVZxjE6VZdTON2puX2MTVkD0z8cLnVWrbFNzMgfNwBtHaAh5t7udtX7iKtBqdR2W2SXWz\nDPZW7OxTc6vuvi57Xq22LcFZwKHsgrUFbH0PfDfHot2zOZbVRvf8HpXdJiYr9r1JgDtgrQGH\nsgvWFnAxB74+qluW5sCH+mHlt6Xe2V7MqdnX918LidaAA9kFawu4GdbW4szKKqoHm7Ll8p+K\nuzn0AYe3A964A3cBB7ILNlsU7PY8cFqdB44OJ4vngavhsul3xu0QOrwdsN4dmd4t/4W+gJ2s\n4lyGOgi4OYj1CG4HrDmDIHbBngU8GF+b/y0VX1vw8f/7v4ph8/+WT1E9Z/GVa2z+5//87/9t\nrpfqxsLnVfjx35+M7eDj5Mei3i2e53tZzxYHnB2NSZpxrJXTSOU6omTw9P1a7ibuzycFYuPd\nmMM5sOrRw1zFi1PXHjiL6guh6yexFHBSD5Sj54AP5tqdTwrE1g04DTiAQbS6gFNzLio+j3aR\nG6/iESf1iar6KPSj+deiPh7dnU8KhO6A/b8kWl3AzV7xEcUPSwFfu/3rqTrkfO3ONh/MPbiA\nty7AbcD+D6LLl+L6l+MOl1JmSWIn4Ec/Ph5ciVWqTgiHNoRWHrD3g2h1AcemvXgjTqwEfGwP\nYlcrG/7QYrkDzgM7iKV+D+b7IFpdwNVFypViV2kjYDMIOKt+Gqm5417Pha/tjyUFQX3Anhbc\nvbbVBVz+sFBz6/rlFHY4E1V79Afswx/hRf/K1xdwfm8PCuePIwFbtvnuy/UcOPdrGtxdmtBG\nqzBgSavwng8B+zSINu1ZENOlTMB4a/vh5w4B+1RwF/Dg1gYFE7Cn/AjYo4KrsfOgWgLGJ9u/\n8HcJWPqBrL+/md/nut6X0TMBY5o3ey7Zf5BfAs4njl8RMCYJ33H9QPYg+m9uwaOA8/Foeg0C\n9pM/AcsuuA34W8ZPF+ITMD6z8KLfZw6cy/7HqAi3bPfrjvgpVALGZz4FLHgaXMdb9ftTwN31\nHKt/q45vAZ+7wwTtVdFBvjODjZ3WfgHLHURX2Vb1Lgx4dcGeBXxvvh33PuAg35nBr4DlFtwG\n/G0S/Frp6Ij0Cn4FfI+6gLvLroN8ZwYbL/gdAxZbcJ/t4oBXFuxVwOfuxxTP5XsyNM/VHv8L\naAcseNa4kNCC1wTc31pTlFcBF4PlLuBz98Um4JB2wJKP2y4ksuBhtZ8K/vLDs2sS9irge/et\nOpjrsfmB/hDfmcHDgEUWvE3AuVmRsFcB54OAu1+pE+I7M1h5re85By4VBUtLeGbA3+sk4O6R\n7cH5S/lrdaqBdP2bdA7VOzME8it1vAy4KFjaTnizgJcfypIZ8PK3mjBm8Hlm/qf9+v9r/qfY\nFf/v/2P+z+5vqWH/o523VPlv9z+X7I9/7+5f9VZBkt5axcUqxv+QBfrODHamwLvvgXNpw+hi\nW8w/03/yxryd68JdsMw98OYBh/XODHZGmhICljWMrgL+WvDMlxwB14+sHxpVv4O6f2uVsN6Z\nQdBrfHtyjkaXxf4z5t/g0ylzX3LLXpqeBpyWb6qSdWeOwnpnBh9PIg1IKfiv2gGXDbefTy42\nu0sCrh5ZP7R5J8ThWyOF884MngcspeB6B5x30+DpgH84QbSoYE8Drt6ZIW6vxgrsnRksvb5F\nzIErIgoue62Hz+U8uHjdTQb8ywneRZdz+BYwAghYwsHov3YHnJfl/fv3bzLgn5JcdEEWAXvH\n1ghaUMD7H4z+63fAtX9Tu+Dfg/z9AT8uv+whAlfhrxAC3nsY3fQ7eJ2aehc83qgFO1QCDp6t\nV7asgHcdRr/2W+6C69+tM/waAeNnvh+D7uw2jP6b6rcK+PlXYxEwfhZMwDvthOtKB1dgNcy/\nl9+NteSgMgGHTsIpFlfKnbDjP26X78tr9F/zu7H6DSJg/M5awMLmwDXn4+i/v/Kc0VRnzTFp\nAnazCl/ZG0GLDNh1wn9/9VUbE9qTSl3BS66sIuDQBRdw2ZS7gv+ep74Dz5dUEjB+Z+/FLDVg\nlwezPvVbHpce9UfAC7TvzJCnkYnS8mcKw3pnBovHoOUG7GoY/ff38dtbTo4Hs+OFP13046M8\nC7h9Z4Y8qS4sLYMN650ZAjqJNOJgJ1xOfz/tgCvd8emlv2gy7IC7d2a4mehefnYL7Z0ZQjqJ\nNGL9jFK5glm/na46yrX4F8UGHXD/zgz1zvZSvj9DWO/MEGzAVWH/5r7d9pInn/3MK/INPOD+\nnRkO5pE3Pwcc1Dsz2BxBC54DN6pBrpWC3R0mCzrg/p0ZBh+CemeGsAMubXw8q3ou4/BUc9AB\n51MBB/XODDZfaDoCrnbD24ykTXNY2en1mgT88iGkd2Yg4LwdSa/+TlThbvfPwdy1ehHw8t9U\n37wzQ/nL8MtNNe3Xg3pnBjsftb0zQ5HdqscX4W7yPJY/fq/pmYY9cDTaEeehvDNDqGeBp6w8\nr7TX7wt4upbrawky98CrA66PQve/2D2Qd2YI+CTShJ8Tbl4b5Umg3X7fx/AFOudslKcBn6pD\nztfRL4b2P2B2wE+qyfDctyb696+d9P6zdTZq3oYMbgYb8OBKrFIg78xgN2A1B7GG6h7//pqL\no6azbBfqP+74W2tHF1MHG3Aed2/wXQrknRns7jZUBlz/Co1/ndcym2PW+1Y70DVrgg44q34a\nqfliIO/MYHkErTTgyl+rT7jaIf9NR72nQcBzZny+BRwyy/M2zQH3mv1tv0cWVW9OwOGyfQjL\nj4DljZmfEHCoOAbthbbZmSdNCNgbnAT2AgEHih2wHwg4UNZ3wL7MgYUj4DDZ3wETsBvt79Tq\n//9x6QUr+P0hAlfhG/szYAJ2Y5Tu14IJ2A8OZsAE7AYBh8jBIWgCdoOAA8QhaH+Mjl4RcBg4\nB+wPAg4PO2CPEHB4nOyAmQO7MboGmoBD4GYHTMBujN7YgYBD4GYGTMBujN6YhYAD4GgGTMBu\njH4RFgH7z9Ub1BPwDgjYf5xC8hgBe49TSD4jYN+5GkBjH98KJmDl3PXLHHgPBOw3hwNoAt4D\nAfvN4QCagPdAwF5zeQSLgPdAwF5zeQSLgPdAwD7jFJL3CNhjnELyHwH7i34DQMDect0vc+A9\nELCvnO9/CXgPBOwp9+NnAt4DAftph/kvAe+BgL20x/ErAt7Fl4IJWKG/fxx/DgYB+6bMl36D\nQcCe+SPfoBCwX/a7eoM58C4I2Cs7Xn1FwLsgYG/8FbPfHY9eEfAuCNgTVb17Tn8JeBcE7AUB\nh54JeBeSAr6dDtX7RhzSm61V+KgeOnPoOUxyAs5i00usrMIzf5W9h87YlZyAUxNd7tWtxzUy\nqY1VaPc39K9BvEGTE3Bk7t3tu4lsrGK1vz39G/sTVi5z4F3ICXi0Ja+bNRhfm/9Kxdd2+Vi0\ns8/Hv79d/9zfPv4nZDtC+2g+3/8ut/d83gPjPfbAErmdA18f1S3mwBoRsEQuTyMlg0FynFlZ\nBewhYIncngdOq/PA0eHEeWBgC1yJBShGwIBiBIyZmANLRMCYiYAlImDMRMASETBmImCJCBgz\nEbBEBAwoRsCAYgQMKEbAmIk5sEQEjJkIWCICxkwELBEBYyYCloiAMRMBS0TAgGIEDChGwIBi\nBIyZmANLRMCYiYAlImDMRMASETBmImCJhAYMef7bewMwZUFd2werbANWULztijedbbf6hOo2\nYAXF265409l2q0+obgNWULztijedbbf6hOo2YAXF265409l2q0+obgNWULztijedbbf6hOo2\nYAXF265409l2q0+obgNWULztijedbbf6hOo2YAXF265409l2q0+obgNWULztijedbbf6hOo2\nYAXF265409l2q0+obgNWULztijedbbf6hOo2YAXF265409l2q08IwB0CBhQjYEAxAgYUI2BA\nMQIGFCNgQDECBhQjYEAxAgYUI2BAMQIGFCNgQDECBhQjYEAxAgYU2zHg83jdaWSiNNtpW37z\nvKmL35nKsZdvMd9zR+y91vf7DtzH3/2k+vuI99qaXzxv6l3Ji+nlW8z33BGLr/XdvgP3aPSH\nupnoXn7tttf2zPeyqXdz2HN75nrZbr7njth8re8V8Nkkoz9Uaq7F/y/mtNP2/OBlU88atnpi\nu/meu2H1tb5XwCbNR3+og3nkSv5dfdnUsznvuDmzvWw333M3rL7W9wr4no//UM0nGiY1L5t6\nMNejidLdNmiml+3me+6G1df6jn95HgVcSXbbonk8C1jF97xFwIJMhHDJ8yyVPqjzKmAl3/MW\nAQvyZlMz6edjvAq4Jv573vIm4OHJu9EfIJL/Ymq3/d2mSt720st2K/iet7R+z1v2XutSAq6P\nzD0kHxFtt/3dpkp/Mb1st4LveUvr97xl77UuZQh9qs6NXY2C44ovmxqZ8qo48SG8bDffc2fs\nvdalBKz5qqC0/KvI6tPzgnl1JZaS73nLvyux8v4PVX+M9ZwWGGxqte1ZVH1B/I7sebv5njtj\n77UuJuCs+gmN/bbmB4NN7bc9ln9CY3K7+Z67YO+1ruQgAIApBAwoRsCAYgQMKEbAgGIEDChG\nwIBiBAwoRsCAYgQMKEbAgGIEDChGwIBiBAwoRsCAYgQMKEbAgGIEDChGwIBiBAwoRsCAYgQM\nKEbAgGIEDChGwIBiBAwoRsCAYgQMKEbAgGIEDChGwIBiBAwoRsCAYgSsxf0YmeN19uLm6W/2\nOvXFF1kaG5OoeeN7ELAWqanEj5nLP7Uam4kvvsiiei1RtmQTsQMC1uFkomIfmhUfZhb81OrX\nfW/laJLi6R+JSX/eQOyDgFV4tOEezXHeIxYFbEy1683mLQ0B+JtSITWn+kZ2qCao59jE1Y0i\nudgc2g/VHVFzR/G/66EYDqfVZ6b8Qh3m4MGPg4lO/WrG4aZRtUOeWN1gPfk1KWbN8+fm2BYB\nq5CY+/jTUpKXRRWNpu2H/DC4oxx3V9JxwMMHV3PevuDUHB/jlVSz4ZfVDdZzrtfBca+dELAK\n413jxUT3/B6ZS3lHkuXdh2v5IUvMtX6EKZe4VI813UGs5wefTdw/c5FqnN6alRR3HstaX1c3\nWE9U/tNyGT4HXCJgFcYBH8pyyorKO271/bf6jjKvrBzk9o94CvjlwaPnvh7L3e61eq5b+VTR\n1OpG62H4vCcCVmEccPNZP6vtPzTarzyup+Qp4JcHPx+xup2iMtOnfwGeHtytJy1G1fd7jp0Q\nsAqHbg58zX4IOGk/+yXgPL+XI+KZAeench499+QWtkbAKpzaI023QVtTAXePKG8eTXy+PuYH\n3N0aLJd/WF3jmsbMgfdCwCp054GT8nhvOyk9PBd16CekXW7PAb88uK/x0BxMrma+ycsc+DC1\nnn5t2APfeB2O1ZGl8rRt/nRYuLq7+VDdkZ/b1MpDTvd2DvzIJ45CDx6bl7t3cy5G6LfqX4lz\neaQ5fTkK/byeuD7SzR54JwSsRDK8Fnp4Yra6t42wvqPcWzcHmCq3srMy/ZfzwMPH5v0Dku65\nXs4DP6/n0q0CeyBgLS6HoqFL88k56i6Nygcfqiuk6osxqq8ci4fcqrHvLe4Cfn7wcPhb/shT\nt5byAPNjcnWD9dRXYtHvXggYUIyAAcUIGFCMgAHFCBhQjIABxQgYUIyAAcUIGFCMgAHFCBhQ\njIABxQgYUIyAAcUIGFCMgAHFCBhQjIABxQgYUIyAAcUIGFCMgAHFCBhQjIABxQgYUIyAAcUI\nGFCMgAHFCBhQjIABxQgYUOz/B/HEJBtGsgHpAAAAAElFTkSuQmCC",
      "text/plain": [
       "Plot with title \"FDR Plot\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#svglite(filename=\"20220928_FDR_plot_comparison.svg\") #saving the plot as svg\n",
    "#png(\"20220928_FDR_plot_comparison.png\") #saving the plot as png\n",
    "plot(Pos_FDR$Breaks,((Pos_FDR$fdr)*100), xlim=c(-1,1),t='l',lwd=2.0, ylim=c(0,130), xlab = 'Correlation Scores', ylab='FDR', main='FDR Plot')\n",
    "lines(Neg_FDR$Breaks,((Neg_FDR$fdr)*100),lwd=2.0)\n",
    "\n",
    "lines(Neg_FDR_subset$Breaks,((Neg_FDR_subset$fdr)*100),lwd=2.0,col=\"blue\")\n",
    "lines(Pos_FDR_subset$Breaks,((Pos_FDR_subset$fdr)*100),lwd=2.0,col=\"blue\")\n",
    "\n",
    "lines(Neg_FDR_newVal$Breaks,((Neg_FDR_newVal$fdr)*100),lwd=1.0,col=\"green\")\n",
    "lines(Pos_FDR_newVal$Breaks,((Pos_FDR_newVal$fdr)*100),lwd=1.0,col=\"green\")\n",
    "\n",
    "abline(h=0,v=0, col=\"gray\",lty=2)\n",
    "abline(h=c(10,15,20), col=\"red\",lty=2)\n",
    "text(-0.8, c(10,15,20), c(\"10%\",\"15%\",\"20%\"), pos=2, cex=1)\n",
    "legend(0.2,130,   # Position\n",
    "       legend = c(\"Full Range of scores\", \"Subset before FDR calculation\",\"Subset after FDR calculation\"),\n",
    "       bty=\"n\", #specifying the box type is none\n",
    "       col = c(\"black\",\"blue\",\"green\"), #color of the lines\n",
    "       y.intersp= 1.5, #line space\n",
    "       lwd = 1, #line width\n",
    "       cex=1 ) # text size\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 339,
   "id": "b1b3f41e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>Breaks</th><th scope=col>T_C</th><th scope=col>D_C</th><th scope=col>fdr</th><th scope=col>round_fdr</th><th scope=col>round1_fdr</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>-0.79</td><td>0    </td><td>0    </td><td>NaN  </td><td>NaN  </td><td>NaN  </td></tr>\n",
       "\t<tr><td>-0.78</td><td>0    </td><td>0    </td><td>NaN  </td><td>NaN  </td><td>NaN  </td></tr>\n",
       "\t<tr><td>-0.77</td><td>0    </td><td>0    </td><td>NaN  </td><td>NaN  </td><td>NaN  </td></tr>\n",
       "\t<tr><td>-0.76</td><td>1    </td><td>0    </td><td>  0  </td><td>  0  </td><td>  0  </td></tr>\n",
       "\t<tr><td>-0.75</td><td>1    </td><td>0    </td><td>  0  </td><td>  0  </td><td>  0  </td></tr>\n",
       "\t<tr><td>-0.74</td><td>1    </td><td>0    </td><td>  0  </td><td>  0  </td><td>  0  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|llllll}\n",
       " Breaks & T\\_C & D\\_C & fdr & round\\_fdr & round1\\_fdr\\\\\n",
       "\\hline\n",
       "\t -0.79 & 0     & 0     & NaN   & NaN   & NaN  \\\\\n",
       "\t -0.78 & 0     & 0     & NaN   & NaN   & NaN  \\\\\n",
       "\t -0.77 & 0     & 0     & NaN   & NaN   & NaN  \\\\\n",
       "\t -0.76 & 1     & 0     &   0   &   0   &   0  \\\\\n",
       "\t -0.75 & 1     & 0     &   0   &   0   &   0  \\\\\n",
       "\t -0.74 & 1     & 0     &   0   &   0   &   0  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| Breaks | T_C | D_C | fdr | round_fdr | round1_fdr |\n",
       "|---|---|---|---|---|---|\n",
       "| -0.79 | 0     | 0     | NaN   | NaN   | NaN   |\n",
       "| -0.78 | 0     | 0     | NaN   | NaN   | NaN   |\n",
       "| -0.77 | 0     | 0     | NaN   | NaN   | NaN   |\n",
       "| -0.76 | 1     | 0     |   0   |   0   |   0   |\n",
       "| -0.75 | 1     | 0     |   0   |   0   |   0   |\n",
       "| -0.74 | 1     | 0     |   0   |   0   |   0   |\n",
       "\n"
      ],
      "text/plain": [
       "  Breaks T_C D_C fdr round_fdr round1_fdr\n",
       "1 -0.79  0   0   NaN NaN       NaN       \n",
       "2 -0.78  0   0   NaN NaN       NaN       \n",
       "3 -0.77  0   0   NaN NaN       NaN       \n",
       "4 -0.76  1   0     0   0         0       \n",
       "5 -0.75  1   0     0   0         0       \n",
       "6 -0.74  1   0     0   0         0       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "FDR_full <- rbind(Neg_FDR_subset,Pos_FDR_subset) #combining both negative and positive fdr\n",
    "head(FDR_full)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 322,
   "id": "4b95c785",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Creating a Cutoff table\n",
    "Cutoff_table <- data.frame(matrix(ncol = 3, nrow = 3))\n",
    "colnames(Cutoff_table) <- c(\"FDR %\",\"Positive Cutoff\",\"Negative Cutoff\")\n",
    "Cutoff_table$\"FDR %\"<- c(10,15,20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "id": "52a1a5ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol>\n",
       "\t<li>'0.09|0.11|0.12'</li>\n",
       "\t<li>'0.14|0.16'</li>\n",
       "\t<li>'0.19|0.21|0.22'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate}\n",
       "\\item '0.09\\textbar{}0.11\\textbar{}0.12'\n",
       "\\item '0.14\\textbar{}0.16'\n",
       "\\item '0.19\\textbar{}0.21\\textbar{}0.22'\n",
       "\\end{enumerate}\n"
      ],
      "text/markdown": [
       "1. '0.09|0.11|0.12'\n",
       "2. '0.14|0.16'\n",
       "3. '0.19|0.21|0.22'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[[1]]\n",
       "[1] \"0.09|0.11|0.12\"\n",
       "\n",
       "[[2]]\n",
       "[1] \"0.14|0.16\"\n",
       "\n",
       "[[3]]\n",
       "[1] \"0.19|0.21|0.22\"\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# listing the desired FDR values and their range\n",
    "desired_FDR <- list(\"0.09|0.11|0.12\",\"0.14|0.16\",\"0.19|0.21|0.22\")\n",
    "desired_FDR\n",
    "FDR_original <- c(0.10,0.15,0.20)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "112c4f7b",
   "metadata": {},
   "source": [
    "<p style='text-align: justify;'> The below for loop gets the cutoff values using the following steps: <br>\n",
    "1) If the rounded fdr values in Pos_FDR and Neg_FDR table contains the actual values of interested FDR as given in \"FDR_original\" &rarr; 0.10,0.15,0.20 then pick the row. When there are multiple rows with the same FDR values, go for the one with minimum FDR value. <br>\n",
    "2) If the actual values are not present, look for the nearest FDR values as mentioned in the desired_FDR list and slice the corresponding row from the FDR tables as mentioned in step 1 <br>\n",
    "3) Then insert the correlation score (i.e Breaks here) corresponding to the FDR into the cutoff table. When no value is present, mention it by NA</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 437,
   "id": "d8a55aa9",
   "metadata": {},
   "outputs": [],
   "source": [
    "for(i in 1:length(desired_FDR)){\n",
    "    for(j in 1:length(FDR_original)){\n",
    "        if(isTRUE(any(Pos_FDR_subset$round_fdr==FDR_original[i]))){\n",
    "            x <- Pos_FDR_subset %>% filter(grepl(FDR_original[i],round_fdr)) %>% slice(which.min(fdr))} else{\n",
    "            x <- Pos_FDR_subset %>% filter(grepl(desired_FDR[[i]], round_fdr)) %>% slice(which.min(fdr))}\n",
    "        if(isTRUE(any(Neg_FDR_subset$round_fdr==FDR_original[i]))){\n",
    "            y <- Neg_FDR_subset %>% filter(grepl(FDR_original[i],round_fdr)) %>% slice(which.min(fdr))} else{\n",
    "            y <- Neg_FDR_subset %>% filter(grepl(desired_FDR[[i]], round_fdr)) %>% slice(which.min(fdr))}\n",
    "        if(nrow(x)==0){Cutoff_table$\"Positive Cutoff\"[i] <- NA} else{Cutoff_table$\"Positive Cutoff\"[i] <- x$Breaks}\n",
    "        if(nrow(y)==0){Cutoff_table$\"Negative Cutoff\"[i] <- NA} else{Cutoff_table$\"Negative Cutoff\"[i] <- y$Breaks}\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 434,
   "id": "184f0d24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "svglite(filename=\"20220929_FDR_plot_Cutoff.svg\") #saving the plot as svg\n",
    "#png(\"20220928_FDR_plot_comparison.png\") #saving the plot as png\n",
    "plot(FDR_full$Breaks,((FDR_full$fdr)*100), xaxt='n',xlab = 'Correlation Scores', ylab='FDR %', main='FDR Plot',pch=20)  \n",
    "axis(1, at = seq(-0.8,0.8,by = 0.1), las=2)\n",
    "abline(h=0,v=0, col=\"gray\",lty=2)\n",
    "abline(h=c(10,15,20), col=\"red\",lty=2)\n",
    "abline(v=Cutoff_table$'Positive Cutoff', col=\"blue\",lty=2)\n",
    "abline(v=Cutoff_table$'Negative Cutoff', col=\"blue\",lty=2)\n",
    "text(-0.6, c(10,15,20), c(\"10%\",\"15%\",\"20%\"), pos=2, cex=1)\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 438,
   "id": "57ac20bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>FDR %</th><th scope=col>Positive Cutoff</th><th scope=col>Negative Cutoff</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>10   </td><td>0.47 </td><td>-0.25</td></tr>\n",
       "\t<tr><td>15   </td><td>0.73 </td><td>-0.23</td></tr>\n",
       "\t<tr><td>20   </td><td>0.29 </td><td>-0.21</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lll}\n",
       " FDR \\% & Positive Cutoff & Negative Cutoff\\\\\n",
       "\\hline\n",
       "\t 10    & 0.47  & -0.25\\\\\n",
       "\t 15    & 0.73  & -0.23\\\\\n",
       "\t 20    & 0.29  & -0.21\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| FDR % | Positive Cutoff | Negative Cutoff |\n",
       "|---|---|---|\n",
       "| 10    | 0.47  | -0.25 |\n",
       "| 15    | 0.73  | -0.23 |\n",
       "| 20    | 0.29  | -0.21 |\n",
       "\n"
      ],
      "text/plain": [
       "  FDR % Positive Cutoff Negative Cutoff\n",
       "1 10    0.47            -0.25          \n",
       "2 15    0.73            -0.23          \n",
       "3 20    0.29            -0.21          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Cutoff_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "791be2f2",
   "metadata": {},
   "source": [
    "<font color='red'> <b> Indicate the cut-off for negative and positive correlation scores in the below cell: </b></font> Scores only above the cutoff (or below for Neg_FDR) will be selected for further processing."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "id": "3b263f20",
   "metadata": {},
   "outputs": [],
   "source": [
    "Neg_FDR_Cutoff <- -0.2\n",
    "Pos_FDR_Cutoff <- 0.4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8744bbdc",
   "metadata": {},
   "outputs": [],
   "source": [
    "FDR_Decoy<- hist_Decoy %>% filter(correlation_coef < Neg_FDR_Cutoff & correlation_coef > Pos_FDR_Cutoff)\n",
    "FDR_Target <- hist_Target %>% filter(correlation_coef < Neg_FDR_Cutoff & correlation_coef > Pos_FDR_Cutoff )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "fcd902c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(FDR_Decoy,\"20220915_Decoy_subsetted_FDR.csv\",row.names = F,quote = F)\n",
    "write.table(FDR_Target,\"20220915_Target_subsetted_FDR.csv\",row.names = F,quote = F)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
